{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### CONTENT NOTEBOOK\n",
    "\n",
    "IMPORTS\n",
    "\n",
    "VARIABLES\n",
    "\n",
    "FUNCTIONS\n",
    "\n",
    "CLASS\n",
    "\n",
    "MAIN CODE\n",
    "\n",
    "BUILD AND TESTS"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### ----------------------------------------------------------------"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### \n",
    "IMPORTS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "import json\n",
    "import pandas\n",
    "import urllib\n",
    "import requests\n",
    "from urllib import parse, request"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "\n",
    "VARIABLES"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "REST_URL = \"http://data.bioontology.org\"\n",
    "API_KEY = '3b00793b-f3cc-489c-9d6e-7a888f4b656e'\n",
    "PATH_TO_PAPERS = 'data/kaggle/'\n",
    "PATH_TO_ONTS_SELECT = 'data/onts_select.csv'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "\n",
    "FUNCTIONS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def parse_directory():\n",
    "    '''For parse and join paths in PATH_TO_PAPERS.\n",
    "       Output:path_files: list with all paths to record \n",
    "    '''\n",
    "    global path_files\n",
    "    path_files = []\n",
    "\n",
    "    for root, dirs, files in os.walk(PATH_TO_PAPERS):\n",
    "        for file in files:\n",
    "            path_files.append(os.path.join(root, file))\n",
    "\n",
    "    return path_files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_metadata_record(path_to_record):\n",
    "    '''Get dict keys from JSON-PAPER data structure.\n",
    "       Input:path_to_json record.\n",
    "       Output: record.keys.\n",
    "    '''\n",
    "    with open(path_to_record,'r') as json_handler:\n",
    "        record=json.load(json_handler)\n",
    "    return record.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def load_json_record(path_to_record):\n",
    "    ''' Load entire json record for a paper.\n",
    "        Input: path to record.\n",
    "        Output: json_record.\n",
    "    '''\n",
    "    global json_record\n",
    "    with open(path_to_record, 'r') as json_handler:\n",
    "        json_record = json.load(json_handler)\n",
    "    return json_record"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "def load_select_onts():\n",
    "    ''' For reading and proccess ontologies selected file in PATH_TO_ONTS_SELECT.\n",
    "        Output: select_onts.\n",
    "    '''\n",
    "    #proces file.\n",
    "    ont_sel = pandas.read_csv(PATH_TO_ONTS_SELECT)\n",
    "    list_onto = [x.replace('\\t','').replace(' ','').replace('\\n','') for x in list(ont_sel['Ontology'].values)if type(x)!=float]\n",
    "    #\n",
    "    global select_onts\n",
    "    select_onts= []\n",
    "    for i in list_onto:\n",
    "        if i not in select_onts:\n",
    "            select_onts.append(i)\n",
    "        else:\n",
    "            next\n",
    "            \n",
    "    return select_onts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_json(url):\n",
    "    \n",
    "    '''\n",
    "    General util: call some bioportal endpoint.\n",
    "    Header Auth method.\n",
    "    Input:general url bioportal.\n",
    "    Output: json response fo a general urls requests.\n",
    "    \n",
    "    '''\n",
    "    \n",
    "    opener = urllib.request.build_opener()\n",
    "    opener.addheaders = [('Authorization', 'apikey token=' + API_KEY)] \n",
    "    return json.loads(opener.open(url).read())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "def load_bio_onts():\n",
    "    '''Use bioportal endPoint to retrive all Bioportal Onts'''\n",
    "    \n",
    "    resources = get_json(REST_URL + \"/\")                    # call to retrieve json\n",
    "    ontologies = get_json(resources[\"links\"][\"ontologies\"]) # call again for especific fiels of json response\n",
    "    \n",
    "    global bio_onts\n",
    "    bio_onts = []\n",
    "    for ontology in ontologies:\n",
    "        bio_onts.append(f\"{ontology['name']}\\n{ontology['@id']}\\n\")\n",
    "        \n",
    "    return bio_onts    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "def onts():\n",
    "    '''\n",
    "       Merge select_onts and bio_onts,\n",
    "       for  build later string-url-params that cotain only ontlogies\n",
    "       selected and existing in bioportal.\n",
    "       Output: ontologies url query parameter \n",
    "    '''\n",
    "    \n",
    "    select_onts = load_select_onts() # call other function\n",
    "    bio_onts = load_bio_onts()       # call function\n",
    "   \n",
    "    \n",
    "    #build an string characteres that contain a unique words: each word is a \n",
    "    # ontologie selected and existing in bio.\n",
    "    global ontologies\n",
    "    ontologies = '' \n",
    "    \n",
    "    for ontosel in select_onts:          \n",
    "        for ontobio in bio_onts:\n",
    "            if ontobio.find(ontosel) > 1:          # bugOne: the word MESH make match twice and twice is append,\n",
    "                if ontologies.find(ontosel) < 1:   # but they should be appended only once. i don't catch .\n",
    "                    ontologies += ontosel +' '     # the reason of this.\n",
    "                else:\n",
    "                    pass\n",
    "    ontologies = ontologies[:-1].replace(' ',',')\n",
    "    \n",
    "    return ontologies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'MESH,MESH,COVID-19,HPIO,SNOMEDCT,OCHV,PREMEDONTO,NCIT,CIDO,VO,IOBC,BAO,COVID19,MEDDRA,COVIDCRFRAPID,EFO,CODO'"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "onts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def str_list_onts(ontologie_list):\n",
    "    '''General util.\n",
    "       make methamorfosis from list to string-of-elements\n",
    "       Input:ontologies_list\n",
    "       Output: ontologies url query parameter\n",
    "    '''\n",
    "    global ontologies\n",
    "    ontologies = \"\"\n",
    "    for ont in ontologie_list:\n",
    "        ontologies += ','+ ont\n",
    "    return ontologies[1:]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### nota: Es necesario estandarizar el archivo onts_select.csv: \n",
    "- se deben registrar las siglas oficiales de cada ontologia registrada. \n",
    "- se deben incluir ontologias que existan en bioportal.\n",
    "\n",
    "\n",
    "para el caso de \"TheCOVID-19InfectiousDiseaseOntology\" la sigla oficias es IDO-COVID19. vease: https://bioportal.bioontology.org/ontologies/IDO-COVID-19\n",
    "\n",
    "para el caso de: \"NHC\" no se encuentra en bioportal ontologias. vease: \n",
    "https://bioportal.bioontology.org/ontologies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ontologies = \"MESH,COVID-19,SNOMEDCT,COVID19\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "def make_url_query(text_to_annotate):\n",
    "    '''return URL'''\n",
    "    global URL \n",
    "    URL = REST_URL + \"/annotator/?text=%s\"%(parse.quote(text_to_annotate))\n",
    "    return URL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Call api V.2\n",
    "def call_endpoint_annotator(URL, PARAMS):\n",
    "    r = requests.get(url = URL, params = PARAMS)\n",
    "    global json_api_results\n",
    "    json_api_results = r.json()\n",
    "    return json_api_results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### \n",
    "CLASS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "class dbMain:\n",
    "    \n",
    "    '''this class store data for tree schemas.\n",
    "    s1: item collection.\n",
    "    s2:chunks of paper\n",
    "    s2:annotations for each chunk.\n",
    "    contain methods for update schemas an store db main.\n",
    "    Methods:\n",
    "    annotations['matchTerm'].append(matchTerm)\n",
    "    annotations['ontology'].append(ontology)\n",
    "    annotations['annotatedClass'].append(annotatedClass)\n",
    "    chunks['text'].append(text_to_annotate)\n",
    "    chunks['sections'].append(section_text_to_annotated)\n",
    "    chunks['chunks_paper'].append(annotations)\n",
    "    collection['chunks_paper'].append(chunks)\n",
    "    collection['paper_id'] += paper_id\n",
    "    db.append(item_collection).\n",
    "    '''\n",
    "    \n",
    "    db = [] # list of items_collections\n",
    "\n",
    "    def __init__(self):\n",
    "\n",
    "        self.item_collection = {\n",
    "                                'paper_id':'',\n",
    "                                'chunks_paper':[]\n",
    "                                  }\n",
    "        \n",
    "        self.chunks = {\n",
    "                        'texts':[],\n",
    "                        'sections':[],\n",
    "                        'annotations':[]\n",
    "                        }\n",
    "\n",
    "        self.annotations = {\n",
    "                            'matchTerm':[],\n",
    "                            'ontology':[],\n",
    "                            'annotatedClass':[]\n",
    "                            }\n",
    "\n",
    "    #methods\n",
    "    def restart_annotations(self):\n",
    "        self.annotations = {\n",
    "                            'matchTerm':[],\n",
    "                            'ontology':[],\n",
    "                            'annotatedClass':[]\n",
    "                            }\n",
    "    def update_annotations(self,matchTerm,\n",
    "                                ontology,\n",
    "                                annotatedClass):\n",
    "        self.annotations['matchTerm'].append(matchTerm)\n",
    "        self.annotations['ontology'].append(ontology)\n",
    "        self.annotations['annotatedClass'].append(annotatedClass)\n",
    "    \n",
    "    def update_chunks(self,text,section,annotations):\n",
    "        self.chunks['texts'].append(text)\n",
    "        self.chunks['sections'].append(section)\n",
    "        self.chunks['annotations'].append(annotations)\n",
    "\n",
    "    def update_item_collection(self,paper_id,chunks_paper):\n",
    "        self.item_collection['paper_id'] += paper_id\n",
    "        self.item_collection['chunks_paper'].append(chunks_paper)\n",
    "        \n",
    "    def update_db_main(self):\n",
    "        dbMain.db.append(self.item_collection)\n",
    "        \n",
    "    def store_db_main(self):\n",
    "        \n",
    "        with open('annotationsDBlist.json','w') as json_file:\n",
    "            json_file.write(json.dumps(dbMain().db))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "MAIN CODE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Mycoplasma pneumoniae is a common cause of upper and lower respiratory tract infections. It remains one of the most frequent causes of atypical pneumonia particularly among young adults.[1,2,3,4,5] Although it is highly transmissible, most infections caused by this organism are relatively minor and include pharyngitis, tracheobronchitis, bronchiolitis, and croup with one fifth of infections being asymptomatic.[6,7] Only 3 -10% of infected subjects develop symptoms consistent with bronchopneumonia and mortality from infection is rare.[6,7] The organism is fastidious and difficult to grow on cultures. Therefore, diagnosis of infections caused by this organism is usually confirmed with serological tests or polymerase chain reaction-gene amplification techniques. At King Abdulaziz University Hospital (KAUH), Jeddah, Saudi Arabia, the facility to perform Mycoplasma culture has been available since January 1997. As published information concerning M. pneumoniae infections in Saudi Arabia is scarce,[8,9,10] we wished to study the epidemiology and clinical features of culture-proven infections caused by this organism at this hospital.\n",
      "http://data.bioontology.org/annotator/?text=Mycoplasma%20pneumoniae%20is%20a%20common%20cause%20of%20upper%20and%20lower%20respiratory%20tract%20infections.%20It%20remains%20one%20of%20the%20most%20frequent%20causes%20of%20atypical%20pneumonia%20particularly%20among%20young%20adults.%5B1%2C2%2C3%2C4%2C5%5D%20Although%20it%20is%20highly%20transmissible%2C%20most%20infections%20caused%20by%20this%20organism%20are%20relatively%20minor%20and%20include%20pharyngitis%2C%20tracheobronchitis%2C%20bronchiolitis%2C%20and%20croup%20with%20one%20fifth%20of%20infections%20being%20asymptomatic.%5B6%2C7%5D%20Only%203%20-10%25%20of%20infected%20subjects%20develop%20symptoms%20consistent%20with%20bronchopneumonia%20and%20mortality%20from%20infection%20is%20rare.%5B6%2C7%5D%20The%20organism%20is%20fastidious%20and%20difficult%20to%20grow%20on%20cultures.%20Therefore%2C%20diagnosis%20of%20infections%20caused%20by%20this%20organism%20is%20usually%20confirmed%20with%20serological%20tests%20or%20polymerase%20chain%20reaction-gene%20amplification%20techniques.%20At%20King%20Abdulaziz%20University%20Hospital%20%28KAUH%29%2C%20Jeddah%2C%20Saudi%20Arabia%2C%20the%20facility%20to%20perform%20Mycoplasma%20culture%20has%20been%20available%20since%20January%201997.%20As%20published%20information%20concerning%20M.%20pneumoniae%20infections%20in%20Saudi%20Arabia%20is%20scarce%2C%5B8%2C9%2C10%5D%20we%20wished%20to%20study%20the%20epidemiology%20and%20clinical%20features%20of%20culture-proven%20infections%20caused%20by%20this%20organism%20at%20this%20hospital.\n",
      "KAUH is a tertiary care teaching hospital with a bed capacity of 265 beds and annual admissions of 18000 to 19000 patients. Patients with M. pneumoniae positive cultures from respiratory specimens were identified over a 24-months\" period from January, 1997 through December, 1998 for this review.\n",
      "http://data.bioontology.org/annotator/?text=KAUH%20is%20a%20tertiary%20care%20teaching%20hospital%20with%20a%20bed%20capacity%20of%20265%20beds%20and%20annual%20admissions%20of%2018000%20to%2019000%20patients.%20Patients%20with%20M.%20pneumoniae%20positive%20cultures%20from%20respiratory%20specimens%20were%20identified%20over%20a%2024-months%22%20period%20from%20January%2C%201997%20through%20December%2C%201998%20for%20this%20review.\n",
      "During the study period, respiratory specimens (sputum, nasopharyngeal aspiration, endotracheal secretion, and bronchoalveolar lavage) for M. pneumoniae culture were obtained from patients with upper or lower respiratory tract infections seen as inpatients or in the outpatient or emergency departments. Respiratory specimens were aslo Gram-stained and cultured for bacteria and viruses. M. pneumoniae serological tests for IgG or IgM were not available at KAUH during the study period. All positive culture results were obtained from the Microbiology laboratory records. Charts of patients were reviewed with standardized data collection. Information collected included patients' demographics, comorbidities, clinical manifestations, complications, and outcome.\n",
      "http://data.bioontology.org/annotator/?text=During%20the%20study%20period%2C%20respiratory%20specimens%20%28sputum%2C%20nasopharyngeal%20aspiration%2C%20endotracheal%20secretion%2C%20and%20bronchoalveolar%20lavage%29%20for%20M.%20pneumoniae%20culture%20were%20obtained%20from%20patients%20with%20upper%20or%20lower%20respiratory%20tract%20infections%20seen%20as%20inpatients%20or%20in%20the%20outpatient%20or%20emergency%20departments.%20Respiratory%20specimens%20were%20aslo%20Gram-stained%20and%20cultured%20for%20bacteria%20and%20viruses.%20M.%20pneumoniae%20serological%20tests%20for%20IgG%20or%20IgM%20were%20not%20available%20at%20KAUH%20during%20the%20study%20period.%20All%20positive%20culture%20results%20were%20obtained%20from%20the%20Microbiology%20laboratory%20records.%20Charts%20of%20patients%20were%20reviewed%20with%20standardized%20data%20collection.%20Information%20collected%20included%20patients%27%20demographics%2C%20comorbidities%2C%20clinical%20manifestations%2C%20complications%2C%20and%20outcome.\n",
      "M. pneumoniae was cultured using the classic M. pneumoniae agar medium (M.P. agar) and the Pneumofast tray (Pneumofast®, International Microbio, Signes, France). Specimens were processed according to the instructions of the manufacturer. The M.P. agars and Pneumofast trays were incubated anaerobically at 37°C and inspected daily for 4 weeks. The organism was identified based on typical colonial morphology (granular colonies, rarely fried-egg-like, 10-150 ∝ in diameter) on the M.P. agar medium and the change in the Pneumofast broth color from red to orange then to yellow (glucose fermentation) in the absence of turbidity of the broth. Antibiotic sensitivity profile on the Pneumofast tray was also used for identification according to the instructions of the manufacturer. Bacterial and viral cultures were performed using standard methods.\n",
      "http://data.bioontology.org/annotator/?text=M.%20pneumoniae%20was%20cultured%20using%20the%20classic%20M.%20pneumoniae%20agar%20medium%20%28M.P.%20agar%29%20and%20the%20Pneumofast%20tray%20%28Pneumofast%C2%AE%2C%20International%20Microbio%2C%20Signes%2C%20France%29.%20Specimens%20were%20processed%20according%20to%20the%20instructions%20of%20the%20manufacturer.%20The%20M.P.%20agars%20and%20Pneumofast%20trays%20were%20incubated%20anaerobically%20at%2037%C2%B0C%20and%20inspected%20daily%20for%204%20weeks.%20The%20organism%20was%20identified%20based%20on%20typical%20colonial%20morphology%20%28granular%20colonies%2C%20rarely%20fried-egg-like%2C%2010-150%20%E2%88%9D%20in%20diameter%29%20on%20the%20M.P.%20agar%20medium%20and%20the%20change%20in%20the%20Pneumofast%20broth%20color%20from%20red%20to%20orange%20then%20to%20yellow%20%28glucose%20fermentation%29%20in%20the%20absence%20of%20turbidity%20of%20the%20broth.%20Antibiotic%20sensitivity%20profile%20on%20the%20Pneumofast%20tray%20was%20also%20used%20for%20identification%20according%20to%20the%20instructions%20of%20the%20manufacturer.%20Bacterial%20and%20viral%20cultures%20were%20performed%20using%20standard%20methods.\n",
      "M. pneumoniae isolates were considered community-acquired if they were recovered from unhospitalized patients or within 72 hours of admission to the hospital, and nosocomial if they were recovered beyond that period.\n",
      "http://data.bioontology.org/annotator/?text=M.%20pneumoniae%20isolates%20were%20considered%20community-acquired%20if%20they%20were%20recovered%20from%20unhospitalized%20patients%20or%20within%2072%20hours%20of%20admission%20to%20the%20hospital%2C%20and%20nosocomial%20if%20they%20were%20recovered%20beyond%20that%20period.\n",
      "Pneumonia was diagnosed based on clinical symptoms and signs, along with radiographic evidence of pneumonia when possible. Severe pneumonia was defined as pneumonia associated with tachycardia (>140 /minute), tachypnoea (>30/minute), hypotension (Systolic blood pressure <90 mmHg), hypoxemia (arterial oxygen partial pressure <8 kPa or oxygen saturation <90%), and/or more than 2 areas of consolidation.\n",
      "http://data.bioontology.org/annotator/?text=Pneumonia%20was%20diagnosed%20based%20on%20clinical%20symptoms%20and%20signs%2C%20along%20with%20radiographic%20evidence%20of%20pneumonia%20when%20possible.%20Severe%20pneumonia%20was%20defined%20as%20pneumonia%20associated%20with%20tachycardia%20%28%3E140%20/minute%29%2C%20tachypnoea%20%28%3E30/minute%29%2C%20hypotension%20%28Systolic%20blood%20pressure%20%3C90%20mmHg%29%2C%20hypoxemia%20%28arterial%20oxygen%20partial%20pressure%20%3C8%20kPa%20or%20oxygen%20saturation%20%3C90%25%29%2C%20and/or%20more%20than%202%20areas%20of%20consolidation.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Outcome of patients with M. pneumoniae infection was classified into 4 categories; uneventful recovery, recovery following complications, death due to M. pneumoniae infection, or death unrelated to M. pneumoniae infection.\n",
      "http://data.bioontology.org/annotator/?text=Outcome%20of%20patients%20with%20M.%20pneumoniae%20infection%20was%20classified%20into%204%20categories%3B%20uneventful%20recovery%2C%20recovery%20following%20complications%2C%20death%20due%20to%20M.%20pneumoniae%20infection%2C%20or%20death%20unrelated%20to%20M.%20pneumoniae%20infection.\n",
      "The Statistical Package for Social Sciences (SPSS) program was used for data analysis. Comparison of categorical data was by Chi-square statistic or Fisher's exact test for small expected values.\n",
      "http://data.bioontology.org/annotator/?text=The%20Statistical%20Package%20for%20Social%20Sciences%20%28SPSS%29%20program%20was%20used%20for%20data%20analysis.%20Comparison%20of%20categorical%20data%20was%20by%20Chi-square%20statistic%20or%20Fisher%27s%20exact%20test%20for%20small%20expected%20values.\n",
      "A total of 40 respiratory specimens from 40 patients were positive for M. pneumoniae over the 24-months study period. The demographic and epidemiological characteristics of the patients are summarized in Table 1. Of all isolates, 37 (92.5%) were community-acquired and 3 (7.5%) were nosocomial. Thirty-three (82.5%) patients required admission to the hospital and the remaining 7 (17.5%) were treated as outpatients. Twenty-four isolates (60%) were associated with pneumonia, 14 (35%) with upper respiratory tract infections, and 2 (5%) with bronchiolitis. Of the 24 cases of pneumonia, 21 were confirmed radiologically and the remaining 3 were diagnosed clinically. The two cases of bronchiolitis occurred in 2 children, one and three years old. Thirty-one patients (77.5%) had comorbidities. Eleven patients (27.5%) had cardiopulmonary comorbidities (asthma, 8, lung fibrosis, 1, congestive heart failure, 1, congenial heart disease, 1), 9 patients (22.5%) were immunocompromised (malignancy, 7, steroid therapy, 3, Human immunodeficiency virus infection, 1), and 11 patients (27.5%) had other comorbidities (premature newborns, 2, and one each of myelodysplastic syndrome, myeloproliferative disorder, sickle cell anemia, Evan's syndrome, Down syndrome, sarcoidosis, demyelinating disease, cerebral palsy, and spinal muscle atrophy). Organisms concomitantly isolated with M. pneumoniae from the respiratory tract included herpes simplex virus type 1 (2 occasions), adenovirus (2 occasions), cytomegalo virus (1 occasion), respiratory syncytial virus (1 occasion), and bacterial isolates (2 occasions: Acinetobacter species, 1, and Enter obacter cloacae, 1).\n",
      "http://data.bioontology.org/annotator/?text=A%20total%20of%2040%20respiratory%20specimens%20from%2040%20patients%20were%20positive%20for%20M.%20pneumoniae%20over%20the%2024-months%20study%20period.%20The%20demographic%20and%20epidemiological%20characteristics%20of%20the%20patients%20are%20summarized%20in%20Table%201.%20Of%20all%20isolates%2C%2037%20%2892.5%25%29%20were%20community-acquired%20and%203%20%287.5%25%29%20were%20nosocomial.%20Thirty-three%20%2882.5%25%29%20patients%20required%20admission%20to%20the%20hospital%20and%20the%20remaining%207%20%2817.5%25%29%20were%20treated%20as%20outpatients.%20Twenty-four%20isolates%20%2860%25%29%20were%20associated%20with%20pneumonia%2C%2014%20%2835%25%29%20with%20upper%20respiratory%20tract%20infections%2C%20and%202%20%285%25%29%20with%20bronchiolitis.%20Of%20the%2024%20cases%20of%20pneumonia%2C%2021%20were%20confirmed%20radiologically%20and%20the%20remaining%203%20were%20diagnosed%20clinically.%20The%20two%20cases%20of%20bronchiolitis%20occurred%20in%202%20children%2C%20one%20and%20three%20years%20old.%20Thirty-one%20patients%20%2877.5%25%29%20had%20comorbidities.%20Eleven%20patients%20%2827.5%25%29%20had%20cardiopulmonary%20comorbidities%20%28asthma%2C%208%2C%20lung%20fibrosis%2C%201%2C%20congestive%20heart%20failure%2C%201%2C%20congenial%20heart%20disease%2C%201%29%2C%209%20patients%20%2822.5%25%29%20were%20immunocompromised%20%28malignancy%2C%207%2C%20steroid%20therapy%2C%203%2C%20Human%20immunodeficiency%20virus%20infection%2C%201%29%2C%20and%2011%20patients%20%2827.5%25%29%20had%20other%20comorbidities%20%28premature%20newborns%2C%202%2C%20and%20one%20each%20of%20myelodysplastic%20syndrome%2C%20myeloproliferative%20disorder%2C%20sickle%20cell%20anemia%2C%20Evan%27s%20syndrome%2C%20Down%20syndrome%2C%20sarcoidosis%2C%20demyelinating%20disease%2C%20cerebral%20palsy%2C%20and%20spinal%20muscle%20atrophy%29.%20Organisms%20concomitantly%20isolated%20with%20M.%20pneumoniae%20from%20the%20respiratory%20tract%20included%20herpes%20simplex%20virus%20type%201%20%282%20occasions%29%2C%20adenovirus%20%282%20occasions%29%2C%20cytomegalo%20virus%20%281%20occasion%29%2C%20respiratory%20syncytial%20virus%20%281%20occasion%29%2C%20and%20bacterial%20isolates%20%282%20occasions%3A%20Acinetobacter%20species%2C%201%2C%20and%20Enter%20obacter%20cloacae%2C%201%29.\n",
      "Clinical manifestations associated with M. pneumoniae infections are summarized in Table 2. Pneumonia was more common than upper respiratory tract infections (57.5 % versus 27.5%, respectively). Immunocompromised patients were more likely to present with pneumonia as opposed to upper respiratory tract infection or bronchiolitis than non-immunocompromised patients (8/9 versus 16/31, P = 0.05). Similarly, there was a tendency for patients 60 years of age or older to present with pneumonia more frequently than those below 60 (4/4 versus 20/36, P = 0.1). Of the 24 patients with clinically or radiologically confirmed pneumonia, 19 (79.2%) had crepitations and only 6 (25%) had bronchial breath sounds on physical examination. Of the 16 patients in whom wheezes were detected, 9 (56.3%) were not known to have asthma or other obstructive airway disease.\n",
      "http://data.bioontology.org/annotator/?text=Clinical%20manifestations%20associated%20with%20M.%20pneumoniae%20infections%20are%20summarized%20in%20Table%202.%20Pneumonia%20was%20more%20common%20than%20upper%20respiratory%20tract%20infections%20%2857.5%20%25%20versus%2027.5%25%2C%20respectively%29.%20Immunocompromised%20patients%20were%20more%20likely%20to%20present%20with%20pneumonia%20as%20opposed%20to%20upper%20respiratory%20tract%20infection%20or%20bronchiolitis%20than%20non-immunocompromised%20patients%20%288/9%20versus%2016/31%2C%20P%20%3D%200.05%29.%20Similarly%2C%20there%20was%20a%20tendency%20for%20patients%2060%20years%20of%20age%20or%20older%20to%20present%20with%20pneumonia%20more%20frequently%20than%20those%20below%2060%20%284/4%20versus%2020/36%2C%20P%20%3D%200.1%29.%20Of%20the%2024%20patients%20with%20clinically%20or%20radiologically%20confirmed%20pneumonia%2C%2019%20%2879.2%25%29%20had%20crepitations%20and%20only%206%20%2825%25%29%20had%20bronchial%20breath%20sounds%20on%20physical%20examination.%20Of%20the%2016%20patients%20in%20whom%20wheezes%20were%20detected%2C%209%20%2856.3%25%29%20were%20not%20known%20to%20have%20asthma%20or%20other%20obstructive%20airway%20disease.\n",
      "Complications and outcome of patients are shown in Table 3. Of the 24 patients with pneumonia, 21 (87.5%) were admitted to the hospital, and 20 (83.3%) had comorbidities. All patients with upper respiratory tract infections (11 patients) or bronchiolitis (2 patients) had uneventful recovery. Of the 24 patients with pneumonia, 14 (58.3%) had uneventful recovery, 4 (16.7%) recovered following some complications (acute respiratory distress syndrome, 2, respiratory failure, 1, septic shock, 1), 3 (12.5%) died because of M pneumoniae infection, and 3 (12.5%) died due to underlying comorbidities. The 3 patients who died of M pneumoniae pneumonia had other comorbidities; one had congestive heart failure, the second had congenital heart disease, and the third was a 3-months old infant born prematurely at 32 weeks of gestation who previously had 3 episodes of pneumonia due to other pathogens.\n",
      "http://data.bioontology.org/annotator/?text=Complications%20and%20outcome%20of%20patients%20are%20shown%20in%20Table%203.%20Of%20the%2024%20patients%20with%20pneumonia%2C%2021%20%2887.5%25%29%20were%20admitted%20to%20the%20hospital%2C%20and%2020%20%2883.3%25%29%20had%20comorbidities.%20All%20patients%20with%20upper%20respiratory%20tract%20infections%20%2811%20patients%29%20or%20bronchiolitis%20%282%20patients%29%20had%20uneventful%20recovery.%20Of%20the%2024%20patients%20with%20pneumonia%2C%2014%20%2858.3%25%29%20had%20uneventful%20recovery%2C%204%20%2816.7%25%29%20recovered%20following%20some%20complications%20%28acute%20respiratory%20distress%20syndrome%2C%202%2C%20respiratory%20failure%2C%201%2C%20septic%20shock%2C%201%29%2C%203%20%2812.5%25%29%20died%20because%20of%20M%20pneumoniae%20infection%2C%20and%203%20%2812.5%25%29%20died%20due%20to%20underlying%20comorbidities.%20The%203%20patients%20who%20died%20of%20M%20pneumoniae%20pneumonia%20had%20other%20comorbidities%3B%20one%20had%20congestive%20heart%20failure%2C%20the%20second%20had%20congenital%20heart%20disease%2C%20and%20the%20third%20was%20a%203-months%20old%20infant%20born%20prematurely%20at%2032%20weeks%20of%20gestation%20who%20previously%20had%203%20episodes%20of%20pneumonia%20due%20to%20other%20pathogens.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Mycoplasma pneumoniae is one of the most common causes of atypical pneumonia accounting for 5-23% of community-acquired pneumonia,[1,2,3,4,5] In a study of 511 children with acute respiratory tract infection in Riyadh, Saudi Arabia, Mycoplasma pneumoniae was found to be the second most common causative agent after Respiratory syncytial virus (RSV) accounting for 9% of all cases,[8] In a study of 112 adult patients with community acquired pneumonia admitted to a military hospital in Riyadh, Saudi Arabia, this organism accounted for 6% of all cases,[9] In another retrospective study of 567 pneumonic episodes in adult patients from Al-Qassim area, the organism accounted for 23% of all episodes,[10] The organism also causes other relatively minor infections such as pharyngitis, tracheobronchitis, bronchiolitis, and croup. It is transmitted from person-to-person by infected respiratory droplets during close contact. It is most common in school-aged children, military recruits, and college students.[11] Most cases occur singly or as family outbreaks. Larger outbreaks can also occur in closed populations such as military recruit camps or boarding schools,[12] Infection occurs most frequently during the fall and winter in temperate climates but may develop year-round,[13] The average incubation period is 3 weeks following exposure,[6] Although rare, complications are protean and may involve virtually any organ system such as the respiratory system (e.g.: pleurisy, pneumothorax, acute respiratory distress syndrome, lung abscess), the hematologic system (e.g.: hemolytic anemia, intravascular coagulation, thrombocytopenia), the dermatologic system (e.g.: maculopapular or urticarial rashes, erythema multiforme, erythema nodosum), the musculoskeletal system (e.g.: myalgias, arthralgias, arthritis), the cardiovascular system (e.g.: pericarditis, myocarditis), the nervous system (e.g.: meningoencephalitis, Guillain-Barre syndrome, neuropathies, acute psychosis), or the eye (optic disc edema, optic nerve atrophy, retinal exudation and hemorrhages).[6,7,14,15,16,17,18] Immunity following infection is not long lasting.[11]\n",
      "http://data.bioontology.org/annotator/?text=Mycoplasma%20pneumoniae%20is%20one%20of%20the%20most%20common%20causes%20of%20atypical%20pneumonia%20accounting%20for%205-23%25%20of%20community-acquired%20pneumonia%2C%5B1%2C2%2C3%2C4%2C5%5D%20In%20a%20study%20of%20511%20children%20with%20acute%20respiratory%20tract%20infection%20in%20Riyadh%2C%20Saudi%20Arabia%2C%20Mycoplasma%20pneumoniae%20was%20found%20to%20be%20the%20second%20most%20common%20causative%20agent%20after%20Respiratory%20syncytial%20virus%20%28RSV%29%20accounting%20for%209%25%20of%20all%20cases%2C%5B8%5D%20In%20a%20study%20of%20112%20adult%20patients%20with%20community%20acquired%20pneumonia%20admitted%20to%20a%20military%20hospital%20in%20Riyadh%2C%20Saudi%20Arabia%2C%20this%20organism%20accounted%20for%206%25%20of%20all%20cases%2C%5B9%5D%20In%20another%20retrospective%20study%20of%20567%20pneumonic%20episodes%20in%20adult%20patients%20from%20Al-Qassim%20area%2C%20the%20organism%20accounted%20for%2023%25%20of%20all%20episodes%2C%5B10%5D%20The%20organism%20also%20causes%20other%20relatively%20minor%20infections%20such%20as%20pharyngitis%2C%20tracheobronchitis%2C%20bronchiolitis%2C%20and%20croup.%20It%20is%20transmitted%20from%20person-to-person%20by%20infected%20respiratory%20droplets%20during%20close%20contact.%20It%20is%20most%20common%20in%20school-aged%20children%2C%20military%20recruits%2C%20and%20college%20students.%5B11%5D%20Most%20cases%20occur%20singly%20or%20as%20family%20outbreaks.%20Larger%20outbreaks%20can%20also%20occur%20in%20closed%20populations%20such%20as%20military%20recruit%20camps%20or%20boarding%20schools%2C%5B12%5D%20Infection%20occurs%20most%20frequently%20during%20the%20fall%20and%20winter%20in%20temperate%20climates%20but%20may%20develop%20year-round%2C%5B13%5D%20The%20average%20incubation%20period%20is%203%20weeks%20following%20exposure%2C%5B6%5D%20Although%20rare%2C%20complications%20are%20protean%20and%20may%20involve%20virtually%20any%20organ%20system%20such%20as%20the%20respiratory%20system%20%28e.g.%3A%20pleurisy%2C%20pneumothorax%2C%20acute%20respiratory%20distress%20syndrome%2C%20lung%20abscess%29%2C%20the%20hematologic%20system%20%28e.g.%3A%20hemolytic%20anemia%2C%20intravascular%20coagulation%2C%20thrombocytopenia%29%2C%20the%20dermatologic%20system%20%28e.g.%3A%20maculopapular%20or%20urticarial%20rashes%2C%20erythema%20multiforme%2C%20erythema%20nodosum%29%2C%20the%20musculoskeletal%20system%20%28e.g.%3A%20myalgias%2C%20arthralgias%2C%20arthritis%29%2C%20the%20cardiovascular%20system%20%28e.g.%3A%20pericarditis%2C%20myocarditis%29%2C%20the%20nervous%20system%20%28e.g.%3A%20meningoencephalitis%2C%20Guillain-Barre%20syndrome%2C%20neuropathies%2C%20acute%20psychosis%29%2C%20or%20the%20eye%20%28optic%20disc%20edema%2C%20optic%20nerve%20atrophy%2C%20retinal%20exudation%20and%20hemorrhages%29.%5B6%2C7%2C14%2C15%2C16%2C17%2C18%5D%20Immunity%20following%20infection%20is%20not%20long%20lasting.%5B11%5D\n",
      "In our study, the infection affected all age groups but was most common in infants (32.5%) and preschool children (22.5%), and least common in adults aged 15 to 30 years (2.5%) and elderly above 70 years of age (5%). This contrasts with data from temperate countries where the infection is most common in school-aged children, and young adults.[11] One possible explanation for this difference is that infants and preschool children perhaps had more severe infections than did school-aged children, and young adults which prompted presentation of the former group to the hospital. The infection occurred year-round but was most common in the fall (35%), and spring (30%), and least common in the summer (10%). Most infections were community-acquired (92.5%).\n",
      "http://data.bioontology.org/annotator/?text=In%20our%20study%2C%20the%20infection%20affected%20all%20age%20groups%20but%20was%20most%20common%20in%20infants%20%2832.5%25%29%20and%20preschool%20children%20%2822.5%25%29%2C%20and%20least%20common%20in%20adults%20aged%2015%20to%2030%20years%20%282.5%25%29%20and%20elderly%20above%2070%20years%20of%20age%20%285%25%29.%20This%20contrasts%20with%20data%20from%20temperate%20countries%20where%20the%20infection%20is%20most%20common%20in%20school-aged%20children%2C%20and%20young%20adults.%5B11%5D%20One%20possible%20explanation%20for%20this%20difference%20is%20that%20infants%20and%20preschool%20children%20perhaps%20had%20more%20severe%20infections%20than%20did%20school-aged%20children%2C%20and%20young%20adults%20which%20prompted%20presentation%20of%20the%20former%20group%20to%20the%20hospital.%20The%20infection%20occurred%20year-round%20but%20was%20most%20common%20in%20the%20fall%20%2835%25%29%2C%20and%20spring%20%2830%25%29%2C%20and%20least%20common%20in%20the%20summer%20%2810%25%29.%20Most%20infections%20were%20community-acquired%20%2892.5%25%29.\n",
      "More than one half of patients (57.5%) presented with pneumonia, and about a third (27.5%) presented with upper respiratory tract infection, Immunocompromised patients and patients 60 years of age or older were more likely to present with pneumonia as opposed to upper respiratory tract infection than non-immunocompromised patients or those below 60 years of age. Cough (82.5%), fever (75%), and malaise (58.8%) were the most common presenting symptoms. Cough was usually dry or slightly productive of white sputum and mild to moderate in severity. Most febrile patients had mild to moderate fever of 39°C or less; high-grade fever of more than 39°C was rare. Crepitations (60%), and wheezes (40%) were the most common signs. Wheezes were as common in patients with no history of obstructive airway disease (9 patients) as it was in those with such a history (7 patients). Bronchial breathing as a sign of consolidation was detected in only one fourth of patients with pneumonia, which is consistent with the known disparity between clinical and radiological signs of M pneumoniae pneumonia. Crepitations, however, were detected in the majority (79.2%) of patients. Pleuritic chest pain and pleural effusion were rare.\n",
      "http://data.bioontology.org/annotator/?text=More%20than%20one%20half%20of%20patients%20%2857.5%25%29%20presented%20with%20pneumonia%2C%20and%20about%20a%20third%20%2827.5%25%29%20presented%20with%20upper%20respiratory%20tract%20infection%2C%20Immunocompromised%20patients%20and%20patients%2060%20years%20of%20age%20or%20older%20were%20more%20likely%20to%20present%20with%20pneumonia%20as%20opposed%20to%20upper%20respiratory%20tract%20infection%20than%20non-immunocompromised%20patients%20or%20those%20below%2060%20years%20of%20age.%20Cough%20%2882.5%25%29%2C%20fever%20%2875%25%29%2C%20and%20malaise%20%2858.8%25%29%20were%20the%20most%20common%20presenting%20symptoms.%20Cough%20was%20usually%20dry%20or%20slightly%20productive%20of%20white%20sputum%20and%20mild%20to%20moderate%20in%20severity.%20Most%20febrile%20patients%20had%20mild%20to%20moderate%20fever%20of%2039%C2%B0C%20or%20less%3B%20high-grade%20fever%20of%20more%20than%2039%C2%B0C%20was%20rare.%20Crepitations%20%2860%25%29%2C%20and%20wheezes%20%2840%25%29%20were%20the%20most%20common%20signs.%20Wheezes%20were%20as%20common%20in%20patients%20with%20no%20history%20of%20obstructive%20airway%20disease%20%289%20patients%29%20as%20it%20was%20in%20those%20with%20such%20a%20history%20%287%20patients%29.%20Bronchial%20breathing%20as%20a%20sign%20of%20consolidation%20was%20detected%20in%20only%20one%20fourth%20of%20patients%20with%20pneumonia%2C%20which%20is%20consistent%20with%20the%20known%20disparity%20between%20clinical%20and%20radiological%20signs%20of%20M%20pneumoniae%20pneumonia.%20Crepitations%2C%20however%2C%20were%20detected%20in%20the%20majority%20%2879.2%25%29%20of%20patients.%20Pleuritic%20chest%20pain%20and%20pleural%20effusion%20were%20rare.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "More than half (56.5%) of the patients with pneumonia had uneventful recovery. Mortality from M. pneumoniae pneumonia was high (12.5%) and occurred only in patients with underlying comorbidities. None of the 9 patients with no underlying comorbidities died of M pneumoniae pneumonia. The relatively high complications rate (16.7%) and mortality (12.5%) related to M. pneumoniae pneumonia are likely due to selection bias as most patients with pneumonia were sick enough to require admission to the hospital (21/24 or 87.5%) and most of them had comorbidities (20/24 or 83.3%).\n",
      "http://data.bioontology.org/annotator/?text=More%20than%20half%20%2856.5%25%29%20of%20the%20patients%20with%20pneumonia%20had%20uneventful%20recovery.%20Mortality%20from%20M.%20pneumoniae%20pneumonia%20was%20high%20%2812.5%25%29%20and%20occurred%20only%20in%20patients%20with%20underlying%20comorbidities.%20None%20of%20the%209%20patients%20with%20no%20underlying%20comorbidities%20died%20of%20M%20pneumoniae%20pneumonia.%20The%20relatively%20high%20complications%20rate%20%2816.7%25%29%20and%20mortality%20%2812.5%25%29%20related%20to%20M.%20pneumoniae%20pneumonia%20are%20likely%20due%20to%20selection%20bias%20as%20most%20patients%20with%20pneumonia%20were%20sick%20enough%20to%20require%20admission%20to%20the%20hospital%20%2821/24%20or%2087.5%25%29%20and%20most%20of%20them%20had%20comorbidities%20%2820/24%20or%2083.3%25%29.\n",
      "In conclusion, our data shed some light on the epidemiology and clinical features of M pneumoniae infections in one of the Saudi tertiary care centers. The data are comparable to those of other countries except for the finding that infections were more common in infants and preschool children than in school children and young adults. Additionally, mortality attributable to M. pneumoniae pneumonia was relatively high in patients with comorbidities. It is hoped this information will assist clinicians in their approach and management of respiratory tract infections.\n",
      "http://data.bioontology.org/annotator/?text=In%20conclusion%2C%20our%20data%20shed%20some%20light%20on%20the%20epidemiology%20and%20clinical%20features%20of%20M%20pneumoniae%20infections%20in%20one%20of%20the%20Saudi%20tertiary%20care%20centers.%20The%20data%20are%20comparable%20to%20those%20of%20other%20countries%20except%20for%20the%20finding%20that%20infections%20were%20more%20common%20in%20infants%20and%20preschool%20children%20than%20in%20school%20children%20and%20young%20adults.%20Additionally%2C%20mortality%20attributable%20to%20M.%20pneumoniae%20pneumonia%20was%20relatively%20high%20in%20patients%20with%20comorbidities.%20It%20is%20hoped%20this%20information%20will%20assist%20clinicians%20in%20their%20approach%20and%20management%20of%20respiratory%20tract%20infections.\n",
      "The pre-publication history for this paper can be accessed here:\n",
      "http://data.bioontology.org/annotator/?text=The%20pre-publication%20history%20for%20this%20paper%20can%20be%20accessed%20here%3A\n",
      "\n",
      "http://data.bioontology.org/annotator/?text=Clinical%20features%20of%20culture-proven%20Mycoplasma%20pneumoniae%20infections%20at%20King%20Abdulaziz%20University%20Hospital%2C%20Jeddah%2C%20Saudi%20Arabia\n",
      "paper index: 17 processed\n",
      "Surfactant protein-D (SP-D) is a member of the collagenous subfamily of calcium-dependent lectins (collectins) that includes pulmonary surfactant protein A (SP-A) and the serum mannose-binding lectin [1,2,3]. Collectins interact with a wide variety of microorganisms, lipids, and organic particulate antigens, and can modulate the function of immune effector cells and their responses to these ligands. This article reviews what is currently known about the sites of production, structure, function, and regulated expression of SP-D. Emphasis will be placed on functional attributes, known ligand interactions, and structure-function relationships believed to be important for host defense. For additional information on SP-A and other members of the collectin family, the reader is referred to other recent reviews [4,5,6].\n",
      "http://data.bioontology.org/annotator/?text=Surfactant%20protein-D%20%28SP-D%29%20is%20a%20member%20of%20the%20collagenous%20subfamily%20of%20calcium-dependent%20lectins%20%28collectins%29%20that%20includes%20pulmonary%20surfactant%20protein%20A%20%28SP-A%29%20and%20the%20serum%20mannose-binding%20lectin%20%5B1%2C2%2C3%5D.%20Collectins%20interact%20with%20a%20wide%20variety%20of%20microorganisms%2C%20lipids%2C%20and%20organic%20particulate%20antigens%2C%20and%20can%20modulate%20the%20function%20of%20immune%20effector%20cells%20and%20their%20responses%20to%20these%20ligands.%20This%20article%20reviews%20what%20is%20currently%20known%20about%20the%20sites%20of%20production%2C%20structure%2C%20function%2C%20and%20regulated%20expression%20of%20SP-D.%20Emphasis%20will%20be%20placed%20on%20functional%20attributes%2C%20known%20ligand%20interactions%2C%20and%20structure-function%20relationships%20believed%20to%20be%20important%20for%20host%20defense.%20For%20additional%20information%20on%20SP-A%20and%20other%20members%20of%20the%20collectin%20family%2C%20the%20reader%20is%20referred%20to%20other%20recent%20reviews%20%5B4%2C5%2C6%5D.\n",
      "SP-D is synthesized and secreted into the airspaces of the lung by the respiratory epithelium [1]. At the alveolar level, SP-D is constitutively synthesized and secreted by alveolar type II cells. More proximally in the lung, SP-D is secreted by a subset of bronchiolar epithelial cells, the non-ciliated Clara cells. Because SP-D is stored within the secretory granules of Clara cells [7,8], it seems likely that SP-D is subject to regulated secretion via granule exocytosis at this level of the respiratory tract. In some species, SP-D is also synthesized by epithelial cells and/or submucosal glands associated with the bronchi and trachea [9]. Although many alveolar macrophages show strong cytoplasmic and/or membrane labeling with antibody against SP-D, they do not contain detectable SP-D message.\n",
      "http://data.bioontology.org/annotator/?text=SP-D%20is%20synthesized%20and%20secreted%20into%20the%20airspaces%20of%20the%20lung%20by%20the%20respiratory%20epithelium%20%5B1%5D.%20At%20the%20alveolar%20level%2C%20SP-D%20is%20constitutively%20synthesized%20and%20secreted%20by%20alveolar%20type%20II%20cells.%20More%20proximally%20in%20the%20lung%2C%20SP-D%20is%20secreted%20by%20a%20subset%20of%20bronchiolar%20epithelial%20cells%2C%20the%20non-ciliated%20Clara%20cells.%20Because%20SP-D%20is%20stored%20within%20the%20secretory%20granules%20of%20Clara%20cells%20%5B7%2C8%5D%2C%20it%20seems%20likely%20that%20SP-D%20is%20subject%20to%20regulated%20secretion%20via%20granule%20exocytosis%20at%20this%20level%20of%20the%20respiratory%20tract.%20In%20some%20species%2C%20SP-D%20is%20also%20synthesized%20by%20epithelial%20cells%20and/or%20submucosal%20glands%20associated%20with%20the%20bronchi%20and%20trachea%20%5B9%5D.%20Although%20many%20alveolar%20macrophages%20show%20strong%20cytoplasmic%20and/or%20membrane%20labeling%20with%20antibody%20against%20SP-D%2C%20they%20do%20not%20contain%20detectable%20SP-D%20message.\n",
      "The lung seems to be the major site of SP-D production. However, there is increasing evidence for extrapulmonary sites of expression as assessed with monoclonal or affinity-purified antibodies, reverse-transcriptase-mediated PCR (RT-PCR), and/or hybridization assays of tissues from humans and other large mammals [10,11,12,13,14] (summarized in Table 1). It is difficult to entirely exclude cross-reactions or amplification of related sequences; however, localization to many of these sites in human tissues was confirmed by using monoclonal antibodies in combination with RT-PCR with sequencing of the amplified products [10]. Non-pulmonary expression seems to be largely restricted to cells lining epithelial surfaces or ducts and certain glandular epithelial cells that are in direct or indirect continuity with the environment. Notable exceptions to this generalization might include heart, brain, pancreatic islets, and testicular Leydig cells. SP-D has also been identified in amnionic epithelial cells by immunohistochemistry [15]; however, it is unclear whether this is synthesized locally or derived from the lung by way of the amniotic fluid. Interestingly, in many of these sites SP-D microscopically co-localizes with gp-340, an SP-D binding protein and putative SP-D receptor [10].\n",
      "http://data.bioontology.org/annotator/?text=The%20lung%20seems%20to%20be%20the%20major%20site%20of%20SP-D%20production.%20However%2C%20there%20is%20increasing%20evidence%20for%20extrapulmonary%20sites%20of%20expression%20as%20assessed%20with%20monoclonal%20or%20affinity-purified%20antibodies%2C%20reverse-transcriptase-mediated%20PCR%20%28RT-PCR%29%2C%20and/or%20hybridization%20assays%20of%20tissues%20from%20humans%20and%20other%20large%20mammals%20%5B10%2C11%2C12%2C13%2C14%5D%20%28summarized%20in%20Table%201%29.%20It%20is%20difficult%20to%20entirely%20exclude%20cross-reactions%20or%20amplification%20of%20related%20sequences%3B%20however%2C%20localization%20to%20many%20of%20these%20sites%20in%20human%20tissues%20was%20confirmed%20by%20using%20monoclonal%20antibodies%20in%20combination%20with%20RT-PCR%20with%20sequencing%20of%20the%20amplified%20products%20%5B10%5D.%20Non-pulmonary%20expression%20seems%20to%20be%20largely%20restricted%20to%20cells%20lining%20epithelial%20surfaces%20or%20ducts%20and%20certain%20glandular%20epithelial%20cells%20that%20are%20in%20direct%20or%20indirect%20continuity%20with%20the%20environment.%20Notable%20exceptions%20to%20this%20generalization%20might%20include%20heart%2C%20brain%2C%20pancreatic%20islets%2C%20and%20testicular%20Leydig%20cells.%20SP-D%20has%20also%20been%20identified%20in%20amnionic%20epithelial%20cells%20by%20immunohistochemistry%20%5B15%5D%3B%20however%2C%20it%20is%20unclear%20whether%20this%20is%20synthesized%20locally%20or%20derived%20from%20the%20lung%20by%20way%20of%20the%20amniotic%20fluid.%20Interestingly%2C%20in%20many%20of%20these%20sites%20SP-D%20microscopically%20co-localizes%20with%20gp-340%2C%20an%20SP-D%20binding%20protein%20and%20putative%20SP-D%20receptor%20%5B10%5D.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sites of extrapulmonary expression have also been described in small mammals. In the rat, SP-D message was identified in RNA extracted from skin and blood vessel [16], and both protein and message were identified in gastric mucosa [17] and mesentery [13]. Using RT-PCR, SP-D message has also been identified in mouse stomach, heart, and kidney [14].\n",
      "http://data.bioontology.org/annotator/?text=Sites%20of%20extrapulmonary%20expression%20have%20also%20been%20described%20in%20small%20mammals.%20In%20the%20rat%2C%20SP-D%20message%20was%20identified%20in%20RNA%20extracted%20from%20skin%20and%20blood%20vessel%20%5B16%5D%2C%20and%20both%20protein%20and%20message%20were%20identified%20in%20gastric%20mucosa%20%5B17%5D%20and%20mesentery%20%5B13%5D.%20Using%20RT-PCR%2C%20SP-D%20message%20has%20also%20been%20identified%20in%20mouse%20stomach%2C%20heart%2C%20and%20kidney%20%5B14%5D.\n",
      "Recent crystallographic and mutagenesis studies suggest that the structural determinants of saccharide binding are similar to those originally described for mannose-binding lectin [19,20,21,22]. At least two bound calcium ions and two intrachain disulfide crosslinks stabilize the required tertiary structure, and Glu321 and Asn323 within the CRD participate in glucose/mannose type recognition. Interactions with at least one glycolipid ligand, phosphatidylinositol (PI), require the participation of the C-terminal end of the protein [23,24].\n",
      "http://data.bioontology.org/annotator/?text=Recent%20crystallographic%20and%20mutagenesis%20studies%20suggest%20that%20the%20structural%20determinants%20of%20saccharide%20binding%20are%20similar%20to%20those%20originally%20described%20for%20mannose-binding%20lectin%20%5B19%2C20%2C21%2C22%5D.%20At%20least%20two%20bound%20calcium%20ions%20and%20two%20intrachain%20disulfide%20crosslinks%20stabilize%20the%20required%20tertiary%20structure%2C%20and%20Glu321%20and%20Asn323%20within%20the%20CRD%20participate%20in%20glucose/mannose%20type%20recognition.%20Interactions%20with%20at%20least%20one%20glycolipid%20ligand%2C%20phosphatidylinositol%20%28PI%29%2C%20require%20the%20participation%20of%20the%20C-terminal%20end%20of%20the%20protein%20%5B23%2C24%5D.\n",
      "A trimeric cluster of CRDs is necessary for high-affinity binding to carbohydrate ligands [21,25]. The crystal structure of human SP-D suggests that the spatial distribution of CRDs within a trimeric subunit permits simultaneous and cooperative interactions with two or three glycoconjugates displayed on the surface of a particulate ligand [21]. Furthermore, solid-phase binding studies have shown that monomeric CRDs have an approximately 10-fold lower binding affinity for multivalent ligands than trimeric CRDs.\n",
      "http://data.bioontology.org/annotator/?text=A%20trimeric%20cluster%20of%20CRDs%20is%20necessary%20for%20high-affinity%20binding%20to%20carbohydrate%20ligands%20%5B21%2C25%5D.%20The%20crystal%20structure%20of%20human%20SP-D%20suggests%20that%20the%20spatial%20distribution%20of%20CRDs%20within%20a%20trimeric%20subunit%20permits%20simultaneous%20and%20cooperative%20interactions%20with%20two%20or%20three%20glycoconjugates%20displayed%20on%20the%20surface%20of%20a%20particulate%20ligand%20%5B21%5D.%20Furthermore%2C%20solid-phase%20binding%20studies%20have%20shown%20that%20monomeric%20CRDs%20have%20an%20approximately%2010-fold%20lower%20binding%20affinity%20for%20multivalent%20ligands%20than%20trimeric%20CRDs.\n",
      "Crystallographic studies of human SP-D further suggest that the spatial organization of CRDs within a trimer is stabilized by interactions of the C-terminal sequence with the trimeric neck domain [21,26]. Interestingly, the three CRDs show a deviation from threefold asymmetry, suggesting some flexibility of the CRDs in relation to the neck. Thus, the dependence of the binding of PI on the C-terminal sequence could reflect conformational effects, rather than the direct participation of this sequence in ligand interactions.\n",
      "http://data.bioontology.org/annotator/?text=Crystallographic%20studies%20of%20human%20SP-D%20further%20suggest%20that%20the%20spatial%20organization%20of%20CRDs%20within%20a%20trimer%20is%20stabilized%20by%20interactions%20of%20the%20C-terminal%20sequence%20with%20the%20trimeric%20neck%20domain%20%5B21%2C26%5D.%20Interestingly%2C%20the%20three%20CRDs%20show%20a%20deviation%20from%20threefold%20asymmetry%2C%20suggesting%20some%20flexibility%20of%20the%20CRDs%20in%20relation%20to%20the%20neck.%20Thus%2C%20the%20dependence%20of%20the%20binding%20of%20PI%20on%20the%20C-terminal%20sequence%20could%20reflect%20conformational%20effects%2C%20rather%20than%20the%20direct%20participation%20of%20this%20sequence%20in%20ligand%20interactions.\n",
      "The collagen domain length of SP-D is highly conserved and lacks interruptions in the repeating Gly-X-Y sequence(in which X and Y are different amino acids). As for other collagenous proteins, this domain is enriched in imino acids and contains hydroxyproline. Unlike SP-A, SP-D also contains hydroxylysine. Although the collagen domain of rat, human, bovine, and mouse SP-D lacks cysteine residues, cDNA sequencing has identified a codon for cysteine within the collagen domain of pig SP-D [27]; this suggests the possibility of alternative patterns of chain association and oligomeric assembly for pig SP-D.\n",
      "http://data.bioontology.org/annotator/?text=The%20collagen%20domain%20length%20of%20SP-D%20is%20highly%20conserved%20and%20lacks%20interruptions%20in%20the%20repeating%20Gly-X-Y%20sequence%28in%20which%20X%20and%20Y%20are%20different%20amino%20acids%29.%20As%20for%20other%20collagenous%20proteins%2C%20this%20domain%20is%20enriched%20in%20imino%20acids%20and%20contains%20hydroxyproline.%20Unlike%20SP-A%2C%20SP-D%20also%20contains%20hydroxylysine.%20Although%20the%20collagen%20domain%20of%20rat%2C%20human%2C%20bovine%2C%20and%20mouse%20SP-D%20lacks%20cysteine%20residues%2C%20cDNA%20sequencing%20has%20identified%20a%20codon%20for%20cysteine%20within%20the%20collagen%20domain%20of%20pig%20SP-D%20%5B27%5D%3B%20this%20suggests%20the%20possibility%20of%20alternative%20patterns%20of%20chain%20association%20and%20oligomeric%20assembly%20for%20pig%20SP-D.\n",
      "The first translated exon of SP-D contains a highly conserved and unusually hydrophilic Gly-X-Y sequence that shows little homology with the remainder of the collagen sequence. The functional significance of this region is unknown. However, it has been suggested that this region contributes to oligomer assembly or mediates interactions with cellular receptors.\n",
      "http://data.bioontology.org/annotator/?text=The%20first%20translated%20exon%20of%20SP-D%20contains%20a%20highly%20conserved%20and%20unusually%20hydrophilic%20Gly-X-Y%20sequence%20that%20shows%20little%20homology%20with%20the%20remainder%20of%20the%20collagen%20sequence.%20The%20functional%20significance%20of%20this%20region%20is%20unknown.%20However%2C%20it%20has%20been%20suggested%20that%20this%20region%20contributes%20to%20oligomer%20assembly%20or%20mediates%20interactions%20with%20cellular%20receptors.\n",
      "The collagen domain determines the maximal spatial separation of trimeric, C-terminal lectin domains within SP-D molecules, but might also contribute to normal oligomeric assembly and secretion. For example, deletion of the entire collagen domain of rat SP-D results in the secretion of trimers rather than dodecamers [28]. In addition, 2,2-dipyridyl, an inhibitor of prolyl hydroxylation that interferes with the formation of a stable collagen helix, causes the intracellular accumulation of 43 kDa monomers and dimers [29]. In any case, the complete conservation of the number of Gly-X-Y triplets suggests that the spatial separation of trimeric CRDs is critical for normal SP-D function.\n",
      "http://data.bioontology.org/annotator/?text=The%20collagen%20domain%20determines%20the%20maximal%20spatial%20separation%20of%20trimeric%2C%20C-terminal%20lectin%20domains%20within%20SP-D%20molecules%2C%20but%20might%20also%20contribute%20to%20normal%20oligomeric%20assembly%20and%20secretion.%20For%20example%2C%20deletion%20of%20the%20entire%20collagen%20domain%20of%20rat%20SP-D%20results%20in%20the%20secretion%20of%20trimers%20rather%20than%20dodecamers%20%5B28%5D.%20In%20addition%2C%202%2C2-dipyridyl%2C%20an%20inhibitor%20of%20prolyl%20hydroxylation%20that%20interferes%20with%20the%20formation%20of%20a%20stable%20collagen%20helix%2C%20causes%20the%20intracellular%20accumulation%20of%2043%20kDa%20monomers%20and%20dimers%20%5B29%5D.%20In%20any%20case%2C%20the%20complete%20conservation%20of%20the%20number%20of%20Gly-X-Y%20triplets%20suggests%20that%20the%20spatial%20separation%20of%20trimeric%20CRDs%20is%20critical%20for%20normal%20SP-D%20function.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The N-terminal peptide of the mature protein contains two conserved cysteine residues at positions 15 and 20. These residues participate in interchain disulfide crosslinks that stabilize the trimer, as well as the N-terminal association of four or more trimeric subunits. Stable oligomerization of trimeric subunits permits cooperative or bridging interactions between spatially separated binding sites on the same surface or on different particles.\n",
      "http://data.bioontology.org/annotator/?text=The%20N-terminal%20peptide%20of%20the%20mature%20protein%20contains%20two%20conserved%20cysteine%20residues%20at%20positions%2015%20and%2020.%20These%20residues%20participate%20in%20interchain%20disulfide%20crosslinks%20that%20stabilize%20the%20trimer%2C%20as%20well%20as%20the%20N-terminal%20association%20of%20four%20or%20more%20trimeric%20subunits.%20Stable%20oligomerization%20of%20trimeric%20subunits%20permits%20cooperative%20or%20bridging%20interactions%20between%20spatially%20separated%20binding%20sites%20on%20the%20same%20surface%20or%20on%20different%20particles.\n",
      "The process of forming interchain disulfide bonds is complex, and appropriate crosslinking of the N-terminal domains might be rate limiting for secretion [30]. Subcellular fractionation studies suggest that interchain bonds form initially between the three chains of a trimeric subunit. Subsequent rearrangements within the rough endoplasmic reticulum might allow the covalent crosslinking of a single chain from one subunit and two crosslinked chains of another, with the associated elimination of free thiol groups. Mutant proteins that contain unpaired N-terminal cysteine residues are not secreted. However, it is unclear whether this results from abnormalities in disulfide bonding itself, or the failure to stabilize the required N-terminal conformation.\n",
      "http://data.bioontology.org/annotator/?text=The%20process%20of%20forming%20interchain%20disulfide%20bonds%20is%20complex%2C%20and%20appropriate%20crosslinking%20of%20the%20N-terminal%20domains%20might%20be%20rate%20limiting%20for%20secretion%20%5B30%5D.%20Subcellular%20fractionation%20studies%20suggest%20that%20interchain%20bonds%20form%20initially%20between%20the%20three%20chains%20of%20a%20trimeric%20subunit.%20Subsequent%20rearrangements%20within%20the%20rough%20endoplasmic%20reticulum%20might%20allow%20the%20covalent%20crosslinking%20of%20a%20single%20chain%20from%20one%20subunit%20and%20two%20crosslinked%20chains%20of%20another%2C%20with%20the%20associated%20elimination%20of%20free%20thiol%20groups.%20Mutant%20proteins%20that%20contain%20unpaired%20N-terminal%20cysteine%20residues%20are%20not%20secreted.%20However%2C%20it%20is%20unclear%20whether%20this%20results%20from%20abnormalities%20in%20disulfide%20bonding%20itself%2C%20or%20the%20failure%20to%20stabilize%20the%20required%20N-terminal%20conformation.\n",
      "The collagen domain contains hydroxylysyl-derived glycosides and a single N-linked oligosaccharide. In most species (human, rat, mouse, and cow) the site of N-linked glycosylation is located near the N-terminal end of the collagenous domain. Recently, it was shown that pig SP-D has an additional potential site of N-linked glycosylation within the CRD [27]. Although rat and human lung lavage SP-D seem to be sialylated, as suggested by charge heterogeneity and cleavage with highly purified neuraminidase, preparations of human amniotic fluid and bovine lavage SP-D recovered from amniotic fluid showed predominantly complex type biantennary structures and no sialic acid [31].\n",
      "http://data.bioontology.org/annotator/?text=The%20collagen%20domain%20contains%20hydroxylysyl-derived%20glycosides%20and%20a%20single%20N-linked%20oligosaccharide.%20In%20most%20species%20%28human%2C%20rat%2C%20mouse%2C%20and%20cow%29%20the%20site%20of%20N-linked%20glycosylation%20is%20located%20near%20the%20N-terminal%20end%20of%20the%20collagenous%20domain.%20Recently%2C%20it%20was%20shown%20that%20pig%20SP-D%20has%20an%20additional%20potential%20site%20of%20N-linked%20glycosylation%20within%20the%20CRD%20%5B27%5D.%20Although%20rat%20and%20human%20lung%20lavage%20SP-D%20seem%20to%20be%20sialylated%2C%20as%20suggested%20by%20charge%20heterogeneity%20and%20cleavage%20with%20highly%20purified%20neuraminidase%2C%20preparations%20of%20human%20amniotic%20fluid%20and%20bovine%20lavage%20SP-D%20recovered%20from%20amniotic%20fluid%20showed%20predominantly%20complex%20type%20biantennary%20structures%20and%20no%20sialic%20acid%20%5B31%5D.\n",
      "A variant form of SP-D (50 kDa) has been identified in lavage from a subset of human lavage samples; this protein shows O-linked glycosylation of threonyl residues within the N-terminal peptide domain [32]. At present, the functional significance of these sugars is not known. The presence of O-linked glycosylation within the N-terminal domain might be predicted to interfere with normal dodecamer assembly. In this regard, the O-glycosylated 50 kDa form of human SP-D is recovered as trimeric subunits or smaller species.\n",
      "http://data.bioontology.org/annotator/?text=A%20variant%20form%20of%20SP-D%20%2850%20kDa%29%20has%20been%20identified%20in%20lavage%20from%20a%20subset%20of%20human%20lavage%20samples%3B%20this%20protein%20shows%20O-linked%20glycosylation%20of%20threonyl%20residues%20within%20the%20N-terminal%20peptide%20domain%20%5B32%5D.%20At%20present%2C%20the%20functional%20significance%20of%20these%20sugars%20is%20not%20known.%20The%20presence%20of%20O-linked%20glycosylation%20within%20the%20N-terminal%20domain%20might%20be%20predicted%20to%20interfere%20with%20normal%20dodecamer%20assembly.%20In%20this%20regard%2C%20the%20O-glycosylated%2050%20kDa%20form%20of%20human%20SP-D%20is%20recovered%20as%20trimeric%20subunits%20or%20smaller%20species.\n",
      "As for many glycoproteins, the functional role of the attached carbohydrate is unknown. Mutational analysis has shown that the N-linked sugar on rat SP-D is not required for secretion, for dodecamer formation, or for interactions with a variety of microorganisms [29,33].\n",
      "http://data.bioontology.org/annotator/?text=As%20for%20many%20glycoproteins%2C%20the%20functional%20role%20of%20the%20attached%20carbohydrate%20is%20unknown.%20Mutational%20analysis%20has%20shown%20that%20the%20N-linked%20sugar%20on%20rat%20SP-D%20is%20not%20required%20for%20secretion%2C%20for%20dodecamer%20formation%2C%20or%20for%20interactions%20with%20a%20variety%20of%20microorganisms%20%5B29%2C33%5D.\n",
      "Consistent with its designation as a `mannose-type' C-type lectin, SP-D preferentially binds to simple and complex saccharides containing mannose, glucose, or inositol [34,35]. SP-D also interacts with specific constituents of pulmonary surfactant including PI [36,37,38] and glucosylceramide [39]. Binding to glucosylceramide involves interactions of the carbohydrate-binding sequences of the CRD with the glucosyl moiety. However, the interaction of SP-D with PI involves interactions with the lipid, as well as CRD-dependent interactions with the inositol moiety [24,40].\n",
      "http://data.bioontology.org/annotator/?text=Consistent%20with%20its%20designation%20as%20a%20%60mannose-type%27%20C-type%20lectin%2C%20SP-D%20preferentially%20binds%20to%20simple%20and%20complex%20saccharides%20containing%20mannose%2C%20glucose%2C%20or%20inositol%20%5B34%2C35%5D.%20SP-D%20also%20interacts%20with%20specific%20constituents%20of%20pulmonary%20surfactant%20including%20PI%20%5B36%2C37%2C38%5D%20and%20glucosylceramide%20%5B39%5D.%20Binding%20to%20glucosylceramide%20involves%20interactions%20of%20the%20carbohydrate-binding%20sequences%20of%20the%20CRD%20with%20the%20glucosyl%20moiety.%20However%2C%20the%20interaction%20of%20SP-D%20with%20PI%20involves%20interactions%20with%20the%20lipid%2C%20as%20well%20as%20CRD-dependent%20interactions%20with%20the%20inositol%20moiety%20%5B24%2C40%5D.\n",
      "Microorganisms are surfaced with a diverse and complex array of polysaccharides and glycoconjugates, and most classes of microorganism contain one or more sugars recognized by SP-D. However, the outcome of this interaction depends on the specific organism and can be modified by the conditions of microbial growth. The potential consequences of this interaction include the following: varying degrees of lectin-dependent aggregation (namely, microbial agglutination), enhanced binding of microorganisms or microbial aggregates to their `receptors' on host cells, phagocyte activation, and opsonic enhancement of phagocytosis and killing, potentially involving one or more cellular receptors for SP-D.\n",
      "http://data.bioontology.org/annotator/?text=Microorganisms%20are%20surfaced%20with%20a%20diverse%20and%20complex%20array%20of%20polysaccharides%20and%20glycoconjugates%2C%20and%20most%20classes%20of%20microorganism%20contain%20one%20or%20more%20sugars%20recognized%20by%20SP-D.%20However%2C%20the%20outcome%20of%20this%20interaction%20depends%20on%20the%20specific%20organism%20and%20can%20be%20modified%20by%20the%20conditions%20of%20microbial%20growth.%20The%20potential%20consequences%20of%20this%20interaction%20include%20the%20following%3A%20varying%20degrees%20of%20lectin-dependent%20aggregation%20%28namely%2C%20microbial%20agglutination%29%2C%20enhanced%20binding%20of%20microorganisms%20or%20microbial%20aggregates%20to%20their%20%60receptors%27%20on%20host%20cells%2C%20phagocyte%20activation%2C%20and%20opsonic%20enhancement%20of%20phagocytosis%20and%20killing%2C%20potentially%20involving%20one%20or%20more%20cellular%20receptors%20for%20SP-D.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SP-D binds to purified lipopolysaccharide (LPS) isolated from a variety of Gram-negative organisms [35,41]. In addition, LPS is the major cell wall component that is labeled on lectin blotting of outer membranes isolated from Escherichia coli [41]. The latter interactions involve the recognition of the core oligosaccharide domain, which contains glucose and heptose [41]. SP-D interacts preferentially with purified LPS molecules characterized by short or absent O-antigens and preferentially agglutinates bacterial strains expressing a predominance of rough (O-antigen-deficient) LPS [41,44].\n",
      "http://data.bioontology.org/annotator/?text=SP-D%20binds%20to%20purified%20lipopolysaccharide%20%28LPS%29%20isolated%20from%20a%20variety%20of%20Gram-negative%20organisms%20%5B35%2C41%5D.%20In%20addition%2C%20LPS%20is%20the%20major%20cell%20wall%20component%20that%20is%20labeled%20on%20lectin%20blotting%20of%20outer%20membranes%20isolated%20from%20Escherichia%20coli%20%5B41%5D.%20The%20latter%20interactions%20involve%20the%20recognition%20of%20the%20core%20oligosaccharide%20domain%2C%20which%20contains%20glucose%20and%20heptose%20%5B41%5D.%20SP-D%20interacts%20preferentially%20with%20purified%20LPS%20molecules%20characterized%20by%20short%20or%20absent%20O-antigens%20and%20preferentially%20agglutinates%20bacterial%20strains%20expressing%20a%20predominance%20of%20rough%20%28O-antigen-deficient%29%20LPS%20%5B41%2C44%5D.\n",
      "Although the core oligosaccharide domain of LPS constitutes the major ligand for SP-D on at least some Gram-negative bacteria, the mechanism of interaction with this group of microorganisms is probably heterogeneous. SP-D binds to some smooth, unencapsulated strains of Gram-negative bacteria by immunofluorescence. The mechanism is uncertain; the quantity or quality of binding differs from that observed for rough strains and does not necessarily result in agglutination. LPS molecules on the surfaces of bacteria show heterogeneity in the extent of maturation, so it is possible that this interaction is mediated by a subpopulation of LPS with deficient O-antigens and that the density of binding sites is too low for high-affinity binding.\n",
      "http://data.bioontology.org/annotator/?text=Although%20the%20core%20oligosaccharide%20domain%20of%20LPS%20constitutes%20the%20major%20ligand%20for%20SP-D%20on%20at%20least%20some%20Gram-negative%20bacteria%2C%20the%20mechanism%20of%20interaction%20with%20this%20group%20of%20microorganisms%20is%20probably%20heterogeneous.%20SP-D%20binds%20to%20some%20smooth%2C%20unencapsulated%20strains%20of%20Gram-negative%20bacteria%20by%20immunofluorescence.%20The%20mechanism%20is%20uncertain%3B%20the%20quantity%20or%20quality%20of%20binding%20differs%20from%20that%20observed%20for%20rough%20strains%20and%20does%20not%20necessarily%20result%20in%20agglutination.%20LPS%20molecules%20on%20the%20surfaces%20of%20bacteria%20show%20heterogeneity%20in%20the%20extent%20of%20maturation%2C%20so%20it%20is%20possible%20that%20this%20interaction%20is%20mediated%20by%20a%20subpopulation%20of%20LPS%20with%20deficient%20O-antigens%20and%20that%20the%20density%20of%20binding%20sites%20is%20too%20low%20for%20high-affinity%20binding.\n",
      "The recognition of the surface glycoconjugates on Gram-negative bacteria by SP-D depends not only on the expression of lectin-specific residues by a given strain or species, but also on the accessibility of these residues [1,45]. For example, SP-D binds inefficiently to the core region of LPS of encapsulated Klebsiella, but efficiently agglutinates the corresponding unencapsulated phase variants. Interactions of SP-D with the core oligosaccharides of Gram-negative organisms are also influenced by the number of repeating saccharide units associated with the terminal O-antigen of the LPS [41,44].\n",
      "http://data.bioontology.org/annotator/?text=The%20recognition%20of%20the%20surface%20glycoconjugates%20on%20Gram-negative%20bacteria%20by%20SP-D%20depends%20not%20only%20on%20the%20expression%20of%20lectin-specific%20residues%20by%20a%20given%20strain%20or%20species%2C%20but%20also%20on%20the%20accessibility%20of%20these%20residues%20%5B1%2C45%5D.%20For%20example%2C%20SP-D%20binds%20inefficiently%20to%20the%20core%20region%20of%20LPS%20of%20encapsulated%20Klebsiella%2C%20but%20efficiently%20agglutinates%20the%20corresponding%20unencapsulated%20phase%20variants.%20Interactions%20of%20SP-D%20with%20the%20core%20oligosaccharides%20of%20Gram-negative%20organisms%20are%20also%20influenced%20by%20the%20number%20of%20repeating%20saccharide%20units%20associated%20with%20the%20terminal%20O-antigen%20of%20the%20LPS%20%5B41%2C44%5D.\n",
      "Other potential ligands include the O-antigen domain of LPS, certain capsular polysaccharides, and membrane-associated glycoproteins. In this regard, SP-D can bind to di-mannose containing O-antigens expressed by a subset of Klebsiella serotypes (I Ofek, H Sahly and EC Crouch, unpublished data). Although other C-type lectins, specifically SP-A and the mannose receptor, can interact with specific capsular polysaccharides [46], a specific interaction of SP-D with capsular glycoconjugates or exopolysaccharides has not been described.\n",
      "http://data.bioontology.org/annotator/?text=Other%20potential%20ligands%20include%20the%20O-antigen%20domain%20of%20LPS%2C%20certain%20capsular%20polysaccharides%2C%20and%20membrane-associated%20glycoproteins.%20In%20this%20regard%2C%20SP-D%20can%20bind%20to%20di-mannose%20containing%20O-antigens%20expressed%20by%20a%20subset%20of%20Klebsiella%20serotypes%20%28I%20Ofek%2C%20H%20Sahly%20and%20EC%20Crouch%2C%20unpublished%20data%29.%20Although%20other%20C-type%20lectins%2C%20specifically%20SP-A%20and%20the%20mannose%20receptor%2C%20can%20interact%20with%20specific%20capsular%20polysaccharides%20%5B46%5D%2C%20a%20specific%20interaction%20of%20SP-D%20with%20capsular%20glycoconjugates%20or%20exopolysaccharides%20has%20not%20been%20described.\n",
      "The mechanism of interaction with Gram-positive organisms has not been elucidated. Lipoteichoic acids, which are the major glycolipids associated with the Gram-positive cell wall, do not detectably compete with LPS for binding to SP-D (I Ofek, A Mesika, M Kalina, Y Keisari, D Chang, D McGregor and EC Crouch, manuscript submitted). In preliminary studies we observed that binding was competed only partly with maltose and/or EDTA, raising the possibility that binding might be more complex than for some Gram-negative organisms.\n",
      "http://data.bioontology.org/annotator/?text=The%20mechanism%20of%20interaction%20with%20Gram-positive%20organisms%20has%20not%20been%20elucidated.%20Lipoteichoic%20acids%2C%20which%20are%20the%20major%20glycolipids%20associated%20with%20the%20Gram-positive%20cell%20wall%2C%20do%20not%20detectably%20compete%20with%20LPS%20for%20binding%20to%20SP-D%20%28I%20Ofek%2C%20A%20Mesika%2C%20M%20Kalina%2C%20Y%20Keisari%2C%20D%20Chang%2C%20D%20McGregor%20and%20EC%20Crouch%2C%20manuscript%20submitted%29.%20In%20preliminary%20studies%20we%20observed%20that%20binding%20was%20competed%20only%20partly%20with%20maltose%20and/or%20EDTA%2C%20raising%20the%20possibility%20that%20binding%20might%20be%20more%20complex%20than%20for%20some%20Gram-negative%20organisms.\n",
      "Recently, SP-D was shown to interact with the terminal mannosyl-oligosaccharides of lipoarabinomannan from Mycobacterium tuberculosis [43].\n",
      "http://data.bioontology.org/annotator/?text=Recently%2C%20SP-D%20was%20shown%20to%20interact%20with%20the%20terminal%20mannosyl-oligosaccharides%20of%20lipoarabinomannan%20from%20Mycobacterium%20tuberculosis%20%5B43%5D.\n",
      "The consequences of binding with regard to host cell function are not always predictable. Opsonization of mucoid strains of P. aeruginosa with rat SP-D enhanced the internalization by alveolar macrophages, and these experiments further suggested the involvement of a cellular receptor for SP-D [47]. Interestingly, the organisms were not detectably aggregated by SP-D. Similarly, human SP-D enhanced the internalization and killing of unencapsulated phase variants of Klebsiella pneumoniae [45] (I Ofek, A Mesika, M Kalina, Y Keisari, D Chang, D McGregor and EC Crouch, manuscript submitted). However, unlike the situation reported for P. aeruginosa, the organisms showed varying degrees of aggregation; some of the largest aggregates were identified in association with the macrophage membrane, but were not internalized. In other studies, human proteinosis SP-D did not enhance the internalization of E. coli J5, a rough laboratory strain, by rat alveolar macrophages, despite efficient bacterial agglutination [48]. In addition, SP-D inhibited the adherence of Mycobacterium tuberculosis to monocyte-derived macrophages, despite efficient microbial aggregation [43].\n",
      "http://data.bioontology.org/annotator/?text=The%20consequences%20of%20binding%20with%20regard%20to%20host%20cell%20function%20are%20not%20always%20predictable.%20Opsonization%20of%20mucoid%20strains%20of%20P.%20aeruginosa%20with%20rat%20SP-D%20enhanced%20the%20internalization%20by%20alveolar%20macrophages%2C%20and%20these%20experiments%20further%20suggested%20the%20involvement%20of%20a%20cellular%20receptor%20for%20SP-D%20%5B47%5D.%20Interestingly%2C%20the%20organisms%20were%20not%20detectably%20aggregated%20by%20SP-D.%20Similarly%2C%20human%20SP-D%20enhanced%20the%20internalization%20and%20killing%20of%20unencapsulated%20phase%20variants%20of%20Klebsiella%20pneumoniae%20%5B45%5D%20%28I%20Ofek%2C%20A%20Mesika%2C%20M%20Kalina%2C%20Y%20Keisari%2C%20D%20Chang%2C%20D%20McGregor%20and%20EC%20Crouch%2C%20manuscript%20submitted%29.%20However%2C%20unlike%20the%20situation%20reported%20for%20P.%20aeruginosa%2C%20the%20organisms%20showed%20varying%20degrees%20of%20aggregation%3B%20some%20of%20the%20largest%20aggregates%20were%20identified%20in%20association%20with%20the%20macrophage%20membrane%2C%20but%20were%20not%20internalized.%20In%20other%20studies%2C%20human%20proteinosis%20SP-D%20did%20not%20enhance%20the%20internalization%20of%20E.%20coli%20J5%2C%20a%20rough%20laboratory%20strain%2C%20by%20rat%20alveolar%20macrophages%2C%20despite%20efficient%20bacterial%20agglutination%20%5B48%5D.%20In%20addition%2C%20SP-D%20inhibited%20the%20adherence%20of%20Mycobacterium%20tuberculosis%20to%20monocyte-derived%20macrophages%2C%20despite%20efficient%20microbial%20aggregation%20%5B43%5D.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Rat and human SP-D enhanced the binding and uptake of E. coli K12, Streptococcus pneumoniae, and Staphylococcus aureus by human neutrophils [42]. However, similar effects were observed when the neutrophils were preincubated with SP-D, and there was only a slight enhancement of uptake when bacteria were incubated with human SP-D and washed before their addition to neutrophils. Notably, the extent of binding and internalization was dependent on the extent of multimerization, with human SP-D multimers demonstrating the highest potency. Differences in cell type, the extent of SP-D multimerization, or differences in size or organization of bacterial aggregates could account for some of the apparent inconsistencies.\n",
      "http://data.bioontology.org/annotator/?text=Rat%20and%20human%20SP-D%20enhanced%20the%20binding%20and%20uptake%20of%20E.%20coli%20K12%2C%20Streptococcus%20pneumoniae%2C%20and%20Staphylococcus%20aureus%20by%20human%20neutrophils%20%5B42%5D.%20However%2C%20similar%20effects%20were%20observed%20when%20the%20neutrophils%20were%20preincubated%20with%20SP-D%2C%20and%20there%20was%20only%20a%20slight%20enhancement%20of%20uptake%20when%20bacteria%20were%20incubated%20with%20human%20SP-D%20and%20washed%20before%20their%20addition%20to%20neutrophils.%20Notably%2C%20the%20extent%20of%20binding%20and%20internalization%20was%20dependent%20on%20the%20extent%20of%20multimerization%2C%20with%20human%20SP-D%20multimers%20demonstrating%20the%20highest%20potency.%20Differences%20in%20cell%20type%2C%20the%20extent%20of%20SP-D%20multimerization%2C%20or%20differences%20in%20size%20or%20organization%20of%20bacterial%20aggregates%20could%20account%20for%20some%20of%20the%20apparent%20inconsistencies.\n",
      "Although LPS mediates the binding of SP-D to at least some Gram-negative bacteria, SP-D can also bind to specific forms of soluble LPS, particularly `rough' or O-antigen-deficient forms [35,41,44]. Because soluble LPS can modulate a variety of cellular responses and is the key mediator of Gram-negative sepsis, such interactions could have considerable biological significance. Several observations suggest a potential protective role of SP-D in the host response to endotoxin. For example, SP-D can bind to intratracheally instilled LPS in rats, and the resulting complexes are rapidly internalized into lysosomal compartments in alveolar macrophages [49]. In addition, as discussed below, preliminary studies suggest that transgenic mice deficient in SP-D develop more intense inflammation in response to intratracheal challenge with LPS [50]. Of particular interest is the recent observation that SP-D can bind directly to oligosaccharides on sCD14, a soluble form of one component of the LPS receptor complex, and modify its interactions with LPS [51].\n",
      "http://data.bioontology.org/annotator/?text=Although%20LPS%20mediates%20the%20binding%20of%20SP-D%20to%20at%20least%20some%20Gram-negative%20bacteria%2C%20SP-D%20can%20also%20bind%20to%20specific%20forms%20of%20soluble%20LPS%2C%20particularly%20%60rough%27%20or%20O-antigen-deficient%20forms%20%5B35%2C41%2C44%5D.%20Because%20soluble%20LPS%20can%20modulate%20a%20variety%20of%20cellular%20responses%20and%20is%20the%20key%20mediator%20of%20Gram-negative%20sepsis%2C%20such%20interactions%20could%20have%20considerable%20biological%20significance.%20Several%20observations%20suggest%20a%20potential%20protective%20role%20of%20SP-D%20in%20the%20host%20response%20to%20endotoxin.%20For%20example%2C%20SP-D%20can%20bind%20to%20intratracheally%20instilled%20LPS%20in%20rats%2C%20and%20the%20resulting%20complexes%20are%20rapidly%20internalized%20into%20lysosomal%20compartments%20in%20alveolar%20macrophages%20%5B49%5D.%20In%20addition%2C%20as%20discussed%20below%2C%20preliminary%20studies%20suggest%20that%20transgenic%20mice%20deficient%20in%20SP-D%20develop%20more%20intense%20inflammation%20in%20response%20to%20intratracheal%20challenge%20with%20LPS%20%5B50%5D.%20Of%20particular%20interest%20is%20the%20recent%20observation%20that%20SP-D%20can%20bind%20directly%20to%20oligosaccharides%20on%20sCD14%2C%20a%20soluble%20form%20of%20one%20component%20of%20the%20LPS%20receptor%20complex%2C%20and%20modify%20its%20interactions%20with%20LPS%20%5B51%5D.\n",
      "http://data.bioontology.org/annotator/?text=Surfactant%20protein-D%20and%20pulmonary%20host%20defense\n",
      "paper index: 25 processed\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m--------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-62-222873f15cfa>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     19\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 20\u001b[0;31m         \u001b[0;32mfor\u001b[0m \u001b[0midx\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0m_chunk\u001b[0m \u001b[0;32min\u001b[0m \u001b[0menumerate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mjson_record\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'abstract'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     21\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'abstract'",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                        Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-62-222873f15cfa>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     92\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     93\u001b[0m                 \u001b[0;31m#call api here\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 94\u001b[0;31m                 \u001b[0mcall_endpoint_annotator\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mURL\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mPARAMS\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     95\u001b[0m                 \u001b[0;31m#init schema annotations\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     96\u001b[0m                 \u001b[0mitem\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrestart_annotations\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-14-0578c38478e2>\u001b[0m in \u001b[0;36mcall_endpoint_annotator\u001b[0;34m(URL, PARAMS)\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;31m#Call api V.2\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0mcall_endpoint_annotator\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mURL\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mPARAMS\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 3\u001b[0;31m     \u001b[0mr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mrequests\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0murl\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mURL\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mparams\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mPARAMS\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      4\u001b[0m     \u001b[0;32mglobal\u001b[0m \u001b[0mjson_api_results\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m     \u001b[0mjson_api_results\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjson\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.6/site-packages/requests/api.py\u001b[0m in \u001b[0;36mget\u001b[0;34m(url, params, **kwargs)\u001b[0m\n\u001b[1;32m     74\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     75\u001b[0m     \u001b[0mkwargs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msetdefault\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'allow_redirects'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 76\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mrequest\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'get'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mparams\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mparams\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     77\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     78\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.6/site-packages/requests/api.py\u001b[0m in \u001b[0;36mrequest\u001b[0;34m(method, url, **kwargs)\u001b[0m\n\u001b[1;32m     59\u001b[0m     \u001b[0;31m# cases, and look like a memory leak in others.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     60\u001b[0m     \u001b[0;32mwith\u001b[0m \u001b[0msessions\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSession\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0msession\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 61\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0msession\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmethod\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mmethod\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     62\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     63\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.6/site-packages/requests/sessions.py\u001b[0m in \u001b[0;36mrequest\u001b[0;34m(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)\u001b[0m\n\u001b[1;32m    528\u001b[0m         }\n\u001b[1;32m    529\u001b[0m         \u001b[0msend_kwargs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mupdate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msettings\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 530\u001b[0;31m         \u001b[0mresp\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mprep\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0msend_kwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    531\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    532\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mresp\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.6/site-packages/requests/sessions.py\u001b[0m in \u001b[0;36msend\u001b[0;34m(self, request, **kwargs)\u001b[0m\n\u001b[1;32m    683\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    684\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mstream\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 685\u001b[0;31m             \u001b[0mr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcontent\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    686\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    687\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mr\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.6/site-packages/requests/models.py\u001b[0m in \u001b[0;36mcontent\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    827\u001b[0m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_content\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    828\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 829\u001b[0;31m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_content\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34mb''\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0miter_content\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mCONTENT_CHUNK_SIZE\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0;34mb''\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    830\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    831\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_content_consumed\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.6/site-packages/requests/models.py\u001b[0m in \u001b[0;36mgenerate\u001b[0;34m()\u001b[0m\n\u001b[1;32m    749\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mhasattr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mraw\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'stream'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    750\u001b[0m                 \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 751\u001b[0;31m                     \u001b[0;32mfor\u001b[0m \u001b[0mchunk\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mraw\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstream\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mchunk_size\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdecode_content\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    752\u001b[0m                         \u001b[0;32myield\u001b[0m \u001b[0mchunk\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    753\u001b[0m                 \u001b[0;32mexcept\u001b[0m \u001b[0mProtocolError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.6/site-packages/urllib3/response.py\u001b[0m in \u001b[0;36mstream\u001b[0;34m(self, amt, decode_content)\u001b[0m\n\u001b[1;32m    573\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    574\u001b[0m             \u001b[0;32mwhile\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mis_fp_closed\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_fp\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 575\u001b[0;31m                 \u001b[0mdata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mamt\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mamt\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdecode_content\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdecode_content\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    576\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    577\u001b[0m                 \u001b[0;32mif\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.6/site-packages/urllib3/response.py\u001b[0m in \u001b[0;36mread\u001b[0;34m(self, amt, decode_content, cache_content)\u001b[0m\n\u001b[1;32m    516\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    517\u001b[0m                 \u001b[0mcache_content\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mFalse\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 518\u001b[0;31m                 \u001b[0mdata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_fp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mamt\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mfp_closed\u001b[0m \u001b[0;32melse\u001b[0m \u001b[0;34mb\"\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    519\u001b[0m                 if (\n\u001b[1;32m    520\u001b[0m                     \u001b[0mamt\u001b[0m \u001b[0;34m!=\u001b[0m \u001b[0;36m0\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python3.6/http/client.py\u001b[0m in \u001b[0;36mread\u001b[0;34m(self, amt)\u001b[0m\n\u001b[1;32m    457\u001b[0m             \u001b[0;31m# Amount is given, implement using readinto\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    458\u001b[0m             \u001b[0mb\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mbytearray\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mamt\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 459\u001b[0;31m             \u001b[0mn\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreadinto\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mb\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    460\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mmemoryview\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mb\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0mn\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtobytes\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    461\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python3.6/http/client.py\u001b[0m in \u001b[0;36mreadinto\u001b[0;34m(self, b)\u001b[0m\n\u001b[1;32m    501\u001b[0m         \u001b[0;31m# connection, and the user is reading more bytes than will be provided\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    502\u001b[0m         \u001b[0;31m# (for example, reading in 1k chunks)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 503\u001b[0;31m         \u001b[0mn\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreadinto\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mb\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    504\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mn\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mb\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    505\u001b[0m             \u001b[0;31m# Ideally, we would raise IncompleteRead if the content-length\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python3.6/socket.py\u001b[0m in \u001b[0;36mreadinto\u001b[0;34m(self, b)\u001b[0m\n\u001b[1;32m    584\u001b[0m         \u001b[0;32mwhile\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    585\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 586\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_sock\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrecv_into\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mb\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    587\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    588\u001b[0m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_timeout_occurred\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "ontologies = \"MESH,COVID-19,HPIO,SNOMEDCT,OCHV,PREMEDONTO,NCIT,CIDO,VO,IOBC,BAO,COVID19,MEDDRA,COVIDCRFRAPID,EFO,CODO\"\n",
    "\n",
    "PARAMS = {'apikey':API_KEY,'ontologies':ontologies,\n",
    "          'expand_class_hierarchy':'false','class_hierarchy_max_level':0,\n",
    "           'longest_only':'true'\n",
    "         }\n",
    "\n",
    "#parse_directory()\n",
    "\n",
    "parse_directory()\n",
    "\n",
    "for idx, path in enumerate(path_files):\n",
    "    \n",
    "    load_json_record(path)\n",
    "    emty_sections = []\n",
    "    item = dbMain()\n",
    "        \n",
    "    try: \n",
    "        \n",
    "        for idx, _chunk in enumerate(json_record['abstract']):\n",
    "\n",
    "            #make url here\n",
    "            text_to_annotate = _chunk['text']\n",
    "            print(text_to_annotate)\n",
    "\n",
    "            if text_to_annotate:\n",
    "                make_url_query(text_to_annotate)\n",
    "                print(URL)\n",
    "\n",
    "                #call api here\n",
    "                call_endpoint_annotator(URL,PARAMS)\n",
    "                #init db schema annotations\n",
    "                item.restart_annotations()\n",
    "\n",
    "                #proccess and store resultsfor section here\n",
    "                for i in range(len(json_api_results)):\n",
    "                    item.update_annotations(json_api_results[i]['annotations'][0]['text'],\n",
    "                                            json_api_results[i]['annotatedClass']['links']['ontology'],\n",
    "                                            json_api_results[i]['annotatedClass']['@id']\n",
    "                                           )\n",
    "                print(item.annotations)\n",
    "                item.update_chunks(_chunk['text'],\n",
    "                                   _chunk['section'],\n",
    "                                   item.annotations\n",
    "                                  )\n",
    "    \n",
    "            else:\n",
    "                emty_sections.append(idx)\n",
    "\n",
    "        print('ok abstract')\n",
    "        #Init api params\n",
    "\n",
    "        for idx, _chunk in enumerate(json_record['body_text']):\n",
    "\n",
    "            #make url here\n",
    "            text_to_annotate = _chunk['text']\n",
    "            print(text_to_annotate)\n",
    "\n",
    "            if text_to_annotate:\n",
    "                make_url_query(text_to_annotate)\n",
    "                print(URL)\n",
    "\n",
    "                #call api here\n",
    "                call_endpoint_annotator(URL,PARAMS)\n",
    "                #init schema annotations\n",
    "                item.restart_annotations()\n",
    "\n",
    "                #proccess and store resultsfor section here\n",
    "                for i in range(len(json_api_results)):\n",
    "                    item.update_annotations(json_api_results[i]['annotations'][0]['text'],\n",
    "                                            json_api_results[i]['annotatedClass']['links']['ontology'],\n",
    "                                            json_api_results[i]['annotatedClass']['@id']\n",
    "                                           )\n",
    "                item.update_chunks(_chunk['text'],\n",
    "                                   _chunk['section'],\n",
    "                                   item.annotations\n",
    "                                  )\n",
    "            else:\n",
    "                emty_sections.append(idx)\n",
    "                \n",
    "    except KeyError:\n",
    "\n",
    "        for idx, _chunk in enumerate(json_record['body_text']):\n",
    "\n",
    "            #make url here\n",
    "            text_to_annotate = _chunk['text']\n",
    "            print(text_to_annotate)\n",
    "\n",
    "            if text_to_annotate:\n",
    "                make_url_query(text_to_annotate)\n",
    "                print(URL)\n",
    "\n",
    "                #call api here\n",
    "                call_endpoint_annotator(URL,PARAMS)\n",
    "                #init schema annotations\n",
    "                item.restart_annotations()\n",
    "                #proccess and store resultsfor section here\n",
    "                for i in range(len(json_api_results)):\n",
    "                    item.update_annotations(json_api_results[i]['annotations'][0]['text'],\n",
    "                                            json_api_results[i]['annotatedClass']['links']['ontology'],\n",
    "                                            json_api_results[i]['annotatedClass']['@id']\n",
    "\n",
    "                                           )\n",
    "                item.update_chunks(_chunk['text'],\n",
    "                                   _chunk['section'],\n",
    "                                   item.annotations\n",
    "                                  )\n",
    "            else:\n",
    "                emty_sections.append(idx)\n",
    "\n",
    "    finally:\n",
    "        #for title\n",
    "        text_to_annotate = json_record['metadata']['title']\n",
    "        \n",
    "        if text_to_annotate:\n",
    "            make_url_query(text_to_annotate)\n",
    "            print(URL)\n",
    "\n",
    "            #call api here\n",
    "            call_endpoint_annotator(URL,PARAMS)\n",
    "            #init schema annotations\n",
    "            \n",
    "            item.restart_annotations()\n",
    "            #proccess and store resultsfor section here\n",
    "            for i in range(len(json_api_results)):\n",
    "                item.update_annotations(json_api_results[i]['annotations'][0]['text'],\n",
    "                                        json_api_results[i]['annotatedClass']['links']['ontology'],\n",
    "                                        json_api_results[i]['annotatedClass']['@id']\n",
    "\n",
    "                                           )\n",
    "            item.update_chunks(_chunk['text'],\n",
    "                                   _chunk['section'],\n",
    "                                   item.annotations\n",
    "                                  )\n",
    "        item.update_item_collection(json_record['paper_id'], item.chunks)\n",
    "        print('paper index:',idx,'processed')\n",
    "        item.update_db_main()\n",
    "item.store_db_main()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### ---------------------------------------------------\n",
    "BUILD AND TESTS\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### load data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#example parse directory\n",
    "parse_directory()\n",
    "\n",
    "samp_r1=path_files[0]# pmc \n",
    "samp_r2=path_files[49] #pdf_json\n",
    "samp_r1\n",
    "\n",
    "#load_json_record(samp_r1)\n",
    "#load_json_record(samp_r2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "load_json_record(samp_r1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### See json schemes "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "get_metadata_record(samp_r1) #for pmc_json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#schema json for pmc_json\n",
    "{\n",
    "    'paper_id':'str',\n",
    "    'metadata': {'title':'str', 'authors':'list'},\n",
    "    #'abstract': {'text': 'str' ,'section':'str'},\n",
    "    'bodytext': [{'text': 'str', 'section':'str'}]\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "get_metadata_record(samp_r2) #for pdf_json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#eschema json for pdf_json\n",
    "{\n",
    "    'paper_id':'str',\n",
    "    'metadata': {'title':'str', 'authors':'list'},\n",
    "    'abstract': [{'text': 'str' ,'section':'str'}],\n",
    "    'bodytext': [{'text': 'str', 'section':'str'}]\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#data['paper_id']            ####pmc\n",
    "#data['metadata']['title']\n",
    "#data['body_text'] #list of paragraphs\n",
    "\n",
    "#data['paper_id']            #### pdf_json\n",
    "#data['metadata']['title']\n",
    "#data['abstract'] #list of paragraps \n",
    "#data['body_text'] #list of paragraps"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Proposal Nosql dbSchemas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Nosql schemaDB proposal V.2:  schema COLLECTION of texts annotated from bioportal\n",
    "list(Item_collection)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Implementation Nosql schemaDB proposal V.2\n",
    "item_collection = {\n",
    "    'paper_id':'',\n",
    "    'chunks_paper':[{'text':[], \n",
    "                         'section': [], \n",
    "                         'annotations':[{'matchTerm':[],\n",
    "                                         'ontology':[],\n",
    "                                         'annotatedClass':[]}\n",
    "                                       ]\n",
    "                        }]    \n",
    "} "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#schema for each chunk of document for annotate\n",
    "chunks = {'texts':[], 'sections':[], 'annotations': []}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#schema for results Bioportal api: annotations for each chunk.\n",
    "annotations = {'matchTerm':[],             \n",
    "                     'ontology':[],\n",
    "                     'annotatedClass':[]}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#implementations of methods for organization and store data\n",
    "#schema_chunks['text'].append('text_to_annotate')\n",
    "#schema_chunks['section'].append('section_text_to_annotate')\n",
    "#schema_chunks['annotations'].append(schema_annotations)\n",
    "#item_collection['chunks_paper'].append(schema_chunks)\n",
    "#item_collection['paper_id'] = 'paper_id'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Tests"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### test cero"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### call api and retrieve annotations for one fragment of text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "parse_directory()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#prepared url params\n",
    "text_to_annotate = load_json_record(path_files[0])['body_text'][0]['text']\n",
    "text_to_annotate \n",
    "\n",
    "#for test param\n",
    "ontologies = \"MESH,COVID-19,HPIO,SNOMEDCT,OCHV,PREMEDONTO\"\n",
    "ontologies\n",
    "\n",
    "#make url2\n",
    "URL=make_url_query(text_to_annotate)\n",
    "URL\n",
    "\n",
    "#longest_only={true|false}\n",
    "PARAMS = {'apikey':API_KEY,'ontologies':ontologies,\n",
    "          'expand_class_hierarchy':'false','class_hierarchy_max_level':0,\n",
    "           'longest_only':'true'\n",
    "         }\n",
    "#PARAMS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#load_json_record(path_files[0]).keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#init item_collection schema\n",
    "item_collection = {\n",
    "    'paper_id':'',\n",
    "    'chunks_paper':[]    \n",
    "} \n",
    "\n",
    "call_endpoint_annotator(URL,PARAMS)\n",
    "\n",
    "#json_api_results\n",
    "#len(json_api_results)\n",
    "#json_api_results[0].keys()\n",
    "\n",
    "chunks = {'texts':[], 'sections':[], 'annotations': []}\n",
    "\n",
    "# init schema_annotations\n",
    "annotations= {'matchTerm':[],             \n",
    "                     'ontology':[],\n",
    "                     'annotatedClass':[]}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#process results api V.1\n",
    "for i in range(len(json_api_results)):\n",
    "    annotations['matchTerm'].append(json_api_results[i]['annotations'][0]['text'])\n",
    "    annotations['ontology'].append(json_api_results[i]['annotatedClass']['links']['ontology'])\n",
    "    annotations['annotatedClass'].append(json_api_results[i]['annotatedClass']['@id'])\n",
    "chunks['annotations'].append(annotations)\n",
    "chunks['texts'].append(text_to_annotate)\n",
    "chunks['sections'].append(load_json_record(path_files[0])['body_text'][0]['section'])\n",
    "item_collection['chunks_paper'].append(chunks)\n",
    "item_collection['paper_id'] += load_json_record(path_files[0])['paper_id']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "test = item_collection.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Some Tests for schema annotations process results api v.1\n",
    "\n",
    "#for marchTerms\n",
    "test_list_matchTerm = []\n",
    "for result in json_api_results:\n",
    "    test_list_matchTerm.append(result['annotations'][0]['text'])\n",
    "                               \n",
    "#for list ontology\n",
    "test_list_ontology = []\n",
    "for result in json_api_results:\n",
    "    test_list_ontology.append(result['annotatedClass']['links']['ontology'])\n",
    "                               \n",
    "#for annotated class\n",
    "test_list_annotated_class = []\n",
    "for result in json_api_results:\n",
    "    test_list_annotated_class.append(result['annotatedClass']['@id'])\n",
    "\n",
    "assert annotations['matchTerm'] == test_list_matchTerm\n",
    "assert annotations['ontology'] == test_list_ontology\n",
    "assert annotations['annotatedClass'] == test_list_annotated_class"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#test class db\n",
    "item = dbMain()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#process results api v.2: employed class db\n",
    "item.restart_annotations()\n",
    "for i in range(len(json_api_results)):\n",
    "    item.update_annotations(json_api_results[i]['annotations'][0]['text'],\n",
    "                            json_api_results[i]['annotatedClass']['links']['ontology'],\n",
    "                            json_api_results[i]['annotatedClass']['@id']\n",
    "                           )\n",
    "item.update_chunks(text_to_annotate,\n",
    "                         load_json_record(path_files[0])['body_text'][0]['section'],\n",
    "                         item.annotations\n",
    "                         )\n",
    "item.update_item_collection(load_json_record(path_files[0])['paper_id'],\n",
    "                                  item.chunks)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Some Tests for schema annotations process results api v.2\n",
    "\n",
    "#for marchTerms\n",
    "test_list_matchTerm = []\n",
    "for result in json_api_results:\n",
    "    test_list_matchTerm.append(result['annotations'][0]['text'])\n",
    "                               \n",
    "#for list ontology\n",
    "test_list_ontology = []\n",
    "for result in json_api_results:\n",
    "    test_list_ontology.append(result['annotatedClass']['links']['ontology'])\n",
    "                               \n",
    "#for annotated class\n",
    "test_list_annotated_class = []\n",
    "for result in json_api_results:\n",
    "    test_list_annotated_class.append(result['annotatedClass']['@id'])\n",
    "\n",
    "assert annotations['matchTerm'] == item.annotations['matchTerm']\n",
    "assert annotations['ontology'] == item.annotations['ontology']\n",
    "assert annotations['annotatedClass'] == item.annotations['annotatedClass']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "item.annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "item.chunks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "item.item_collection['chunks_paper']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "test['chunks_paper']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#test for results belong process results api v.1 v.2\n",
    "assert test['paper_id'] == item.item_collection['paper_id']\n",
    "assert test['chunks_paper'] == item.item_collection['chunks_paper']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Test one"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### call api for each fragment of text in one pmc_json record"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# for body text pmc_json\n",
    "ontologies = \"MESH,COVID-19,HPIO,SNOMEDCT,OCHV,PREMEDONTO\"\n",
    "\n",
    "PARAMS = {'apikey':API_KEY,'ontologies':ontologies,\n",
    "          'expand_class_hierarchy':'false','class_hierarchy_max_level':0,\n",
    "           'longest_only':'true'\n",
    "         }\n",
    "\n",
    "#init dbs schemas\n",
    "emty_sections = []\n",
    "item_collection = {'paper_id':'',\n",
    "                   'chunks_paper':[]    \n",
    "                  } \n",
    "chunks = {'texts':[], \n",
    "          'sections':[], \n",
    "          'annotations': []\n",
    "         }\n",
    "for idx, _chunk in enumerate(load_json_record(samp_r1)['body_text']):\n",
    "    \n",
    "    #make url here\n",
    "    text_to_annotate = _chunk['text']\n",
    "    print(text_to_annotate)\n",
    "    \n",
    "    if text_to_annotate:\n",
    "        make_url_query(text_to_annotate)\n",
    "        print(URL)\n",
    "\n",
    "        #call api here\n",
    "        call_endpoint_annotator(URL,PARAMS)\n",
    "        #init schema annotations\n",
    "        annotations= {'matchTerm':[],             \n",
    "                         'ontology':[],\n",
    "                         'annotatedClass':[]}\n",
    "        #proccess and store resultsfor section here\n",
    "        for i in range(len(json_api_results)):\n",
    "            annotations['matchTerm'].append(json_api_results[i]['annotations'][0]['text'])\n",
    "            annotations['ontology'].append(json_api_results[i]['annotatedClass']['links']['ontology'])\n",
    "            annotations['annotatedClass'].append(json_api_results[i]['annotatedClass']['@id'])\n",
    "        chunks['texts'].append(_chunk['text'])\n",
    "        chunks['sections'].append(_chunk['section'])\n",
    "        chunks['annotations'].append(annotations)\n",
    "    else:\n",
    "        emty_sections.append(idx)\n",
    "        pass\n",
    "item_collection['paper_id'] += load_json_record(samp_r1)['paper_id']\n",
    "item_collection['chunks_paper'].append(chunks)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "emty_sections"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "test = load_json_record(samp_r1)\n",
    "test['body_text'][17]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "item_collection['chunks_paper'][0]['texts'][1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "item_collection['chunks_paper'][0]['sections']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "item_collection['chunks_paper'][0]['annotations']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# for body text pmc_json\n",
    "ontologies = \"MESH,COVID-19,HPIO,SNOMEDCT,OCHV,PREMEDONTO\"\n",
    "\n",
    "PARAMS = {'apikey':API_KEY,'ontologies':ontologies,\n",
    "          'expand_class_hierarchy':'false','class_hierarchy_max_level':0,\n",
    "           'longest_only':'true'\n",
    "         }\n",
    "\n",
    "#init db class\n",
    "emty_sections = []\n",
    "\n",
    "item = dbMain()\n",
    "for idx, _chunk in enumerate(load_json_record(samp_r1)['body_text']):\n",
    "    \n",
    "    #make url here\n",
    "    text_to_annotate = _chunk['text']\n",
    "    print(text_to_annotate)\n",
    "    \n",
    "    if text_to_annotate:\n",
    "        make_url_query(text_to_annotate)\n",
    "        print(URL)\n",
    "\n",
    "        #call api here\n",
    "        call_endpoint_annotator(URL,PARAMS)\n",
    "        #init schema annotations\n",
    "        item.restart_annotations()\n",
    "        #proccess and store resultsfor section here\n",
    "        for i in range(len(json_api_results)):\n",
    "            item.update_annotations(json_api_results[i]['annotations'][0]['text'],\n",
    "                                    json_api_results[i]['annotatedClass']['links']['ontology'],\n",
    "                                    json_api_results[i]['annotatedClass']['@id']\n",
    "                                    \n",
    "                                   )\n",
    "        item.update_chunks(_chunk['text'],\n",
    "                           _chunk['section'],\n",
    "                           item.annotations\n",
    "                          )\n",
    "    else:\n",
    "        emty_sections.append(idx)\n",
    "        pass\n",
    "item.update_item_collection(load_json_record(samp_r1)['paper_id'],\n",
    "                            item.chunks)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "item.item_collection['chunks_paper'][0]['annotations']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Test two"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### call api for each fragment of text in pdf_json record"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#test for pdf_json\n",
    "ontologies = \"MESH,COVID-19,HPIO,SNOMEDCT,OCHV,PREMEDONTO\"\n",
    "\n",
    "PARAMS = {'apikey':API_KEY,'ontologies':ontologies,\n",
    "          'expand_class_hierarchy':'false','class_hierarchy_max_level':0,\n",
    "           'longest_only':'true'\n",
    "         }\n",
    "\n",
    "#init dbs schemas\n",
    "chunks = {'texts':[], 'sections':[], 'annotations': []}\n",
    "for _chunk in load_json_record(samp_r2)['abstract']:\n",
    "    \n",
    "    chunks['texts'].append(_chunk['text'])  # text to annotate\n",
    "    chunks['sections'].append(_chunk['section'])\n",
    "    \n",
    "    #make url here\n",
    "    text_to_annotate = _chunk['text']\n",
    "    print(text_to_annotate)\n",
    "    \n",
    "    if text_to_annotate:\n",
    "        make_url_query(text_to_annotate)\n",
    "        print(URL)\n",
    "\n",
    "        #call api here\n",
    "        call_endpoint_annotator(URL,PARAMS)\n",
    "        #init schema annotations\n",
    "        annotations= {'matchTerm':[],             \n",
    "                         'ontology':[],\n",
    "                         'annotatedClass':[]}\n",
    "        #proccess and store resultsfor section here\n",
    "        for i in range(len(json_api_results)):\n",
    "            annotations['matchTerm'].append(json_api_results[i]['annotations'][0]['text'])\n",
    "            annotations['ontology'].append(json_api_results[i]['annotatedClass']['links']['ontology'])\n",
    "            annotations['annotatedClass'].append(json_api_results[i]['annotatedClass']['@id'])\n",
    "        chunks['annotations'].append(annotations)\n",
    "    else:\n",
    "        pass\n",
    "    \n",
    "print('ok abstract')\n",
    "#Init api params\n",
    "\n",
    "for _chunk in load_json_record(samp_r2)['body_text']:\n",
    "    \n",
    "    chunks['texts'].append(_chunk['text'])  # text to annotate\n",
    "    chunks['sections'].append(_chunk['section'])\n",
    "    \n",
    "    #make url here\n",
    "    text_to_annotate = _chunk['text']\n",
    "    print(text_to_annotate)\n",
    "    \n",
    "    if text_to_annotate:\n",
    "        make_url_query(text_to_annotate)\n",
    "        print(URL)\n",
    "\n",
    "        #call api here\n",
    "        call_endpoint_annotator(URL,PARAMS)\n",
    "        #init schema annotations\n",
    "        annotations= {'matchTerm':[],             \n",
    "                         'ontology':[],\n",
    "                         'annotatedClass':[]}\n",
    "        #proccess and store resultsfor section here\n",
    "        for i in range(len(json_api_results)):\n",
    "            annotations['matchTerm'].append(json_api_results[i]['annotations'][0]['text'])\n",
    "            annotations['ontology'].append(json_api_results[i]['annotatedClass']['links']['ontology'])\n",
    "            annotations['annotatedClass'].append(json_api_results[i]['annotatedClass']['@id'])\n",
    "        chunks['annotations'].append(annotations)\n",
    "    else:\n",
    "        pass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "test_list_sections = []\n",
    "for chunk_ in load_json_record(samp_r2)['abstract']:\n",
    "    test_list_sections.append(chunk_['section'])\n",
    "for chunk_ in load_json_record(samp_r2)['body_text']:\n",
    "    test_list_sections.append(chunk_['section'])\n",
    "if load_json_record(samp_r2)['metadata']['title']:\n",
    "    test_list_sections.append('title')\n",
    "len(test_list_sections)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#test for pdf_json\n",
    "ontologies = \"MESH,COVID-19,HPIO,SNOMEDCT,OCHV,PREMEDONTO\"\n",
    "\n",
    "PARAMS = {'apikey':API_KEY,'ontologies':ontologies,\n",
    "          'expand_class_hierarchy':'false','class_hierarchy_max_level':0,\n",
    "           'longest_only':'true'\n",
    "         }\n",
    "\n",
    "#init dbs schemas\n",
    "emty_sections = []\n",
    "item = dbMain()\n",
    "for idx,_chunk in enumerate(load_json_record(samp_r2)['abstract']):\n",
    "    \n",
    "    #make url here\n",
    "    text_to_annotate = _chunk['text']\n",
    "    print(text_to_annotate)\n",
    "    \n",
    "    if text_to_annotate:\n",
    "        make_url_query(text_to_annotate)\n",
    "        print(URL)\n",
    "\n",
    "        #call api here\n",
    "        call_endpoint_annotator(URL,PARAMS)\n",
    "        #init db schema annotations\n",
    "        item.restart_annotations()\n",
    "       \n",
    "        #proccess and store resultsfor section here\n",
    "        for i in range(len(json_api_results)):\n",
    "            item.update_annotations(json_api_results[i]['annotations'][0]['text'],\n",
    "                                    json_api_results[i]['annotatedClass']['links']['ontology'],\n",
    "                                    json_api_results[i]['annotatedClass']['@id']\n",
    "                                   )\n",
    "        item.update_chunks(_chunk['text'],\n",
    "                           _chunk['section'],\n",
    "                           item.annotations\n",
    "                          )\n",
    "    else:\n",
    "        emty_sections.append(idx)\n",
    "    \n",
    "print('ok abstract')\n",
    "#Init api params\n",
    "\n",
    "for idx, _chunk in enumerate(load_json_record(samp_r2)['body_text']):\n",
    "    \n",
    "    #make url here\n",
    "    text_to_annotate = _chunk['text']\n",
    "    print(text_to_annotate)\n",
    "    \n",
    "    if text_to_annotate:\n",
    "        make_url_query(text_to_annotate)\n",
    "        print(URL)\n",
    "\n",
    "        #call api here\n",
    "        call_endpoint_annotator(URL,PARAMS)\n",
    "        #init schema annotations\n",
    "        item.restart_annotations()\n",
    "       \n",
    "        #proccess and store resultsfor section here\n",
    "        for i in range(len(json_api_results)):\n",
    "            item.update_annotations(json_api_results[i]['annotations'][0]['text'],\n",
    "                                    json_api_results[i]['annotatedClass']['links']['ontology'],\n",
    "                                    json_api_results[i]['annotatedClass']['@id']\n",
    "                                   )\n",
    "        item.update_chunks(_chunk['text'],\n",
    "                           _chunk['section'],\n",
    "                           item.annotations\n",
    "                          )\n",
    "    else:\n",
    "        emty_sections.append(idx)\n",
    "item.update_db_main()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "test_list_sections = []\n",
    "for chunk_ in load_json_record(samp_r2)['abstract']:\n",
    "    test_list_sections.append(chunk_['section'])\n",
    "for chunk_ in load_json_record(samp_r2)['body_text']:\n",
    "    test_list_sections.append(chunk_['section'])\n",
    "'''if load_json_record(samp_r2)['metadata']['title']:\n",
    "    test_list_sections.append('title')'''\n",
    "len(test_list_sections)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "assert test_list_sections == item.chunks['sections']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Test tree "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Call api for each fragment of text in pcm_json and pdf_json records"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "db = []\n",
    "\n",
    "ontologies = \"MESH,COVID-19,HPIO,SNOMEDCT,OCHV,PREMEDONTO\"\n",
    "\n",
    "PARAMS = {'apikey':API_KEY,'ontologies':ontologies,\n",
    "          'expand_class_hierarchy':'false','class_hierarchy_max_level':0,\n",
    "           'longest_only':'true'\n",
    "         }\n",
    "\n",
    "#parse_directory()\n",
    "\n",
    "path_files = [samp_r1,samp_r2]\n",
    "\n",
    "for idx, path in enumerate(path_files):\n",
    "    \n",
    "    item_collection = {\n",
    "    'paper_id':'',\n",
    "    'chunks_paper':[]    \n",
    "    } \n",
    "         \n",
    "    load_json_record(path)\n",
    "        \n",
    "    try: \n",
    "        \n",
    "        schema_chunks = {'texts':[], 'sections':[], 'annotations': []}\n",
    "        for _chunk in load_json_record(path)['abstract']:\n",
    "    \n",
    "            schema_chunks['texts'].append(_chunk['text'])  # text to annotate\n",
    "            schema_chunks['sections'].append(_chunk['section'])\n",
    "    \n",
    "            #make url here\n",
    "            text_to_annotate = _chunk['text']\n",
    "            print(text_to_annotate)\n",
    "\n",
    "            if text_to_annotate:\n",
    "                make_url_query(text_to_annotate)\n",
    "                print(URL)\n",
    "\n",
    "                #call api here\n",
    "                call_endpoint_annotator(URL,PARAMS)\n",
    "                #init schema annotations\n",
    "                schema_annotations= {'matchTerm':[],             \n",
    "                                 'ontology':[],\n",
    "                                 'annotatedClass':[]}\n",
    "            #proccess and store resultsfor section here\n",
    "                for i in range(len(json_api_results)):\n",
    "                    schema_annotations['matchTerm'].append(json_api_results[i]['annotations'][0]['text'])\n",
    "                    schema_annotations['ontology'].append(json_api_results[i]['annotatedClass']['links']['ontology'])\n",
    "                    schema_annotations['annotatedClass'].append(json_api_results[i]['annotatedClass']['@id'])\n",
    "                schema_chunks['annotations'].append(schema_annotations)\n",
    "            else:\n",
    "                pass\n",
    "    \n",
    "        print('ok abstract')\n",
    "   \n",
    "\n",
    "        for _chunk in load_json_record(path)['body_text']: # fragments of text in body\n",
    "    \n",
    "            schema_chunks['texts'].append(_chunk['text'])  # text to annotate\n",
    "            schema_chunks['sections'].append(_chunk['section'])\n",
    "\n",
    "            #make url here\n",
    "            text_to_annotate = _chunk['text']\n",
    "            print(text_to_annotate)\n",
    "    \n",
    "            if text_to_annotate:\n",
    "                make_url_query(text_to_annotate)\n",
    "                print(URL)\n",
    "\n",
    "                #call api here\n",
    "                call_endpoint_annotator(URL,PARAMS)\n",
    "                #init schema annotations\n",
    "                schema_annotations= {'matchTerm':[],             \n",
    "                                 'ontology':[],\n",
    "                                 'annotatedClass':[]}\n",
    "                #proccess and store resultsfor section here\n",
    "                for i in range(len(json_api_results)):\n",
    "                    schema_annotations['matchTerm'].append(json_api_results[i]['annotations'][0]['text'])\n",
    "                    schema_annotations['ontology'].append(json_api_results[i]['annotatedClass']['links']['ontology'])\n",
    "                    schema_annotations['annotatedClass'].append(json_api_results[i]['annotatedClass']['@id'])\n",
    "                schema_chunks['annotations'].append(schema_annotations)\n",
    "            else:\n",
    "                pass\n",
    "            \n",
    "    except:\n",
    "        schema_chunks = {'texts':[], 'sections':[], 'annotations': []}\n",
    "        for _chunk in load_json_record(path)['body_text']:\n",
    "\n",
    "            schema_chunks['texts'].append(_chunk['text'])  # text to annotate\n",
    "            schema_chunks['sections'].append(_chunk['section'])\n",
    "\n",
    "            #make url here\n",
    "            text_to_annotate = _chunk['text']\n",
    "            print(text_to_annotate)\n",
    "\n",
    "            if text_to_annotate:\n",
    "                make_url_query(text_to_annotate)\n",
    "                print(URL)\n",
    "\n",
    "                #call api here\n",
    "                call_endpoint_annotator(URL,PARAMS)\n",
    "                #init schema annotations\n",
    "                schema_annotations= {'matchTerm':[],             \n",
    "                                 'ontology':[],\n",
    "                                 'annotatedClass':[]}\n",
    "                #proccess and store resultsfor section here\n",
    "                for i in range(len(json_api_results)):\n",
    "                    schema_annotations['matchTerm'].append(json_api_results[i]['annotations'][0]['text'])\n",
    "                    schema_annotations['ontology'].append(json_api_results[i]['annotatedClass']['links']['ontology'])\n",
    "                    schema_annotations['annotatedClass'].append(json_api_results[i]['annotatedClass']['@id'])\n",
    "                schema_chunks['annotations'].append(schema_annotations)\n",
    "            else:\n",
    "                pass\n",
    "                \n",
    "    finally:\n",
    "        #for title\n",
    "        text_to_annotate = json_record['metadata']['title']\n",
    "        \n",
    "        if text_to_annotate:\n",
    "            make_url_query(text_to_annotate)\n",
    "            print(URL)\n",
    "\n",
    "            #call api here\n",
    "            call_endpoint_annotator(URL,PARAMS)\n",
    "            #init schema annotations\n",
    "            schema_annotations= {'matchTerm':[],             \n",
    "                                 'ontology':[],\n",
    "                                 'annotatedClass':[]}\n",
    "            #proccess and store resultsfor section here\n",
    "            for i in range(len(json_api_results)):\n",
    "                schema_annotations['matchTerm'].append(json_api_results[i]['annotations'][0]['text'])\n",
    "                schema_annotations['ontology'].append(json_api_results[i]['annotatedClass']['links']['ontology'])\n",
    "                schema_annotations['annotatedClass'].append(json_api_results[i]['annotatedClass']['@id'])\n",
    "            schema_chunks['annotations'].append(schema_annotations)\n",
    "            schema_chunks['sections'].append('title')\n",
    "            schema_chunks['texts'].append(text_to_annotate)\n",
    "        item_collection['paper_id'] += json_record['paper_id'] \n",
    "        item_collection['chunks_paper'].append(schema_chunks)\n",
    "        print('paper index:',idx,'processed')\n",
    "        db.append(item_collection)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "db[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['.ipynb_checkpoints',\n",
       " '.git',\n",
       " 'README.md',\n",
       " 'SectionPaperAnnotator.ipynb',\n",
       " 'data']"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.listdir('.')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "parse_directory()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "samp_r1 = path_files[0]\n",
    "samp_r2 = path_files[48]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Mycoplasma pneumoniae is a common cause of upper and lower respiratory tract infections. It remains one of the most frequent causes of atypical pneumonia particularly among young adults.[1,2,3,4,5] Although it is highly transmissible, most infections caused by this organism are relatively minor and include pharyngitis, tracheobronchitis, bronchiolitis, and croup with one fifth of infections being asymptomatic.[6,7] Only 3 -10% of infected subjects develop symptoms consistent with bronchopneumonia and mortality from infection is rare.[6,7] The organism is fastidious and difficult to grow on cultures. Therefore, diagnosis of infections caused by this organism is usually confirmed with serological tests or polymerase chain reaction-gene amplification techniques. At King Abdulaziz University Hospital (KAUH), Jeddah, Saudi Arabia, the facility to perform Mycoplasma culture has been available since January 1997. As published information concerning M. pneumoniae infections in Saudi Arabia is scarce,[8,9,10] we wished to study the epidemiology and clinical features of culture-proven infections caused by this organism at this hospital.\n",
      "http://data.bioontology.org/annotator/?text=Mycoplasma%20pneumoniae%20is%20a%20common%20cause%20of%20upper%20and%20lower%20respiratory%20tract%20infections.%20It%20remains%20one%20of%20the%20most%20frequent%20causes%20of%20atypical%20pneumonia%20particularly%20among%20young%20adults.%5B1%2C2%2C3%2C4%2C5%5D%20Although%20it%20is%20highly%20transmissible%2C%20most%20infections%20caused%20by%20this%20organism%20are%20relatively%20minor%20and%20include%20pharyngitis%2C%20tracheobronchitis%2C%20bronchiolitis%2C%20and%20croup%20with%20one%20fifth%20of%20infections%20being%20asymptomatic.%5B6%2C7%5D%20Only%203%20-10%25%20of%20infected%20subjects%20develop%20symptoms%20consistent%20with%20bronchopneumonia%20and%20mortality%20from%20infection%20is%20rare.%5B6%2C7%5D%20The%20organism%20is%20fastidious%20and%20difficult%20to%20grow%20on%20cultures.%20Therefore%2C%20diagnosis%20of%20infections%20caused%20by%20this%20organism%20is%20usually%20confirmed%20with%20serological%20tests%20or%20polymerase%20chain%20reaction-gene%20amplification%20techniques.%20At%20King%20Abdulaziz%20University%20Hospital%20%28KAUH%29%2C%20Jeddah%2C%20Saudi%20Arabia%2C%20the%20facility%20to%20perform%20Mycoplasma%20culture%20has%20been%20available%20since%20January%201997.%20As%20published%20information%20concerning%20M.%20pneumoniae%20infections%20in%20Saudi%20Arabia%20is%20scarce%2C%5B8%2C9%2C10%5D%20we%20wished%20to%20study%20the%20epidemiology%20and%20clinical%20features%20of%20culture-proven%20infections%20caused%20by%20this%20organism%20at%20this%20hospital.\n",
      "KAUH is a tertiary care teaching hospital with a bed capacity of 265 beds and annual admissions of 18000 to 19000 patients. Patients with M. pneumoniae positive cultures from respiratory specimens were identified over a 24-months\" period from January, 1997 through December, 1998 for this review.\n",
      "http://data.bioontology.org/annotator/?text=KAUH%20is%20a%20tertiary%20care%20teaching%20hospital%20with%20a%20bed%20capacity%20of%20265%20beds%20and%20annual%20admissions%20of%2018000%20to%2019000%20patients.%20Patients%20with%20M.%20pneumoniae%20positive%20cultures%20from%20respiratory%20specimens%20were%20identified%20over%20a%2024-months%22%20period%20from%20January%2C%201997%20through%20December%2C%201998%20for%20this%20review.\n",
      "During the study period, respiratory specimens (sputum, nasopharyngeal aspiration, endotracheal secretion, and bronchoalveolar lavage) for M. pneumoniae culture were obtained from patients with upper or lower respiratory tract infections seen as inpatients or in the outpatient or emergency departments. Respiratory specimens were aslo Gram-stained and cultured for bacteria and viruses. M. pneumoniae serological tests for IgG or IgM were not available at KAUH during the study period. All positive culture results were obtained from the Microbiology laboratory records. Charts of patients were reviewed with standardized data collection. Information collected included patients' demographics, comorbidities, clinical manifestations, complications, and outcome.\n",
      "http://data.bioontology.org/annotator/?text=During%20the%20study%20period%2C%20respiratory%20specimens%20%28sputum%2C%20nasopharyngeal%20aspiration%2C%20endotracheal%20secretion%2C%20and%20bronchoalveolar%20lavage%29%20for%20M.%20pneumoniae%20culture%20were%20obtained%20from%20patients%20with%20upper%20or%20lower%20respiratory%20tract%20infections%20seen%20as%20inpatients%20or%20in%20the%20outpatient%20or%20emergency%20departments.%20Respiratory%20specimens%20were%20aslo%20Gram-stained%20and%20cultured%20for%20bacteria%20and%20viruses.%20M.%20pneumoniae%20serological%20tests%20for%20IgG%20or%20IgM%20were%20not%20available%20at%20KAUH%20during%20the%20study%20period.%20All%20positive%20culture%20results%20were%20obtained%20from%20the%20Microbiology%20laboratory%20records.%20Charts%20of%20patients%20were%20reviewed%20with%20standardized%20data%20collection.%20Information%20collected%20included%20patients%27%20demographics%2C%20comorbidities%2C%20clinical%20manifestations%2C%20complications%2C%20and%20outcome.\n",
      "M. pneumoniae was cultured using the classic M. pneumoniae agar medium (M.P. agar) and the Pneumofast tray (Pneumofast®, International Microbio, Signes, France). Specimens were processed according to the instructions of the manufacturer. The M.P. agars and Pneumofast trays were incubated anaerobically at 37°C and inspected daily for 4 weeks. The organism was identified based on typical colonial morphology (granular colonies, rarely fried-egg-like, 10-150 ∝ in diameter) on the M.P. agar medium and the change in the Pneumofast broth color from red to orange then to yellow (glucose fermentation) in the absence of turbidity of the broth. Antibiotic sensitivity profile on the Pneumofast tray was also used for identification according to the instructions of the manufacturer. Bacterial and viral cultures were performed using standard methods.\n",
      "http://data.bioontology.org/annotator/?text=M.%20pneumoniae%20was%20cultured%20using%20the%20classic%20M.%20pneumoniae%20agar%20medium%20%28M.P.%20agar%29%20and%20the%20Pneumofast%20tray%20%28Pneumofast%C2%AE%2C%20International%20Microbio%2C%20Signes%2C%20France%29.%20Specimens%20were%20processed%20according%20to%20the%20instructions%20of%20the%20manufacturer.%20The%20M.P.%20agars%20and%20Pneumofast%20trays%20were%20incubated%20anaerobically%20at%2037%C2%B0C%20and%20inspected%20daily%20for%204%20weeks.%20The%20organism%20was%20identified%20based%20on%20typical%20colonial%20morphology%20%28granular%20colonies%2C%20rarely%20fried-egg-like%2C%2010-150%20%E2%88%9D%20in%20diameter%29%20on%20the%20M.P.%20agar%20medium%20and%20the%20change%20in%20the%20Pneumofast%20broth%20color%20from%20red%20to%20orange%20then%20to%20yellow%20%28glucose%20fermentation%29%20in%20the%20absence%20of%20turbidity%20of%20the%20broth.%20Antibiotic%20sensitivity%20profile%20on%20the%20Pneumofast%20tray%20was%20also%20used%20for%20identification%20according%20to%20the%20instructions%20of%20the%20manufacturer.%20Bacterial%20and%20viral%20cultures%20were%20performed%20using%20standard%20methods.\n",
      "M. pneumoniae isolates were considered community-acquired if they were recovered from unhospitalized patients or within 72 hours of admission to the hospital, and nosocomial if they were recovered beyond that period.\n",
      "http://data.bioontology.org/annotator/?text=M.%20pneumoniae%20isolates%20were%20considered%20community-acquired%20if%20they%20were%20recovered%20from%20unhospitalized%20patients%20or%20within%2072%20hours%20of%20admission%20to%20the%20hospital%2C%20and%20nosocomial%20if%20they%20were%20recovered%20beyond%20that%20period.\n",
      "Pneumonia was diagnosed based on clinical symptoms and signs, along with radiographic evidence of pneumonia when possible. Severe pneumonia was defined as pneumonia associated with tachycardia (>140 /minute), tachypnoea (>30/minute), hypotension (Systolic blood pressure <90 mmHg), hypoxemia (arterial oxygen partial pressure <8 kPa or oxygen saturation <90%), and/or more than 2 areas of consolidation.\n",
      "http://data.bioontology.org/annotator/?text=Pneumonia%20was%20diagnosed%20based%20on%20clinical%20symptoms%20and%20signs%2C%20along%20with%20radiographic%20evidence%20of%20pneumonia%20when%20possible.%20Severe%20pneumonia%20was%20defined%20as%20pneumonia%20associated%20with%20tachycardia%20%28%3E140%20/minute%29%2C%20tachypnoea%20%28%3E30/minute%29%2C%20hypotension%20%28Systolic%20blood%20pressure%20%3C90%20mmHg%29%2C%20hypoxemia%20%28arterial%20oxygen%20partial%20pressure%20%3C8%20kPa%20or%20oxygen%20saturation%20%3C90%25%29%2C%20and/or%20more%20than%202%20areas%20of%20consolidation.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Outcome of patients with M. pneumoniae infection was classified into 4 categories; uneventful recovery, recovery following complications, death due to M. pneumoniae infection, or death unrelated to M. pneumoniae infection.\n",
      "http://data.bioontology.org/annotator/?text=Outcome%20of%20patients%20with%20M.%20pneumoniae%20infection%20was%20classified%20into%204%20categories%3B%20uneventful%20recovery%2C%20recovery%20following%20complications%2C%20death%20due%20to%20M.%20pneumoniae%20infection%2C%20or%20death%20unrelated%20to%20M.%20pneumoniae%20infection.\n",
      "The Statistical Package for Social Sciences (SPSS) program was used for data analysis. Comparison of categorical data was by Chi-square statistic or Fisher's exact test for small expected values.\n",
      "http://data.bioontology.org/annotator/?text=The%20Statistical%20Package%20for%20Social%20Sciences%20%28SPSS%29%20program%20was%20used%20for%20data%20analysis.%20Comparison%20of%20categorical%20data%20was%20by%20Chi-square%20statistic%20or%20Fisher%27s%20exact%20test%20for%20small%20expected%20values.\n",
      "A total of 40 respiratory specimens from 40 patients were positive for M. pneumoniae over the 24-months study period. The demographic and epidemiological characteristics of the patients are summarized in Table 1. Of all isolates, 37 (92.5%) were community-acquired and 3 (7.5%) were nosocomial. Thirty-three (82.5%) patients required admission to the hospital and the remaining 7 (17.5%) were treated as outpatients. Twenty-four isolates (60%) were associated with pneumonia, 14 (35%) with upper respiratory tract infections, and 2 (5%) with bronchiolitis. Of the 24 cases of pneumonia, 21 were confirmed radiologically and the remaining 3 were diagnosed clinically. The two cases of bronchiolitis occurred in 2 children, one and three years old. Thirty-one patients (77.5%) had comorbidities. Eleven patients (27.5%) had cardiopulmonary comorbidities (asthma, 8, lung fibrosis, 1, congestive heart failure, 1, congenial heart disease, 1), 9 patients (22.5%) were immunocompromised (malignancy, 7, steroid therapy, 3, Human immunodeficiency virus infection, 1), and 11 patients (27.5%) had other comorbidities (premature newborns, 2, and one each of myelodysplastic syndrome, myeloproliferative disorder, sickle cell anemia, Evan's syndrome, Down syndrome, sarcoidosis, demyelinating disease, cerebral palsy, and spinal muscle atrophy). Organisms concomitantly isolated with M. pneumoniae from the respiratory tract included herpes simplex virus type 1 (2 occasions), adenovirus (2 occasions), cytomegalo virus (1 occasion), respiratory syncytial virus (1 occasion), and bacterial isolates (2 occasions: Acinetobacter species, 1, and Enter obacter cloacae, 1).\n",
      "http://data.bioontology.org/annotator/?text=A%20total%20of%2040%20respiratory%20specimens%20from%2040%20patients%20were%20positive%20for%20M.%20pneumoniae%20over%20the%2024-months%20study%20period.%20The%20demographic%20and%20epidemiological%20characteristics%20of%20the%20patients%20are%20summarized%20in%20Table%201.%20Of%20all%20isolates%2C%2037%20%2892.5%25%29%20were%20community-acquired%20and%203%20%287.5%25%29%20were%20nosocomial.%20Thirty-three%20%2882.5%25%29%20patients%20required%20admission%20to%20the%20hospital%20and%20the%20remaining%207%20%2817.5%25%29%20were%20treated%20as%20outpatients.%20Twenty-four%20isolates%20%2860%25%29%20were%20associated%20with%20pneumonia%2C%2014%20%2835%25%29%20with%20upper%20respiratory%20tract%20infections%2C%20and%202%20%285%25%29%20with%20bronchiolitis.%20Of%20the%2024%20cases%20of%20pneumonia%2C%2021%20were%20confirmed%20radiologically%20and%20the%20remaining%203%20were%20diagnosed%20clinically.%20The%20two%20cases%20of%20bronchiolitis%20occurred%20in%202%20children%2C%20one%20and%20three%20years%20old.%20Thirty-one%20patients%20%2877.5%25%29%20had%20comorbidities.%20Eleven%20patients%20%2827.5%25%29%20had%20cardiopulmonary%20comorbidities%20%28asthma%2C%208%2C%20lung%20fibrosis%2C%201%2C%20congestive%20heart%20failure%2C%201%2C%20congenial%20heart%20disease%2C%201%29%2C%209%20patients%20%2822.5%25%29%20were%20immunocompromised%20%28malignancy%2C%207%2C%20steroid%20therapy%2C%203%2C%20Human%20immunodeficiency%20virus%20infection%2C%201%29%2C%20and%2011%20patients%20%2827.5%25%29%20had%20other%20comorbidities%20%28premature%20newborns%2C%202%2C%20and%20one%20each%20of%20myelodysplastic%20syndrome%2C%20myeloproliferative%20disorder%2C%20sickle%20cell%20anemia%2C%20Evan%27s%20syndrome%2C%20Down%20syndrome%2C%20sarcoidosis%2C%20demyelinating%20disease%2C%20cerebral%20palsy%2C%20and%20spinal%20muscle%20atrophy%29.%20Organisms%20concomitantly%20isolated%20with%20M.%20pneumoniae%20from%20the%20respiratory%20tract%20included%20herpes%20simplex%20virus%20type%201%20%282%20occasions%29%2C%20adenovirus%20%282%20occasions%29%2C%20cytomegalo%20virus%20%281%20occasion%29%2C%20respiratory%20syncytial%20virus%20%281%20occasion%29%2C%20and%20bacterial%20isolates%20%282%20occasions%3A%20Acinetobacter%20species%2C%201%2C%20and%20Enter%20obacter%20cloacae%2C%201%29.\n",
      "Clinical manifestations associated with M. pneumoniae infections are summarized in Table 2. Pneumonia was more common than upper respiratory tract infections (57.5 % versus 27.5%, respectively). Immunocompromised patients were more likely to present with pneumonia as opposed to upper respiratory tract infection or bronchiolitis than non-immunocompromised patients (8/9 versus 16/31, P = 0.05). Similarly, there was a tendency for patients 60 years of age or older to present with pneumonia more frequently than those below 60 (4/4 versus 20/36, P = 0.1). Of the 24 patients with clinically or radiologically confirmed pneumonia, 19 (79.2%) had crepitations and only 6 (25%) had bronchial breath sounds on physical examination. Of the 16 patients in whom wheezes were detected, 9 (56.3%) were not known to have asthma or other obstructive airway disease.\n",
      "http://data.bioontology.org/annotator/?text=Clinical%20manifestations%20associated%20with%20M.%20pneumoniae%20infections%20are%20summarized%20in%20Table%202.%20Pneumonia%20was%20more%20common%20than%20upper%20respiratory%20tract%20infections%20%2857.5%20%25%20versus%2027.5%25%2C%20respectively%29.%20Immunocompromised%20patients%20were%20more%20likely%20to%20present%20with%20pneumonia%20as%20opposed%20to%20upper%20respiratory%20tract%20infection%20or%20bronchiolitis%20than%20non-immunocompromised%20patients%20%288/9%20versus%2016/31%2C%20P%20%3D%200.05%29.%20Similarly%2C%20there%20was%20a%20tendency%20for%20patients%2060%20years%20of%20age%20or%20older%20to%20present%20with%20pneumonia%20more%20frequently%20than%20those%20below%2060%20%284/4%20versus%2020/36%2C%20P%20%3D%200.1%29.%20Of%20the%2024%20patients%20with%20clinically%20or%20radiologically%20confirmed%20pneumonia%2C%2019%20%2879.2%25%29%20had%20crepitations%20and%20only%206%20%2825%25%29%20had%20bronchial%20breath%20sounds%20on%20physical%20examination.%20Of%20the%2016%20patients%20in%20whom%20wheezes%20were%20detected%2C%209%20%2856.3%25%29%20were%20not%20known%20to%20have%20asthma%20or%20other%20obstructive%20airway%20disease.\n",
      "Complications and outcome of patients are shown in Table 3. Of the 24 patients with pneumonia, 21 (87.5%) were admitted to the hospital, and 20 (83.3%) had comorbidities. All patients with upper respiratory tract infections (11 patients) or bronchiolitis (2 patients) had uneventful recovery. Of the 24 patients with pneumonia, 14 (58.3%) had uneventful recovery, 4 (16.7%) recovered following some complications (acute respiratory distress syndrome, 2, respiratory failure, 1, septic shock, 1), 3 (12.5%) died because of M pneumoniae infection, and 3 (12.5%) died due to underlying comorbidities. The 3 patients who died of M pneumoniae pneumonia had other comorbidities; one had congestive heart failure, the second had congenital heart disease, and the third was a 3-months old infant born prematurely at 32 weeks of gestation who previously had 3 episodes of pneumonia due to other pathogens.\n",
      "http://data.bioontology.org/annotator/?text=Complications%20and%20outcome%20of%20patients%20are%20shown%20in%20Table%203.%20Of%20the%2024%20patients%20with%20pneumonia%2C%2021%20%2887.5%25%29%20were%20admitted%20to%20the%20hospital%2C%20and%2020%20%2883.3%25%29%20had%20comorbidities.%20All%20patients%20with%20upper%20respiratory%20tract%20infections%20%2811%20patients%29%20or%20bronchiolitis%20%282%20patients%29%20had%20uneventful%20recovery.%20Of%20the%2024%20patients%20with%20pneumonia%2C%2014%20%2858.3%25%29%20had%20uneventful%20recovery%2C%204%20%2816.7%25%29%20recovered%20following%20some%20complications%20%28acute%20respiratory%20distress%20syndrome%2C%202%2C%20respiratory%20failure%2C%201%2C%20septic%20shock%2C%201%29%2C%203%20%2812.5%25%29%20died%20because%20of%20M%20pneumoniae%20infection%2C%20and%203%20%2812.5%25%29%20died%20due%20to%20underlying%20comorbidities.%20The%203%20patients%20who%20died%20of%20M%20pneumoniae%20pneumonia%20had%20other%20comorbidities%3B%20one%20had%20congestive%20heart%20failure%2C%20the%20second%20had%20congenital%20heart%20disease%2C%20and%20the%20third%20was%20a%203-months%20old%20infant%20born%20prematurely%20at%2032%20weeks%20of%20gestation%20who%20previously%20had%203%20episodes%20of%20pneumonia%20due%20to%20other%20pathogens.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Mycoplasma pneumoniae is one of the most common causes of atypical pneumonia accounting for 5-23% of community-acquired pneumonia,[1,2,3,4,5] In a study of 511 children with acute respiratory tract infection in Riyadh, Saudi Arabia, Mycoplasma pneumoniae was found to be the second most common causative agent after Respiratory syncytial virus (RSV) accounting for 9% of all cases,[8] In a study of 112 adult patients with community acquired pneumonia admitted to a military hospital in Riyadh, Saudi Arabia, this organism accounted for 6% of all cases,[9] In another retrospective study of 567 pneumonic episodes in adult patients from Al-Qassim area, the organism accounted for 23% of all episodes,[10] The organism also causes other relatively minor infections such as pharyngitis, tracheobronchitis, bronchiolitis, and croup. It is transmitted from person-to-person by infected respiratory droplets during close contact. It is most common in school-aged children, military recruits, and college students.[11] Most cases occur singly or as family outbreaks. Larger outbreaks can also occur in closed populations such as military recruit camps or boarding schools,[12] Infection occurs most frequently during the fall and winter in temperate climates but may develop year-round,[13] The average incubation period is 3 weeks following exposure,[6] Although rare, complications are protean and may involve virtually any organ system such as the respiratory system (e.g.: pleurisy, pneumothorax, acute respiratory distress syndrome, lung abscess), the hematologic system (e.g.: hemolytic anemia, intravascular coagulation, thrombocytopenia), the dermatologic system (e.g.: maculopapular or urticarial rashes, erythema multiforme, erythema nodosum), the musculoskeletal system (e.g.: myalgias, arthralgias, arthritis), the cardiovascular system (e.g.: pericarditis, myocarditis), the nervous system (e.g.: meningoencephalitis, Guillain-Barre syndrome, neuropathies, acute psychosis), or the eye (optic disc edema, optic nerve atrophy, retinal exudation and hemorrhages).[6,7,14,15,16,17,18] Immunity following infection is not long lasting.[11]\n",
      "http://data.bioontology.org/annotator/?text=Mycoplasma%20pneumoniae%20is%20one%20of%20the%20most%20common%20causes%20of%20atypical%20pneumonia%20accounting%20for%205-23%25%20of%20community-acquired%20pneumonia%2C%5B1%2C2%2C3%2C4%2C5%5D%20In%20a%20study%20of%20511%20children%20with%20acute%20respiratory%20tract%20infection%20in%20Riyadh%2C%20Saudi%20Arabia%2C%20Mycoplasma%20pneumoniae%20was%20found%20to%20be%20the%20second%20most%20common%20causative%20agent%20after%20Respiratory%20syncytial%20virus%20%28RSV%29%20accounting%20for%209%25%20of%20all%20cases%2C%5B8%5D%20In%20a%20study%20of%20112%20adult%20patients%20with%20community%20acquired%20pneumonia%20admitted%20to%20a%20military%20hospital%20in%20Riyadh%2C%20Saudi%20Arabia%2C%20this%20organism%20accounted%20for%206%25%20of%20all%20cases%2C%5B9%5D%20In%20another%20retrospective%20study%20of%20567%20pneumonic%20episodes%20in%20adult%20patients%20from%20Al-Qassim%20area%2C%20the%20organism%20accounted%20for%2023%25%20of%20all%20episodes%2C%5B10%5D%20The%20organism%20also%20causes%20other%20relatively%20minor%20infections%20such%20as%20pharyngitis%2C%20tracheobronchitis%2C%20bronchiolitis%2C%20and%20croup.%20It%20is%20transmitted%20from%20person-to-person%20by%20infected%20respiratory%20droplets%20during%20close%20contact.%20It%20is%20most%20common%20in%20school-aged%20children%2C%20military%20recruits%2C%20and%20college%20students.%5B11%5D%20Most%20cases%20occur%20singly%20or%20as%20family%20outbreaks.%20Larger%20outbreaks%20can%20also%20occur%20in%20closed%20populations%20such%20as%20military%20recruit%20camps%20or%20boarding%20schools%2C%5B12%5D%20Infection%20occurs%20most%20frequently%20during%20the%20fall%20and%20winter%20in%20temperate%20climates%20but%20may%20develop%20year-round%2C%5B13%5D%20The%20average%20incubation%20period%20is%203%20weeks%20following%20exposure%2C%5B6%5D%20Although%20rare%2C%20complications%20are%20protean%20and%20may%20involve%20virtually%20any%20organ%20system%20such%20as%20the%20respiratory%20system%20%28e.g.%3A%20pleurisy%2C%20pneumothorax%2C%20acute%20respiratory%20distress%20syndrome%2C%20lung%20abscess%29%2C%20the%20hematologic%20system%20%28e.g.%3A%20hemolytic%20anemia%2C%20intravascular%20coagulation%2C%20thrombocytopenia%29%2C%20the%20dermatologic%20system%20%28e.g.%3A%20maculopapular%20or%20urticarial%20rashes%2C%20erythema%20multiforme%2C%20erythema%20nodosum%29%2C%20the%20musculoskeletal%20system%20%28e.g.%3A%20myalgias%2C%20arthralgias%2C%20arthritis%29%2C%20the%20cardiovascular%20system%20%28e.g.%3A%20pericarditis%2C%20myocarditis%29%2C%20the%20nervous%20system%20%28e.g.%3A%20meningoencephalitis%2C%20Guillain-Barre%20syndrome%2C%20neuropathies%2C%20acute%20psychosis%29%2C%20or%20the%20eye%20%28optic%20disc%20edema%2C%20optic%20nerve%20atrophy%2C%20retinal%20exudation%20and%20hemorrhages%29.%5B6%2C7%2C14%2C15%2C16%2C17%2C18%5D%20Immunity%20following%20infection%20is%20not%20long%20lasting.%5B11%5D\n",
      "In our study, the infection affected all age groups but was most common in infants (32.5%) and preschool children (22.5%), and least common in adults aged 15 to 30 years (2.5%) and elderly above 70 years of age (5%). This contrasts with data from temperate countries where the infection is most common in school-aged children, and young adults.[11] One possible explanation for this difference is that infants and preschool children perhaps had more severe infections than did school-aged children, and young adults which prompted presentation of the former group to the hospital. The infection occurred year-round but was most common in the fall (35%), and spring (30%), and least common in the summer (10%). Most infections were community-acquired (92.5%).\n",
      "http://data.bioontology.org/annotator/?text=In%20our%20study%2C%20the%20infection%20affected%20all%20age%20groups%20but%20was%20most%20common%20in%20infants%20%2832.5%25%29%20and%20preschool%20children%20%2822.5%25%29%2C%20and%20least%20common%20in%20adults%20aged%2015%20to%2030%20years%20%282.5%25%29%20and%20elderly%20above%2070%20years%20of%20age%20%285%25%29.%20This%20contrasts%20with%20data%20from%20temperate%20countries%20where%20the%20infection%20is%20most%20common%20in%20school-aged%20children%2C%20and%20young%20adults.%5B11%5D%20One%20possible%20explanation%20for%20this%20difference%20is%20that%20infants%20and%20preschool%20children%20perhaps%20had%20more%20severe%20infections%20than%20did%20school-aged%20children%2C%20and%20young%20adults%20which%20prompted%20presentation%20of%20the%20former%20group%20to%20the%20hospital.%20The%20infection%20occurred%20year-round%20but%20was%20most%20common%20in%20the%20fall%20%2835%25%29%2C%20and%20spring%20%2830%25%29%2C%20and%20least%20common%20in%20the%20summer%20%2810%25%29.%20Most%20infections%20were%20community-acquired%20%2892.5%25%29.\n",
      "More than one half of patients (57.5%) presented with pneumonia, and about a third (27.5%) presented with upper respiratory tract infection, Immunocompromised patients and patients 60 years of age or older were more likely to present with pneumonia as opposed to upper respiratory tract infection than non-immunocompromised patients or those below 60 years of age. Cough (82.5%), fever (75%), and malaise (58.8%) were the most common presenting symptoms. Cough was usually dry or slightly productive of white sputum and mild to moderate in severity. Most febrile patients had mild to moderate fever of 39°C or less; high-grade fever of more than 39°C was rare. Crepitations (60%), and wheezes (40%) were the most common signs. Wheezes were as common in patients with no history of obstructive airway disease (9 patients) as it was in those with such a history (7 patients). Bronchial breathing as a sign of consolidation was detected in only one fourth of patients with pneumonia, which is consistent with the known disparity between clinical and radiological signs of M pneumoniae pneumonia. Crepitations, however, were detected in the majority (79.2%) of patients. Pleuritic chest pain and pleural effusion were rare.\n",
      "http://data.bioontology.org/annotator/?text=More%20than%20one%20half%20of%20patients%20%2857.5%25%29%20presented%20with%20pneumonia%2C%20and%20about%20a%20third%20%2827.5%25%29%20presented%20with%20upper%20respiratory%20tract%20infection%2C%20Immunocompromised%20patients%20and%20patients%2060%20years%20of%20age%20or%20older%20were%20more%20likely%20to%20present%20with%20pneumonia%20as%20opposed%20to%20upper%20respiratory%20tract%20infection%20than%20non-immunocompromised%20patients%20or%20those%20below%2060%20years%20of%20age.%20Cough%20%2882.5%25%29%2C%20fever%20%2875%25%29%2C%20and%20malaise%20%2858.8%25%29%20were%20the%20most%20common%20presenting%20symptoms.%20Cough%20was%20usually%20dry%20or%20slightly%20productive%20of%20white%20sputum%20and%20mild%20to%20moderate%20in%20severity.%20Most%20febrile%20patients%20had%20mild%20to%20moderate%20fever%20of%2039%C2%B0C%20or%20less%3B%20high-grade%20fever%20of%20more%20than%2039%C2%B0C%20was%20rare.%20Crepitations%20%2860%25%29%2C%20and%20wheezes%20%2840%25%29%20were%20the%20most%20common%20signs.%20Wheezes%20were%20as%20common%20in%20patients%20with%20no%20history%20of%20obstructive%20airway%20disease%20%289%20patients%29%20as%20it%20was%20in%20those%20with%20such%20a%20history%20%287%20patients%29.%20Bronchial%20breathing%20as%20a%20sign%20of%20consolidation%20was%20detected%20in%20only%20one%20fourth%20of%20patients%20with%20pneumonia%2C%20which%20is%20consistent%20with%20the%20known%20disparity%20between%20clinical%20and%20radiological%20signs%20of%20M%20pneumoniae%20pneumonia.%20Crepitations%2C%20however%2C%20were%20detected%20in%20the%20majority%20%2879.2%25%29%20of%20patients.%20Pleuritic%20chest%20pain%20and%20pleural%20effusion%20were%20rare.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "More than half (56.5%) of the patients with pneumonia had uneventful recovery. Mortality from M. pneumoniae pneumonia was high (12.5%) and occurred only in patients with underlying comorbidities. None of the 9 patients with no underlying comorbidities died of M pneumoniae pneumonia. The relatively high complications rate (16.7%) and mortality (12.5%) related to M. pneumoniae pneumonia are likely due to selection bias as most patients with pneumonia were sick enough to require admission to the hospital (21/24 or 87.5%) and most of them had comorbidities (20/24 or 83.3%).\n",
      "http://data.bioontology.org/annotator/?text=More%20than%20half%20%2856.5%25%29%20of%20the%20patients%20with%20pneumonia%20had%20uneventful%20recovery.%20Mortality%20from%20M.%20pneumoniae%20pneumonia%20was%20high%20%2812.5%25%29%20and%20occurred%20only%20in%20patients%20with%20underlying%20comorbidities.%20None%20of%20the%209%20patients%20with%20no%20underlying%20comorbidities%20died%20of%20M%20pneumoniae%20pneumonia.%20The%20relatively%20high%20complications%20rate%20%2816.7%25%29%20and%20mortality%20%2812.5%25%29%20related%20to%20M.%20pneumoniae%20pneumonia%20are%20likely%20due%20to%20selection%20bias%20as%20most%20patients%20with%20pneumonia%20were%20sick%20enough%20to%20require%20admission%20to%20the%20hospital%20%2821/24%20or%2087.5%25%29%20and%20most%20of%20them%20had%20comorbidities%20%2820/24%20or%2083.3%25%29.\n",
      "In conclusion, our data shed some light on the epidemiology and clinical features of M pneumoniae infections in one of the Saudi tertiary care centers. The data are comparable to those of other countries except for the finding that infections were more common in infants and preschool children than in school children and young adults. Additionally, mortality attributable to M. pneumoniae pneumonia was relatively high in patients with comorbidities. It is hoped this information will assist clinicians in their approach and management of respiratory tract infections.\n",
      "http://data.bioontology.org/annotator/?text=In%20conclusion%2C%20our%20data%20shed%20some%20light%20on%20the%20epidemiology%20and%20clinical%20features%20of%20M%20pneumoniae%20infections%20in%20one%20of%20the%20Saudi%20tertiary%20care%20centers.%20The%20data%20are%20comparable%20to%20those%20of%20other%20countries%20except%20for%20the%20finding%20that%20infections%20were%20more%20common%20in%20infants%20and%20preschool%20children%20than%20in%20school%20children%20and%20young%20adults.%20Additionally%2C%20mortality%20attributable%20to%20M.%20pneumoniae%20pneumonia%20was%20relatively%20high%20in%20patients%20with%20comorbidities.%20It%20is%20hoped%20this%20information%20will%20assist%20clinicians%20in%20their%20approach%20and%20management%20of%20respiratory%20tract%20infections.\n",
      "The pre-publication history for this paper can be accessed here:\n",
      "http://data.bioontology.org/annotator/?text=The%20pre-publication%20history%20for%20this%20paper%20can%20be%20accessed%20here%3A\n",
      "\n",
      "http://data.bioontology.org/annotator/?text=Clinical%20features%20of%20culture-proven%20Mycoplasma%20pneumoniae%20infections%20at%20King%20Abdulaziz%20University%20Hospital%2C%20Jeddah%2C%20Saudi%20Arabia\n",
      "paper index: 17 processed\n",
      "Objective: This retrospective chart review describes the epidemiology and clinical features of 40 patients with culture-proven Mycoplasma pneumoniae infections at King Abdulaziz University Hospital, Jeddah, Saudi Arabia.\n",
      "http://data.bioontology.org/annotator/?text=Objective%3A%20This%20retrospective%20chart%20review%20describes%20the%20epidemiology%20and%20clinical%20features%20of%2040%20patients%20with%20culture-proven%20Mycoplasma%20pneumoniae%20infections%20at%20King%20Abdulaziz%20University%20Hospital%2C%20Jeddah%2C%20Saudi%20Arabia.\n",
      "{'matchTerm': ['OBJECTIVE', 'OBJECTIVE', 'OBJECTIVE', 'OBJECTIVE', 'RETROSPECTIVE', 'RETROSPECTIVE', 'CHART REVIEW', 'CHART REVIEW', 'EPIDEMIOLOGY', 'EPIDEMIOLOGY', 'EPIDEMIOLOGY', 'EPIDEMIOLOGY', 'CLINICAL', 'CLINICAL', 'CLINICAL', 'CLINICAL', 'PATIENTS', 'PATIENTS', 'PATIENTS', 'CULTURE', 'CULTURE', 'CULTURE', 'CULTURE', 'CULTURE', 'PROVEN', 'PROVEN', 'PROVEN', 'PROVEN', 'PROVEN', 'MYCOPLASMA PNEUMONIAE', 'MYCOPLASMA PNEUMONIAE', 'MYCOPLASMA PNEUMONIAE', 'MYCOPLASMA PNEUMONIAE', 'INFECTIONS', 'INFECTIONS', 'UNIVERSITY HOSPITAL', 'UNIVERSITY HOSPITAL', 'SAUDI ARABIA', 'SAUDI ARABIA', 'SAUDI ARABIA', 'SAUDI ARABIA'], 'ontology': ['http://data.bioontology.org/ontologies/PREMEDONTO', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/PREMEDONTO', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/MESH', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/MESH', 'http://data.bioontology.org/ontologies/PREMEDONTO', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/MESH', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/PREMEDONTO', 'http://data.bioontology.org/ontologies/MESH', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/MESH', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/MESH', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/MESH'], 'annotatedClass': ['http://purl.obolibrary.org/obo/NCIT_C88181', 'http://purl.bioontology.org/ontology/SNOMEDCT/260224007', 'http://sbmi.uth.tmc.edu/ontology/ochv#5636', 'http://purl.obolibrary.org/obo/NCIT_C49155', 'http://sbmi.uth.tmc.edu/ontology/ochv#10821', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0035363', 'http://sbmi.uth.tmc.edu/ontology/ochv#38503', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0541653', 'http://purl.bioontology.org/ontology/MESH/D004813', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0014507', 'http://sbmi.uth.tmc.edu/ontology/ochv#4543', 'http://purl.bioontology.org/ontology/MESH/Q000453', 'http://purl.obolibrary.org/obo/NCIT_C25398', 'http://purl.bioontology.org/ontology/SNOMEDCT/58147004', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0205210', 'http://sbmi.uth.tmc.edu/ontology/ochv#20405', 'http://purl.bioontology.org/ontology/MESH/D010361', 'http://sbmi.uth.tmc.edu/ontology/ochv#9372', 'http://purl.obolibrary.org/obo/NCIT_C16960', 'http://purl.bioontology.org/ontology/MESH/D003469', 'http://purl.bioontology.org/ontology/SNOMEDCT/703725008', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0010453', 'http://purl.bioontology.org/ontology/SNOMEDCT/61594008', 'http://sbmi.uth.tmc.edu/ontology/ochv#3501', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0700164', 'http://sbmi.uth.tmc.edu/ontology/ochv#36586', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0456369', 'http://purl.bioontology.org/ontology/SNOMEDCT/277025001', 'http://sbmi.uth.tmc.edu/ontology/ochv#44676', 'http://purl.bioontology.org/ontology/MESH/D009177', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0026941', 'http://sbmi.uth.tmc.edu/ontology/ochv#8394', 'http://purl.bioontology.org/ontology/SNOMEDCT/58720004', 'http://purl.bioontology.org/ontology/MESH/D007239', 'http://sbmi.uth.tmc.edu/ontology/ochv#6691', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0020028', 'http://sbmi.uth.tmc.edu/ontology/ochv#6283', 'http://purl.bioontology.org/ontology/SNOMEDCT/223616007', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0036243', 'http://sbmi.uth.tmc.edu/ontology/ochv#11061', 'http://purl.bioontology.org/ontology/MESH/D012529']}\n",
      "Patients with positive M. pneumoniae cultures from respiratory specimens from January 1997 through December 1998 were identified through the Microbiology records. Charts of patients were reviewed.\n",
      "http://data.bioontology.org/annotator/?text=Patients%20with%20positive%20M.%20pneumoniae%20cultures%20from%20respiratory%20specimens%20from%20January%201997%20through%20December%201998%20were%20identified%20through%20the%20Microbiology%20records.%20Charts%20of%20patients%20were%20reviewed.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'matchTerm': ['PATIENTS', 'PATIENTS', 'PATIENTS', 'POSITIVE', 'POSITIVE', 'POSITIVE', 'POSITIVE', 'CULTURES', 'SPECIMENS', '1997', '1998', 'IDENTIFIED', 'IDENTIFIED', 'IDENTIFIED', 'IDENTIFIED', 'MICROBIOLOGY', 'MICROBIOLOGY', 'MICROBIOLOGY', 'MICROBIOLOGY', 'RECORDS', 'RECORDS', 'CHARTS', 'CHARTS', 'REVIEWED', 'REVIEWED'], 'ontology': ['http://data.bioontology.org/ontologies/MESH', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/PREMEDONTO', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/PREMEDONTO', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/PREMEDONTO', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/MESH', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/MESH', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/MESH', 'http://data.bioontology.org/ontologies/MESH', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/PREMEDONTO', 'http://data.bioontology.org/ontologies/OCHV'], 'annotatedClass': ['http://purl.bioontology.org/ontology/MESH/D010361', 'http://sbmi.uth.tmc.edu/ontology/ochv#9372', 'http://purl.obolibrary.org/obo/NCIT_C16960', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0205159', 'http://purl.obolibrary.org/obo/NCIT_C25246', 'http://sbmi.uth.tmc.edu/ontology/ochv#20358', 'http://purl.bioontology.org/ontology/SNOMEDCT/10828004', 'http://sbmi.uth.tmc.edu/ontology/ochv#3501', 'http://sbmi.uth.tmc.edu/ontology/ochv#31600', 'http://purl.bioontology.org/ontology/SNOMEDCT/312848003', 'http://sbmi.uth.tmc.edu/ontology/ochv#1998', 'http://purl.bioontology.org/ontology/SNOMEDCT/7882003', 'http://sbmi.uth.tmc.edu/ontology/ochv#20560', 'http://purl.obolibrary.org/obo/NCIT_C25737', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0205396', 'http://purl.bioontology.org/ontology/MESH/Q000382', 'http://purl.bioontology.org/ontology/SNOMEDCT/269934003', 'http://purl.bioontology.org/ontology/MESH/D008829', 'http://sbmi.uth.tmc.edu/ontology/ochv#8091', 'http://sbmi.uth.tmc.edu/ontology/ochv#10642', 'http://purl.bioontology.org/ontology/MESH/D011996', 'http://purl.bioontology.org/ontology/MESH/D020468', 'http://sbmi.uth.tmc.edu/ontology/ochv#43962', 'http://purl.obolibrary.org/obo/NCIT_C45314', 'http://sbmi.uth.tmc.edu/ontology/ochv#27595']}\n",
      "Results: 40 patients were identified, 33 (82.5%) of whom required admission. Most infections (92.5%) were community-acquired. The infection affected all age groups but was most common in infants (32.5%) and pre-school children (22.5%). It occurred year-round but was most common in the fall (35%) and spring (30%). More than three-quarters of patients (77.5%) had comorbidities. Twenty-four isolates (60%) were associated with pneumonia, 14 (35%) with upper respiratory tract infections, and 2 (5%) with bronchiolitis. Cough (82.5%), fever (75%), and malaise (58.8%) were the most common symptoms, and crepitations (60%), and wheezes (40%) were the most common signs. Most patients with pneumonia had crepitations (79.2%) but only 25% had bronchial breathing. Immunocompromised patients were more likely than non-immunocompromised patients to present with pneumonia (8/9 versus 16/31, P = 0.05). Of the 24 patients with pneumonia, 14 (58.3%) had uneventful recovery, 4 (16.7%) recovered following some complications, 3 (12.5%) died because of M pneumoniae infection, and 3 (12.5%) died due to underlying comorbidities. The 3 patients who died of M pneumoniae pneumonia had other comorbidities.\n",
      "http://data.bioontology.org/annotator/?text=Results%3A%2040%20patients%20were%20identified%2C%2033%20%2882.5%25%29%20of%20whom%20required%20admission.%20Most%20infections%20%2892.5%25%29%20were%20community-acquired.%20The%20infection%20affected%20all%20age%20groups%20but%20was%20most%20common%20in%20infants%20%2832.5%25%29%20and%20pre-school%20children%20%2822.5%25%29.%20It%20occurred%20year-round%20but%20was%20most%20common%20in%20the%20fall%20%2835%25%29%20and%20spring%20%2830%25%29.%20More%20than%20three-quarters%20of%20patients%20%2877.5%25%29%20had%20comorbidities.%20Twenty-four%20isolates%20%2860%25%29%20were%20associated%20with%20pneumonia%2C%2014%20%2835%25%29%20with%20upper%20respiratory%20tract%20infections%2C%20and%202%20%285%25%29%20with%20bronchiolitis.%20Cough%20%2882.5%25%29%2C%20fever%20%2875%25%29%2C%20and%20malaise%20%2858.8%25%29%20were%20the%20most%20common%20symptoms%2C%20and%20crepitations%20%2860%25%29%2C%20and%20wheezes%20%2840%25%29%20were%20the%20most%20common%20signs.%20Most%20patients%20with%20pneumonia%20had%20crepitations%20%2879.2%25%29%20but%20only%2025%25%20had%20bronchial%20breathing.%20Immunocompromised%20patients%20were%20more%20likely%20than%20non-immunocompromised%20patients%20to%20present%20with%20pneumonia%20%288/9%20versus%2016/31%2C%20P%20%3D%200.05%29.%20Of%20the%2024%20patients%20with%20pneumonia%2C%2014%20%2858.3%25%29%20had%20uneventful%20recovery%2C%204%20%2816.7%25%29%20recovered%20following%20some%20complications%2C%203%20%2812.5%25%29%20died%20because%20of%20M%20pneumoniae%20infection%2C%20and%203%20%2812.5%25%29%20died%20due%20to%20underlying%20comorbidities.%20The%203%20patients%20who%20died%20of%20M%20pneumoniae%20pneumonia%20had%20other%20comorbidities.\n",
      "{'matchTerm': ['PATIENTS', 'PATIENTS', 'PATIENTS', 'IDENTIFIED', 'IDENTIFIED', 'IDENTIFIED', 'IDENTIFIED', '82.5', 'ADMISSION', 'INFECTIONS', 'INFECTIONS', 'COMMUNITY', 'COMMUNITY', 'COMMUNITY', 'COMMUNITY', 'ACQUIRED', 'ACQUIRED', 'ACQUIRED', 'INFECTION', 'INFECTION', 'AFFECTED', 'AGE GROUPS', 'AGE GROUPS', 'BUT', 'COMMON', 'COMMON', 'COMMON', 'COMMON', 'INFANTS', 'INFANTS', '32.5', 'PRE-SCHOOL', 'PRE-SCHOOL', 'PRE-SCHOOL', 'SCHOOL CHILDREN', 'SCHOOL CHILDREN', '22.5', 'YEAR', 'YEAR', 'YEAR', 'YEAR', 'ROUND', 'ROUND', 'FALL', 'FALL', 'SPRING', 'SPRING', 'COMORBIDITIES', 'TWENTY', 'TWENTY', 'ISOLATES', 'ASSOCIATED WITH', 'ASSOCIATED WITH', 'PNEUMONIA', 'PNEUMONIA', 'PNEUMONIA', 'PNEUMONIA', 'PNEUMONIA', 'PNEUMONIA', 'UPPER RESPIRATORY TRACT', 'UPPER RESPIRATORY TRACT', 'RESPIRATORY TRACT INFECTIONS', 'RESPIRATORY TRACT INFECTIONS', 'BRONCHIOLITIS', 'BRONCHIOLITIS', 'BRONCHIOLITIS', 'COUGH', 'COUGH', 'COUGH', 'COUGH', 'COUGH', 'FEVER', 'FEVER', 'FEVER', 'FEVER', 'FEVER', 'FEVER', 'MALAISE', 'MALAISE', 'MALAISE', 'CREPITATIONS', 'WHEEZES', 'SIGNS', 'BRONCHIAL BREATHING', 'BRONCHIAL BREATHING', 'BRONCHIAL BREATHING', 'IMMUNOCOMPROMISED PATIENTS', 'LIKELY', 'LIKELY', 'LIKELY', 'PRESENT', 'PRESENT', 'PRESENT', 'PRESENT', '0.05', 'RECOVERY', 'RECOVERY', '16.7', 'RECOVERED', 'FOLLOWING', 'FOLLOWING', 'FOLLOWING', 'FOLLOWING', 'COMPLICATIONS', 'COMPLICATIONS', '12.5', 'DIED', 'DIED', 'DUE TO', 'DUE TO', 'DUE TO', 'DUE TO'], 'ontology': ['http://data.bioontology.org/ontologies/MESH', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/PREMEDONTO', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/PREMEDONTO', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/MESH', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/PREMEDONTO', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/HPIO', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/MESH', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/PREMEDONTO', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/MESH', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/PREMEDONTO', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/MESH', 'http://data.bioontology.org/ontologies/HPIO', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/COVID-19', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/MESH', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/MESH', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/MESH', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/COVID-19', 'http://data.bioontology.org/ontologies/HPIO', 'http://data.bioontology.org/ontologies/HPIO', 'http://data.bioontology.org/ontologies/COVID-19', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/MESH', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/COVID-19', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/PREMEDONTO', 'http://data.bioontology.org/ontologies/PREMEDONTO', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/MESH', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/PREMEDONTO', 'http://data.bioontology.org/ontologies/SNOMEDCT'], 'annotatedClass': ['http://purl.bioontology.org/ontology/MESH/D010361', 'http://sbmi.uth.tmc.edu/ontology/ochv#9372', 'http://purl.obolibrary.org/obo/NCIT_C16960', 'http://purl.bioontology.org/ontology/SNOMEDCT/7882003', 'http://sbmi.uth.tmc.edu/ontology/ochv#20560', 'http://purl.obolibrary.org/obo/NCIT_C25737', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0205396', 'http://purl.bioontology.org/ontology/SNOMEDCT/732884009', 'http://sbmi.uth.tmc.edu/ontology/ochv#49077', 'http://purl.bioontology.org/ontology/MESH/D007239', 'http://sbmi.uth.tmc.edu/ontology/ochv#6691', 'http://sbmi.uth.tmc.edu/ontology/ochv#3211', 'http://purl.obolibrary.org/obo/NCIT_C16453', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0009462', 'http://purl.bioontology.org/ontology/SNOMEDCT/133928008', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0439661', 'http://sbmi.uth.tmc.edu/ontology/ochv#34579', 'http://purl.bioontology.org/ontology/SNOMEDCT/255396000', 'http://www.semanticweb.org/ontologies/2010/5/22/Ontology1277229984000.owl#HPI:0000022', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0021311', 'http://purl.bioontology.org/ontology/SNOMEDCT/247591002', 'http://purl.bioontology.org/ontology/MESH/D009273', 'http://sbmi.uth.tmc.edu/ontology/ochv#8475', 'http://sbmi.uth.tmc.edu/ontology/ochv#23132', 'http://purl.obolibrary.org/obo/NCIT_C43461', 'http://purl.bioontology.org/ontology/SNOMEDCT/72906007', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0205214', 'http://sbmi.uth.tmc.edu/ontology/ochv#20409', 'http://sbmi.uth.tmc.edu/ontology/ochv#6675', 'http://purl.bioontology.org/ontology/MESH/D007223', 'http://purl.bioontology.org/ontology/SNOMEDCT/732671003', 'http://sbmi.uth.tmc.edu/ontology/ochv#33419', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0424930', 'http://purl.bioontology.org/ontology/SNOMEDCT/161116001', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0260267', 'http://sbmi.uth.tmc.edu/ontology/ochv#25331', 'http://purl.bioontology.org/ontology/SNOMEDCT/732599004', 'http://purl.bioontology.org/ontology/SNOMEDCT/258707000', 'http://purl.obolibrary.org/obo/NCIT_C29848', 'http://sbmi.uth.tmc.edu/ontology/ochv#34458', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0439234', 'http://sbmi.uth.tmc.edu/ontology/ochv#29446', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0332490', 'http://sbmi.uth.tmc.edu/ontology/ochv#15704', 'http://purl.bioontology.org/ontology/SNOMEDCT/1912002', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0241232', 'http://sbmi.uth.tmc.edu/ontology/ochv#24555', 'http://sbmi.uth.tmc.edu/ontology/ochv#3226', 'http://sbmi.uth.tmc.edu/ontology/ochv#46382', 'http://purl.bioontology.org/ontology/SNOMEDCT/732587006', 'http://sbmi.uth.tmc.edu/ontology/ochv#31609', 'http://purl.bioontology.org/ontology/SNOMEDCT/47429007', 'http://sbmi.uth.tmc.edu/ontology/ochv#29390', 'http://sbmi.uth.tmc.edu/ontology/ochv#9867', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0032285', 'http://purl.bioontology.org/ontology/MESH/D011014', 'http://www.semanticweb.org/ontologies/2010/5/22/Ontology1277229984000.owl#HPI:0000229', 'http://purl.bioontology.org/ontology/SNOMEDCT/233604007', 'http://purl.obolibrary.org/obo/DOID_552', 'http://sbmi.uth.tmc.edu/ontology/ochv#36815', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0458578', 'http://purl.bioontology.org/ontology/MESH/D012141', 'http://sbmi.uth.tmc.edu/ontology/ochv#10770', 'http://purl.bioontology.org/ontology/MESH/D001988', 'http://sbmi.uth.tmc.edu/ontology/ochv#2186', 'http://purl.bioontology.org/ontology/SNOMEDCT/4120002', 'http://purl.bioontology.org/ontology/MESH/D003371', 'http://purl.bioontology.org/ontology/SNOMEDCT/49727002', 'http://sbmi.uth.tmc.edu/ontology/ochv#3404', 'http://purl.obolibrary.org/obo/SYMP_0000614', 'http://www.semanticweb.org/ontologies/2010/5/22/Ontology1277229984000.owl#HPI:0000171', 'http://www.semanticweb.org/ontologies/2010/5/22/Ontology1277229984000.owl#HPI:0000113', 'http://purl.obolibrary.org/obo/SYMP_0000613', 'http://sbmi.uth.tmc.edu/ontology/ochv#5010', 'http://purl.bioontology.org/ontology/MESH/D005334', 'http://purl.bioontology.org/ontology/SNOMEDCT/386661006', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0015967', 'http://purl.obolibrary.org/obo/SYMP_0019176', 'http://sbmi.uth.tmc.edu/ontology/ochv#22637', 'http://purl.bioontology.org/ontology/SNOMEDCT/367391008', 'http://sbmi.uth.tmc.edu/ontology/ochv#10564', 'http://sbmi.uth.tmc.edu/ontology/ochv#13155', 'http://sbmi.uth.tmc.edu/ontology/ochv#21234', 'http://purl.bioontology.org/ontology/SNOMEDCT/79451004', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0231873', 'http://sbmi.uth.tmc.edu/ontology/ochv#22813', 'http://sbmi.uth.tmc.edu/ontology/ochv#15579', 'http://sbmi.uth.tmc.edu/ontology/ochv#48325', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0750504', 'http://purl.obolibrary.org/obo/NCIT_C85550', 'http://purl.obolibrary.org/obo/NCIT_C25626', 'http://sbmi.uth.tmc.edu/ontology/ochv#16843', 'http://purl.bioontology.org/ontology/SNOMEDCT/52101004', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0150312', 'http://purl.bioontology.org/ontology/SNOMEDCT/732323009', 'http://sbmi.uth.tmc.edu/ontology/ochv#23876', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0237820', 'http://purl.bioontology.org/ontology/SNOMEDCT/732527009', 'http://sbmi.uth.tmc.edu/ontology/ochv#37863', 'http://sbmi.uth.tmc.edu/ontology/ochv#29391', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0332282', 'http://purl.bioontology.org/ontology/SNOMEDCT/255260001', 'http://purl.bioontology.org/ontology/SNOMEDCT/255234002', 'http://sbmi.uth.tmc.edu/ontology/ochv#3248', 'http://purl.bioontology.org/ontology/MESH/Q000150', 'http://purl.bioontology.org/ontology/SNOMEDCT/732470006', 'http://sbmi.uth.tmc.edu/ontology/ochv#3631', 'http://purl.bioontology.org/ontology/SNOMEDCT/419099009', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0678226', 'http://sbmi.uth.tmc.edu/ontology/ochv#42785', 'http://purl.obolibrary.org/obo/NCIT_C89272', 'http://purl.bioontology.org/ontology/SNOMEDCT/42752001']}\n",
      "Conclusion: our results were similar to published data except for the finding that infections were more common in infants and preschool children and that the mortality rate of pneumonia in patients with comorbidities was high.\n",
      "http://data.bioontology.org/annotator/?text=Conclusion%3A%20our%20results%20were%20similar%20to%20published%20data%20except%20for%20the%20finding%20that%20infections%20were%20more%20common%20in%20infants%20and%20preschool%20children%20and%20that%20the%20mortality%20rate%20of%20pneumonia%20in%20patients%20with%20comorbidities%20was%20high.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'matchTerm': ['PUBLISHED', 'PUBLISHED', 'DATA', 'EXCEPT FOR', 'EXCEPT FOR', 'EXCEPT FOR', 'FINDING', 'FINDING', 'FINDING', 'INFECTIONS', 'INFECTIONS', 'COMMON', 'COMMON', 'COMMON', 'COMMON', 'INFANTS', 'INFANTS', 'PRESCHOOL CHILDREN', 'MORTALITY RATE', 'MORTALITY RATE', 'PNEUMONIA', 'PNEUMONIA', 'PNEUMONIA', 'PNEUMONIA', 'PNEUMONIA', 'PNEUMONIA', 'PATIENTS', 'PATIENTS', 'PATIENTS', 'COMORBIDITIES', 'HIGH', 'HIGH', 'HIGH', 'HIGH'], 'ontology': ['http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/PREMEDONTO', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/PREMEDONTO', 'http://data.bioontology.org/ontologies/MESH', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/MESH', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/PREMEDONTO', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/MESH', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/MESH', 'http://data.bioontology.org/ontologies/HPIO', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/COVID-19', 'http://data.bioontology.org/ontologies/MESH', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/PREMEDONTO', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/SNOMEDCT', 'http://data.bioontology.org/ontologies/OCHV', 'http://data.bioontology.org/ontologies/PREMEDONTO'], 'annotatedClass': ['http://sbmi.uth.tmc.edu/ontology/ochv#C0034037', 'http://sbmi.uth.tmc.edu/ontology/ochv#10375', 'http://purl.obolibrary.org/obo/NCIT_C25474', 'http://sbmi.uth.tmc.edu/ontology/ochv#29406', 'http://purl.bioontology.org/ontology/SNOMEDCT/5185003', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0332300', 'http://purl.obolibrary.org/obo/NCIT_C3367', 'http://purl.bioontology.org/ontology/STY/T033', 'http://purl.bioontology.org/ontology/STY/T033', 'http://purl.bioontology.org/ontology/MESH/D007239', 'http://sbmi.uth.tmc.edu/ontology/ochv#6691', 'http://purl.obolibrary.org/obo/NCIT_C43461', 'http://purl.bioontology.org/ontology/SNOMEDCT/72906007', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0205214', 'http://sbmi.uth.tmc.edu/ontology/ochv#20409', 'http://sbmi.uth.tmc.edu/ontology/ochv#6675', 'http://purl.bioontology.org/ontology/MESH/D007223', 'http://sbmi.uth.tmc.edu/ontology/ochv#2789', 'http://purl.bioontology.org/ontology/SNOMEDCT/409651001', 'http://sbmi.uth.tmc.edu/ontology/ochv#8258', 'http://sbmi.uth.tmc.edu/ontology/ochv#9867', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0032285', 'http://purl.bioontology.org/ontology/MESH/D011014', 'http://www.semanticweb.org/ontologies/2010/5/22/Ontology1277229984000.owl#HPI:0000229', 'http://purl.bioontology.org/ontology/SNOMEDCT/233604007', 'http://purl.obolibrary.org/obo/DOID_552', 'http://purl.bioontology.org/ontology/MESH/D010361', 'http://sbmi.uth.tmc.edu/ontology/ochv#9372', 'http://purl.obolibrary.org/obo/NCIT_C16960', 'http://sbmi.uth.tmc.edu/ontology/ochv#3226', 'http://sbmi.uth.tmc.edu/ontology/ochv#C0205250', 'http://purl.bioontology.org/ontology/SNOMEDCT/75540009', 'http://sbmi.uth.tmc.edu/ontology/ochv#20438', 'http://purl.obolibrary.org/obo/NCIT_C25227']}\n",
      "ok abstract\n",
      "Mycoplasma pneumoniae is a common cause of upper and lower respiratory tract infections. It remains one of the most frequent causes of atypical pneumonia particu-larly among young adults. [1, 2, 3, 4, 5] Although it is highly transmissible, most infections caused by this organism are relatively minor and include pharyngitis, tracheobronchitis, bronchiolitis, and croup with one fifth of in-fections being asymptomatic. [6, 7] Only 3 -10% of infected subjects develop symptoms consistent with bronchopneumonia and mortality from infection is rare. [6, 7] The organism is fastidious and difficult to grow on cultures. Therefore, diagnosis of infections caused by this organism is usually confirmed with serological tests or polymerase chain reaction-gene amplification techniques. At King Abdulaziz University Hospital (KAUH), Jeddah, Saudi Arabia, the facility to perform Mycoplasma culture has been available since January 1997. As published information concerning M. pneumoniae infections in Saudi Arabia is scarce, [8, 9, 10] we wished to study the epidemiology and clinical features of cultureproven infections caused by this organism at this hospital.\n",
      "http://data.bioontology.org/annotator/?text=Mycoplasma%20pneumoniae%20is%20a%20common%20cause%20of%20upper%20and%20lower%20respiratory%20tract%20infections.%20It%20remains%20one%20of%20the%20most%20frequent%20causes%20of%20atypical%20pneumonia%20particu-larly%20among%20young%20adults.%20%5B1%2C%202%2C%203%2C%204%2C%205%5D%20Although%20it%20is%20highly%20transmissible%2C%20most%20infections%20caused%20by%20this%20organism%20are%20relatively%20minor%20and%20include%20pharyngitis%2C%20tracheobronchitis%2C%20bronchiolitis%2C%20and%20croup%20with%20one%20fifth%20of%20in-fections%20being%20asymptomatic.%20%5B6%2C%207%5D%20Only%203%20-10%25%20of%20infected%20subjects%20develop%20symptoms%20consistent%20with%20bronchopneumonia%20and%20mortality%20from%20infection%20is%20rare.%20%5B6%2C%207%5D%20The%20organism%20is%20fastidious%20and%20difficult%20to%20grow%20on%20cultures.%20Therefore%2C%20diagnosis%20of%20infections%20caused%20by%20this%20organism%20is%20usually%20confirmed%20with%20serological%20tests%20or%20polymerase%20chain%20reaction-gene%20amplification%20techniques.%20At%20King%20Abdulaziz%20University%20Hospital%20%28KAUH%29%2C%20Jeddah%2C%20Saudi%20Arabia%2C%20the%20facility%20to%20perform%20Mycoplasma%20culture%20has%20been%20available%20since%20January%201997.%20As%20published%20information%20concerning%20M.%20pneumoniae%20infections%20in%20Saudi%20Arabia%20is%20scarce%2C%20%5B8%2C%209%2C%2010%5D%20we%20wished%20to%20study%20the%20epidemiology%20and%20clinical%20features%20of%20cultureproven%20infections%20caused%20by%20this%20organism%20at%20this%20hospital.\n",
      "KAUH is a tertiary care teaching hospital with a bed capacity of 265 beds and annual admissions of 18000 to 19000 patients. Patients with M. pneumoniae positive cultures from respiratory specimens were identified over a 24-months\" period from January, 1997 through December, 1998 for this review.\n",
      "http://data.bioontology.org/annotator/?text=KAUH%20is%20a%20tertiary%20care%20teaching%20hospital%20with%20a%20bed%20capacity%20of%20265%20beds%20and%20annual%20admissions%20of%2018000%20to%2019000%20patients.%20Patients%20with%20M.%20pneumoniae%20positive%20cultures%20from%20respiratory%20specimens%20were%20identified%20over%20a%2024-months%22%20period%20from%20January%2C%201997%20through%20December%2C%201998%20for%20this%20review.\n",
      "During the study period, respiratory specimens (sputum, nasopharyngeal aspiration, endotracheal secretion, and bronchoalveolar lavage) for M. pneumoniae culture were obtained from patients with upper or lower respiratory tract infections seen as inpatients or in the outpatient or emergency departments. Respiratory specimens were aslo Gram-stained and cultured for bacteria and viruses. M. pneumoniae serological tests for IgG or IgM were not available at KAUH during the study period. All positive culture results were obtained from the Microbiology laboratory records. Charts of patients were reviewed with standardized data collection. Information collected included patients' demographics, comorbidities, clinical manifestations, complications, and outcome.\n",
      "http://data.bioontology.org/annotator/?text=During%20the%20study%20period%2C%20respiratory%20specimens%20%28sputum%2C%20nasopharyngeal%20aspiration%2C%20endotracheal%20secretion%2C%20and%20bronchoalveolar%20lavage%29%20for%20M.%20pneumoniae%20culture%20were%20obtained%20from%20patients%20with%20upper%20or%20lower%20respiratory%20tract%20infections%20seen%20as%20inpatients%20or%20in%20the%20outpatient%20or%20emergency%20departments.%20Respiratory%20specimens%20were%20aslo%20Gram-stained%20and%20cultured%20for%20bacteria%20and%20viruses.%20M.%20pneumoniae%20serological%20tests%20for%20IgG%20or%20IgM%20were%20not%20available%20at%20KAUH%20during%20the%20study%20period.%20All%20positive%20culture%20results%20were%20obtained%20from%20the%20Microbiology%20laboratory%20records.%20Charts%20of%20patients%20were%20reviewed%20with%20standardized%20data%20collection.%20Information%20collected%20included%20patients%27%20demographics%2C%20comorbidities%2C%20clinical%20manifestations%2C%20complications%2C%20and%20outcome.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "M. pneumoniae was cultured using the classic M. pneumoniae agar medium (M.P. agar) and the Pneumofast tray (Pneumofast ® , International Microbio, Signes, France). Specimens were processed according to the instructions of the manufacturer. The M.P. agars and Pneumofast trays were incubated anaerobically at 37°C and inspected daily for 4 weeks. The organism was identified based on typical colonial morphology (granular colonies, rarely fried-egg-like, 10-150 ∝ in diameter) on the M.P. agar medium and the change in the Pneumofast broth color from red to orange then to yellow (glucose fermentation) in the absence of turbidity of the broth. Antibiotic sensitivity profile on the Pneumofast tray was also used for identification according to the instructions of the manufacturer. Bacterial and viral cultures were performed using standard methods.\n",
      "http://data.bioontology.org/annotator/?text=M.%20pneumoniae%20was%20cultured%20using%20the%20classic%20M.%20pneumoniae%20agar%20medium%20%28M.P.%20agar%29%20and%20the%20Pneumofast%20tray%20%28Pneumofast%20%C2%AE%20%2C%20International%20Microbio%2C%20Signes%2C%20France%29.%20Specimens%20were%20processed%20according%20to%20the%20instructions%20of%20the%20manufacturer.%20The%20M.P.%20agars%20and%20Pneumofast%20trays%20were%20incubated%20anaerobically%20at%2037%C2%B0C%20and%20inspected%20daily%20for%204%20weeks.%20The%20organism%20was%20identified%20based%20on%20typical%20colonial%20morphology%20%28granular%20colonies%2C%20rarely%20fried-egg-like%2C%2010-150%20%E2%88%9D%20in%20diameter%29%20on%20the%20M.P.%20agar%20medium%20and%20the%20change%20in%20the%20Pneumofast%20broth%20color%20from%20red%20to%20orange%20then%20to%20yellow%20%28glucose%20fermentation%29%20in%20the%20absence%20of%20turbidity%20of%20the%20broth.%20Antibiotic%20sensitivity%20profile%20on%20the%20Pneumofast%20tray%20was%20also%20used%20for%20identification%20according%20to%20the%20instructions%20of%20the%20manufacturer.%20Bacterial%20and%20viral%20cultures%20were%20performed%20using%20standard%20methods.\n",
      "M. pneumoniae isolates were considered community-acquired if they were recovered from unhospitalized patients or within 72 hours of admission to the hospital, and nosocomial if they were recovered beyond that period.\n",
      "http://data.bioontology.org/annotator/?text=M.%20pneumoniae%20isolates%20were%20considered%20community-acquired%20if%20they%20were%20recovered%20from%20unhospitalized%20patients%20or%20within%2072%20hours%20of%20admission%20to%20the%20hospital%2C%20and%20nosocomial%20if%20they%20were%20recovered%20beyond%20that%20period.\n",
      "Pneumonia was diagnosed based on clinical symptoms and signs, along with radiographic evidence of pneumonia when possible. Severe pneumonia was defined as pneumonia associated with tachycardia (>140 /minute), tachypnoea (>30/minute), hypotension (Systolic blood pressure <90 mmHg), hypoxemia (arterial oxygen partial pressure <8 kPa or oxygen saturation <90%), and/or more than 2 areas of consolidation.\n",
      "http://data.bioontology.org/annotator/?text=Pneumonia%20was%20diagnosed%20based%20on%20clinical%20symptoms%20and%20signs%2C%20along%20with%20radiographic%20evidence%20of%20pneumonia%20when%20possible.%20Severe%20pneumonia%20was%20defined%20as%20pneumonia%20associated%20with%20tachycardia%20%28%3E140%20/minute%29%2C%20tachypnoea%20%28%3E30/minute%29%2C%20hypotension%20%28Systolic%20blood%20pressure%20%3C90%20mmHg%29%2C%20hypoxemia%20%28arterial%20oxygen%20partial%20pressure%20%3C8%20kPa%20or%20oxygen%20saturation%20%3C90%25%29%2C%20and/or%20more%20than%202%20areas%20of%20consolidation.\n",
      "Outcome of patients with M. pneumoniae infection was classified into 4 categories; uneventful recovery, recovery following complications, death due to M. pneumoniae infection, or death unrelated to M. pneumoniae infection.\n",
      "http://data.bioontology.org/annotator/?text=Outcome%20of%20patients%20with%20M.%20pneumoniae%20infection%20was%20classified%20into%204%20categories%3B%20uneventful%20recovery%2C%20recovery%20following%20complications%2C%20death%20due%20to%20M.%20pneumoniae%20infection%2C%20or%20death%20unrelated%20to%20M.%20pneumoniae%20infection.\n",
      "The Statistical Package for Social Sciences (SPSS) program was used for data analysis. Comparison of categorical data was by Chi-square statistic or Fisher's exact test for small expected values.\n",
      "http://data.bioontology.org/annotator/?text=The%20Statistical%20Package%20for%20Social%20Sciences%20%28SPSS%29%20program%20was%20used%20for%20data%20analysis.%20Comparison%20of%20categorical%20data%20was%20by%20Chi-square%20statistic%20or%20Fisher%27s%20exact%20test%20for%20small%20expected%20values.\n",
      "A total of 40 respiratory specimens from 40 patients were positive for M. pneumoniae over the 24-months study period. The demographic and epidemiological characteristics of the patients are summarized in Table 1 . Of all isolates, 37 (92.5%) were community-acquired and 3 (7.5%) were nosocomial. Thirty-three (82.5%) patients required admission to the hospital and the remaining 7 (17.5%) were treated as outpatients. Twenty-four isolates (60%) were associated with pneumonia, 14 (35%) with upper respiratory tract infections, and 2 (5%) with bronchiolitis. Of the 24 cases of pneumonia, 21 were confirmed radiologically and the remaining 3 were diagnosed clinically. The two cases of bronchiolitis occurred in 2 children, one and three years old. Thirty-one patients (77.5%) had comorbidities. Eleven patients (27.5%) had cardiopulmonary comorbidities (asthma, 8, lung fibrosis, 1, congestive heart failure, 1, congenial heart disease, 1), 9 patients (22.5%) were immunocompromised (malignancy, 7, steroid therapy, 3, Human immunodeficiency virus infection, 1), and 11 patients (27.5%) had other comorbidities (premature newborns, 2, and one each of myelodysplastic syndrome, myelopro-liferative disorder, sickle cell anemia, Evan's syndrome, Down syndrome, sarcoidosis, demyelinating disease, cerebral palsy, and spinal muscle atrophy). Organisms concomitantly isolated with M. pneumoniae from the respiratory tract included herpes simplex virus type 1 (2 occasions), adenovirus (2 occasions), cytomegalo virus (1 occasion), respiratory syncytial virus (1 occasion), and bacterial isolates (2 occasions: Acinetobacter species, 1, and Enter obacter cloacae, 1).\n",
      "http://data.bioontology.org/annotator/?text=A%20total%20of%2040%20respiratory%20specimens%20from%2040%20patients%20were%20positive%20for%20M.%20pneumoniae%20over%20the%2024-months%20study%20period.%20The%20demographic%20and%20epidemiological%20characteristics%20of%20the%20patients%20are%20summarized%20in%20Table%201%20.%20Of%20all%20isolates%2C%2037%20%2892.5%25%29%20were%20community-acquired%20and%203%20%287.5%25%29%20were%20nosocomial.%20Thirty-three%20%2882.5%25%29%20patients%20required%20admission%20to%20the%20hospital%20and%20the%20remaining%207%20%2817.5%25%29%20were%20treated%20as%20outpatients.%20Twenty-four%20isolates%20%2860%25%29%20were%20associated%20with%20pneumonia%2C%2014%20%2835%25%29%20with%20upper%20respiratory%20tract%20infections%2C%20and%202%20%285%25%29%20with%20bronchiolitis.%20Of%20the%2024%20cases%20of%20pneumonia%2C%2021%20were%20confirmed%20radiologically%20and%20the%20remaining%203%20were%20diagnosed%20clinically.%20The%20two%20cases%20of%20bronchiolitis%20occurred%20in%202%20children%2C%20one%20and%20three%20years%20old.%20Thirty-one%20patients%20%2877.5%25%29%20had%20comorbidities.%20Eleven%20patients%20%2827.5%25%29%20had%20cardiopulmonary%20comorbidities%20%28asthma%2C%208%2C%20lung%20fibrosis%2C%201%2C%20congestive%20heart%20failure%2C%201%2C%20congenial%20heart%20disease%2C%201%29%2C%209%20patients%20%2822.5%25%29%20were%20immunocompromised%20%28malignancy%2C%207%2C%20steroid%20therapy%2C%203%2C%20Human%20immunodeficiency%20virus%20infection%2C%201%29%2C%20and%2011%20patients%20%2827.5%25%29%20had%20other%20comorbidities%20%28premature%20newborns%2C%202%2C%20and%20one%20each%20of%20myelodysplastic%20syndrome%2C%20myelopro-liferative%20disorder%2C%20sickle%20cell%20anemia%2C%20Evan%27s%20syndrome%2C%20Down%20syndrome%2C%20sarcoidosis%2C%20demyelinating%20disease%2C%20cerebral%20palsy%2C%20and%20spinal%20muscle%20atrophy%29.%20Organisms%20concomitantly%20isolated%20with%20M.%20pneumoniae%20from%20the%20respiratory%20tract%20included%20herpes%20simplex%20virus%20type%201%20%282%20occasions%29%2C%20adenovirus%20%282%20occasions%29%2C%20cytomegalo%20virus%20%281%20occasion%29%2C%20respiratory%20syncytial%20virus%20%281%20occasion%29%2C%20and%20bacterial%20isolates%20%282%20occasions%3A%20Acinetobacter%20species%2C%201%2C%20and%20Enter%20obacter%20cloacae%2C%201%29.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Clinical manifestations associated with M. pneumoniae infections are summarized in Table 2 . Pneumonia was more common than upper respiratory tract infections (57.5 % versus 27.5%, respectively). Immunocompromised patients were more likely to present with pneumonia as opposed to upper respiratory tract infection or bronchiolitis than non-immunocompromised patients (8/9 versus 16/31, P = 0.05). Similarly, there was a tendency for patients 60 years of age or older to present with pneumonia more frequently than those below 60 (4/4 versus 20/36, P = 0.1). Of the 24 patients with clinically or radiologically confirmed pneumonia, 19 (79.2%) had crepitations and only 6 (25%) had bronchial breath sounds on physical examination. Of the 16 patients in whom wheezes were detected, 9 (56.3%) were not known to have asthma or other obstructive airway disease. Table 3 . Of the 24 patients with pneumonia, 21 (87.5%) were admitted to the hospital, and 20 (83.3%) had comorbidities. All patients with upper respiratory tract infections (11 patients) or bronchiolitis (2 patients) had uneventful recovery. Of the 24 patients with pneumonia, 14 (58.3%) had uneventful recovery, 4 (16.7%) recovered following some complications (acute respiratory distress syndrome, 2, respiratory failure, 1, septic shock, 1), 3 (12.5%) died because of M pneumoniae infection, and 3 (12.5%) died due to underlying comorbidities. The 3 patients who died of M pneumoniae pneumonia had other comorbidities; one had congestive heart failure, the second had congenital heart disease, and the third was a 3months old infant born prematurely at 32 weeks of gestation who previously had 3 episodes of pneumonia due to other pathogens.\n",
      "http://data.bioontology.org/annotator/?text=Clinical%20manifestations%20associated%20with%20M.%20pneumoniae%20infections%20are%20summarized%20in%20Table%202%20.%20Pneumonia%20was%20more%20common%20than%20upper%20respiratory%20tract%20infections%20%2857.5%20%25%20versus%2027.5%25%2C%20respectively%29.%20Immunocompromised%20patients%20were%20more%20likely%20to%20present%20with%20pneumonia%20as%20opposed%20to%20upper%20respiratory%20tract%20infection%20or%20bronchiolitis%20than%20non-immunocompromised%20patients%20%288/9%20versus%2016/31%2C%20P%20%3D%200.05%29.%20Similarly%2C%20there%20was%20a%20tendency%20for%20patients%2060%20years%20of%20age%20or%20older%20to%20present%20with%20pneumonia%20more%20frequently%20than%20those%20below%2060%20%284/4%20versus%2020/36%2C%20P%20%3D%200.1%29.%20Of%20the%2024%20patients%20with%20clinically%20or%20radiologically%20confirmed%20pneumonia%2C%2019%20%2879.2%25%29%20had%20crepitations%20and%20only%206%20%2825%25%29%20had%20bronchial%20breath%20sounds%20on%20physical%20examination.%20Of%20the%2016%20patients%20in%20whom%20wheezes%20were%20detected%2C%209%20%2856.3%25%29%20were%20not%20known%20to%20have%20asthma%20or%20other%20obstructive%20airway%20disease.%20Table%203%20.%20Of%20the%2024%20patients%20with%20pneumonia%2C%2021%20%2887.5%25%29%20were%20admitted%20to%20the%20hospital%2C%20and%2020%20%2883.3%25%29%20had%20comorbidities.%20All%20patients%20with%20upper%20respiratory%20tract%20infections%20%2811%20patients%29%20or%20bronchiolitis%20%282%20patients%29%20had%20uneventful%20recovery.%20Of%20the%2024%20patients%20with%20pneumonia%2C%2014%20%2858.3%25%29%20had%20uneventful%20recovery%2C%204%20%2816.7%25%29%20recovered%20following%20some%20complications%20%28acute%20respiratory%20distress%20syndrome%2C%202%2C%20respiratory%20failure%2C%201%2C%20septic%20shock%2C%201%29%2C%203%20%2812.5%25%29%20died%20because%20of%20M%20pneumoniae%20infection%2C%20and%203%20%2812.5%25%29%20died%20due%20to%20underlying%20comorbidities.%20The%203%20patients%20who%20died%20of%20M%20pneumoniae%20pneumonia%20had%20other%20comorbidities%3B%20one%20had%20congestive%20heart%20failure%2C%20the%20second%20had%20congenital%20heart%20disease%2C%20and%20the%20third%20was%20a%203months%20old%20infant%20born%20prematurely%20at%2032%20weeks%20of%20gestation%20who%20previously%20had%203%20episodes%20of%20pneumonia%20due%20to%20other%20pathogens.\n",
      "Mycoplasma pneumoniae is one of the most common causes of atypical pneumonia accounting for 5-23% of community-acquired pneumonia, [1, 2, 3, 4, 5] In a study of 511 children with acute respiratory tract infection in Riyadh, Saudi Arabia, Mycoplasma pneumoniae was found to be the second most common causative agent after Respiratory syncytial virus (RSV) accounting for 9% of all cases, [8] In a study of 112 adult patients with community acquired pneumonia admitted to a military hospital in Riyadh, Saudi Arabia, this organism accounted for 6% of all cases, [9] In another retrospective study of 567 pneumonic episodes in adult patients from Al-Qassim area, the organism accounted for 23% of all episodes, [10] The organism also causes other relatively minor infections such as pharyngitis, tracheobronchitis, bronchiolitis, and croup. It is transmitted from person-to-person by infected respiratory droplets during close contact. It is most common in school-aged children, military recruits, and college students. [11] Most cases occur singly or as family outbreaks. Larger outbreaks can also occur in closed populations such as military recruit camps or boarding schools, [12] Infection occurs most frequently during the fall and winter in temperate climates but may develop year-round, [13] The average incubation period is 3 weeks following exposure, [6] Although rare, complications are protean and may involve virtually any organ system such as the respiratory system (e.g.: pleurisy, pneumothorax, acute respiratory distress syndrome, lung abscess), the hematologic system (e.g.: hemolytic anemia, intravascular coagulation, thrombocytopenia), the dermatologic system (e.g.: maculopapular or urticarial rashes, erythema multiforme, erythema nodosum), the musculoskeletal system (e.g.: myalgias, arthralgias, arthritis), the cardiovascular system (e.g.: pericarditis, myocarditis), the nervous system (e.g.: meningoencephalitis, Guillain-Barre syndrome, neuropathies, acute psychosis), or the eye (optic disc edema, optic nerve atrophy, retinal exudation and hemorrhages). [6, 7, 14, 15, 16, 17, 18] Immunity following infection is not long lasting. [11] In our study, the infection affected all age groups but was most common in infants (32.5%) and preschool children (22.5%), and least common in adults aged 15 to 30 years (2.5%) and elderly above 70 years of age (5%). This contrasts with data from temperate countries where the infection is most common in school-aged children, and young adults. [11] One possible explanation for this difference is that infants and preschool children perhaps had more severe infections than did school-aged children, and young adults which prompted presentation of the former group to the hospital. The infection occurred year-round but was most common in the fall (35%), and spring (30%), and least common in the summer (10%). Most infections were community-acquired (92.5%).\n",
      "http://data.bioontology.org/annotator/?text=Mycoplasma%20pneumoniae%20is%20one%20of%20the%20most%20common%20causes%20of%20atypical%20pneumonia%20accounting%20for%205-23%25%20of%20community-acquired%20pneumonia%2C%20%5B1%2C%202%2C%203%2C%204%2C%205%5D%20In%20a%20study%20of%20511%20children%20with%20acute%20respiratory%20tract%20infection%20in%20Riyadh%2C%20Saudi%20Arabia%2C%20Mycoplasma%20pneumoniae%20was%20found%20to%20be%20the%20second%20most%20common%20causative%20agent%20after%20Respiratory%20syncytial%20virus%20%28RSV%29%20accounting%20for%209%25%20of%20all%20cases%2C%20%5B8%5D%20In%20a%20study%20of%20112%20adult%20patients%20with%20community%20acquired%20pneumonia%20admitted%20to%20a%20military%20hospital%20in%20Riyadh%2C%20Saudi%20Arabia%2C%20this%20organism%20accounted%20for%206%25%20of%20all%20cases%2C%20%5B9%5D%20In%20another%20retrospective%20study%20of%20567%20pneumonic%20episodes%20in%20adult%20patients%20from%20Al-Qassim%20area%2C%20the%20organism%20accounted%20for%2023%25%20of%20all%20episodes%2C%20%5B10%5D%20The%20organism%20also%20causes%20other%20relatively%20minor%20infections%20such%20as%20pharyngitis%2C%20tracheobronchitis%2C%20bronchiolitis%2C%20and%20croup.%20It%20is%20transmitted%20from%20person-to-person%20by%20infected%20respiratory%20droplets%20during%20close%20contact.%20It%20is%20most%20common%20in%20school-aged%20children%2C%20military%20recruits%2C%20and%20college%20students.%20%5B11%5D%20Most%20cases%20occur%20singly%20or%20as%20family%20outbreaks.%20Larger%20outbreaks%20can%20also%20occur%20in%20closed%20populations%20such%20as%20military%20recruit%20camps%20or%20boarding%20schools%2C%20%5B12%5D%20Infection%20occurs%20most%20frequently%20during%20the%20fall%20and%20winter%20in%20temperate%20climates%20but%20may%20develop%20year-round%2C%20%5B13%5D%20The%20average%20incubation%20period%20is%203%20weeks%20following%20exposure%2C%20%5B6%5D%20Although%20rare%2C%20complications%20are%20protean%20and%20may%20involve%20virtually%20any%20organ%20system%20such%20as%20the%20respiratory%20system%20%28e.g.%3A%20pleurisy%2C%20pneumothorax%2C%20acute%20respiratory%20distress%20syndrome%2C%20lung%20abscess%29%2C%20the%20hematologic%20system%20%28e.g.%3A%20hemolytic%20anemia%2C%20intravascular%20coagulation%2C%20thrombocytopenia%29%2C%20the%20dermatologic%20system%20%28e.g.%3A%20maculopapular%20or%20urticarial%20rashes%2C%20erythema%20multiforme%2C%20erythema%20nodosum%29%2C%20the%20musculoskeletal%20system%20%28e.g.%3A%20myalgias%2C%20arthralgias%2C%20arthritis%29%2C%20the%20cardiovascular%20system%20%28e.g.%3A%20pericarditis%2C%20myocarditis%29%2C%20the%20nervous%20system%20%28e.g.%3A%20meningoencephalitis%2C%20Guillain-Barre%20syndrome%2C%20neuropathies%2C%20acute%20psychosis%29%2C%20or%20the%20eye%20%28optic%20disc%20edema%2C%20optic%20nerve%20atrophy%2C%20retinal%20exudation%20and%20hemorrhages%29.%20%5B6%2C%207%2C%2014%2C%2015%2C%2016%2C%2017%2C%2018%5D%20Immunity%20following%20infection%20is%20not%20long%20lasting.%20%5B11%5D%20In%20our%20study%2C%20the%20infection%20affected%20all%20age%20groups%20but%20was%20most%20common%20in%20infants%20%2832.5%25%29%20and%20preschool%20children%20%2822.5%25%29%2C%20and%20least%20common%20in%20adults%20aged%2015%20to%2030%20years%20%282.5%25%29%20and%20elderly%20above%2070%20years%20of%20age%20%285%25%29.%20This%20contrasts%20with%20data%20from%20temperate%20countries%20where%20the%20infection%20is%20most%20common%20in%20school-aged%20children%2C%20and%20young%20adults.%20%5B11%5D%20One%20possible%20explanation%20for%20this%20difference%20is%20that%20infants%20and%20preschool%20children%20perhaps%20had%20more%20severe%20infections%20than%20did%20school-aged%20children%2C%20and%20young%20adults%20which%20prompted%20presentation%20of%20the%20former%20group%20to%20the%20hospital.%20The%20infection%20occurred%20year-round%20but%20was%20most%20common%20in%20the%20fall%20%2835%25%29%2C%20and%20spring%20%2830%25%29%2C%20and%20least%20common%20in%20the%20summer%20%2810%25%29.%20Most%20infections%20were%20community-acquired%20%2892.5%25%29.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "More than one half of patients (57.5%) presented with pneumonia, and about a third (27.5%) presented with upper respiratory tract infection, Immunocompromised patients and patients 60 years of age or older were more likely to present with pneumonia as opposed to upper respiratory tract infection than non-immunocompromised patients or those below 60 years of age. Cough (82.5%), fever (75%), and malaise (58.8%) were the most common presenting symptoms. Cough was usually dry or slightly productive of white sputum and mild to moderate in severity. Most febrile patients had mild to mod- erate fever of 39°C or less; high-grade fever of more than 39°C was rare. Crepitations (60%), and wheezes (40%) were the most common signs. Wheezes were as common in patients with no history of obstructive airway disease (9 patients) as it was in those with such a history (7 patients). Bronchial breathing as a sign of consolidation was detected in only one fourth of patients with pneumonia, which is consistent with the known disparity between clinical and radiological signs of M pneumoniae pneumonia. Crepitations, however, were detected in the majority (79.2%) of patients. Pleuritic chest pain and pleural effusion were rare.\n",
      "http://data.bioontology.org/annotator/?text=More%20than%20one%20half%20of%20patients%20%2857.5%25%29%20presented%20with%20pneumonia%2C%20and%20about%20a%20third%20%2827.5%25%29%20presented%20with%20upper%20respiratory%20tract%20infection%2C%20Immunocompromised%20patients%20and%20patients%2060%20years%20of%20age%20or%20older%20were%20more%20likely%20to%20present%20with%20pneumonia%20as%20opposed%20to%20upper%20respiratory%20tract%20infection%20than%20non-immunocompromised%20patients%20or%20those%20below%2060%20years%20of%20age.%20Cough%20%2882.5%25%29%2C%20fever%20%2875%25%29%2C%20and%20malaise%20%2858.8%25%29%20were%20the%20most%20common%20presenting%20symptoms.%20Cough%20was%20usually%20dry%20or%20slightly%20productive%20of%20white%20sputum%20and%20mild%20to%20moderate%20in%20severity.%20Most%20febrile%20patients%20had%20mild%20to%20mod-%20erate%20fever%20of%2039%C2%B0C%20or%20less%3B%20high-grade%20fever%20of%20more%20than%2039%C2%B0C%20was%20rare.%20Crepitations%20%2860%25%29%2C%20and%20wheezes%20%2840%25%29%20were%20the%20most%20common%20signs.%20Wheezes%20were%20as%20common%20in%20patients%20with%20no%20history%20of%20obstructive%20airway%20disease%20%289%20patients%29%20as%20it%20was%20in%20those%20with%20such%20a%20history%20%287%20patients%29.%20Bronchial%20breathing%20as%20a%20sign%20of%20consolidation%20was%20detected%20in%20only%20one%20fourth%20of%20patients%20with%20pneumonia%2C%20which%20is%20consistent%20with%20the%20known%20disparity%20between%20clinical%20and%20radiological%20signs%20of%20M%20pneumoniae%20pneumonia.%20Crepitations%2C%20however%2C%20were%20detected%20in%20the%20majority%20%2879.2%25%29%20of%20patients.%20Pleuritic%20chest%20pain%20and%20pleural%20effusion%20were%20rare.\n",
      "More than half (56.5%) of the patients with pneumonia had uneventful recovery. Mortality from M. pneumoniae pneumonia was high (12.5%) and occurred only in patients with underlying comorbidities. None of the 9 patients with no underlying comorbidities died of M pneumoniae pneumonia. The relatively high complications rate (16.7%) and mortality (12.5%) related to M. pneumoniae pneumonia are likely due to selection bias as most patients with pneumonia were sick enough to require admission to the hospital (21/24 or 87.5%) and most of them had comorbidities (20/24 or 83.3%).\n",
      "http://data.bioontology.org/annotator/?text=More%20than%20half%20%2856.5%25%29%20of%20the%20patients%20with%20pneumonia%20had%20uneventful%20recovery.%20Mortality%20from%20M.%20pneumoniae%20pneumonia%20was%20high%20%2812.5%25%29%20and%20occurred%20only%20in%20patients%20with%20underlying%20comorbidities.%20None%20of%20the%209%20patients%20with%20no%20underlying%20comorbidities%20died%20of%20M%20pneumoniae%20pneumonia.%20The%20relatively%20high%20complications%20rate%20%2816.7%25%29%20and%20mortality%20%2812.5%25%29%20related%20to%20M.%20pneumoniae%20pneumonia%20are%20likely%20due%20to%20selection%20bias%20as%20most%20patients%20with%20pneumonia%20were%20sick%20enough%20to%20require%20admission%20to%20the%20hospital%20%2821/24%20or%2087.5%25%29%20and%20most%20of%20them%20had%20comorbidities%20%2820/24%20or%2083.3%25%29.\n",
      "In conclusion, our data shed some light on the epidemiology and clinical features of M pneumoniae infections in one of the Saudi tertiary care centers. The data are comparable to those of other countries except for the finding that infections were more common in infants and preschool children than in school children and young adults. Additionally, mortality attributable to M. pneumoniae pneumonia was relatively high in patients with comorbidities. It is hoped this information will assist clinicians in their approach and management of respiratory tract infections.\n",
      "http://data.bioontology.org/annotator/?text=In%20conclusion%2C%20our%20data%20shed%20some%20light%20on%20the%20epidemiology%20and%20clinical%20features%20of%20M%20pneumoniae%20infections%20in%20one%20of%20the%20Saudi%20tertiary%20care%20centers.%20The%20data%20are%20comparable%20to%20those%20of%20other%20countries%20except%20for%20the%20finding%20that%20infections%20were%20more%20common%20in%20infants%20and%20preschool%20children%20than%20in%20school%20children%20and%20young%20adults.%20Additionally%2C%20mortality%20attributable%20to%20M.%20pneumoniae%20pneumonia%20was%20relatively%20high%20in%20patients%20with%20comorbidities.%20It%20is%20hoped%20this%20information%20will%20assist%20clinicians%20in%20their%20approach%20and%20management%20of%20respiratory%20tract%20infections.\n",
      "http://data.bioontology.org/annotator/?text=Clinical%20features%20of%20culture-proven%20Mycoplasma%20pneumoniae%20infections%20at%20King\n",
      "paper index: 13 processed\n"
     ]
    }
   ],
   "source": [
    "ontologies = \"MESH,COVID-19,HPIO,SNOMEDCT,OCHV,PREMEDONTO\"\n",
    "\n",
    "PARAMS = {'apikey':API_KEY,'ontologies':ontologies,\n",
    "          'expand_class_hierarchy':'false','class_hierarchy_max_level':0,\n",
    "           'longest_only':'true'\n",
    "         }\n",
    "\n",
    "#parse_directory()\n",
    "\n",
    "path_files = [samp_r1,samp_r2]\n",
    "\n",
    "for idx, path in enumerate(path_files):\n",
    "    \n",
    "    load_json_record(path)\n",
    "    emty_sections = []\n",
    "    item = dbMain()\n",
    "        \n",
    "    try: \n",
    "        \n",
    "        for idx, _chunk in enumerate(json_record['abstract']):\n",
    "\n",
    "            #make url here\n",
    "            text_to_annotate = _chunk['text']\n",
    "            print(text_to_annotate)\n",
    "\n",
    "            if text_to_annotate:\n",
    "                make_url_query(text_to_annotate)\n",
    "                print(URL)\n",
    "\n",
    "                #call api here\n",
    "                call_endpoint_annotator(URL,PARAMS)\n",
    "                #init db schema annotations\n",
    "                item.restart_annotations()\n",
    "\n",
    "                #proccess and store resultsfor section here\n",
    "                for i in range(len(json_api_results)):\n",
    "                    item.update_annotations(json_api_results[i]['annotations'][0]['text'],\n",
    "                                            json_api_results[i]['annotatedClass']['links']['ontology'],\n",
    "                                            json_api_results[i]['annotatedClass']['@id']\n",
    "                                           )\n",
    "                print(item.annotations)\n",
    "                item.update_chunks(_chunk['text'],\n",
    "                                   _chunk['section'],\n",
    "                                   item.annotations\n",
    "                                  )\n",
    "    \n",
    "            else:\n",
    "                emty_sections.append(idx)\n",
    "\n",
    "        print('ok abstract')\n",
    "        #Init api params\n",
    "\n",
    "        for idx, _chunk in enumerate(json_record['body_text']):\n",
    "\n",
    "            #make url here\n",
    "            text_to_annotate = _chunk['text']\n",
    "            print(text_to_annotate)\n",
    "\n",
    "            if text_to_annotate:\n",
    "                make_url_query(text_to_annotate)\n",
    "                print(URL)\n",
    "\n",
    "                #call api here\n",
    "                call_endpoint_annotator(URL,PARAMS)\n",
    "                #init schema annotations\n",
    "                item.restart_annotations()\n",
    "\n",
    "                #proccess and store resultsfor section here\n",
    "                for i in range(len(json_api_results)):\n",
    "                    item.update_annotations(json_api_results[i]['annotations'][0]['text'],\n",
    "                                            json_api_results[i]['annotatedClass']['links']['ontology'],\n",
    "                                            json_api_results[i]['annotatedClass']['@id']\n",
    "                                           )\n",
    "                item.update_chunks(_chunk['text'],\n",
    "                                   _chunk['section'],\n",
    "                                   item.annotations\n",
    "                                  )\n",
    "            else:\n",
    "                emty_sections.append(idx)\n",
    "                \n",
    "    except KeyError:\n",
    "\n",
    "        for idx, _chunk in enumerate(json_record['body_text']):\n",
    "\n",
    "            #make url here\n",
    "            text_to_annotate = _chunk['text']\n",
    "            print(text_to_annotate)\n",
    "\n",
    "            if text_to_annotate:\n",
    "                make_url_query(text_to_annotate)\n",
    "                print(URL)\n",
    "\n",
    "                #call api here\n",
    "                call_endpoint_annotator(URL,PARAMS)\n",
    "                #init schema annotations\n",
    "                item.restart_annotations()\n",
    "                #proccess and store resultsfor section here\n",
    "                for i in range(len(json_api_results)):\n",
    "                    item.update_annotations(json_api_results[i]['annotations'][0]['text'],\n",
    "                                            json_api_results[i]['annotatedClass']['links']['ontology'],\n",
    "                                            json_api_results[i]['annotatedClass']['@id']\n",
    "\n",
    "                                           )\n",
    "                item.update_chunks(_chunk['text'],\n",
    "                                   _chunk['section'],\n",
    "                                   item.annotations\n",
    "                                  )\n",
    "            else:\n",
    "                emty_sections.append(idx)\n",
    "\n",
    "    finally:\n",
    "        #for title\n",
    "        text_to_annotate = json_record['metadata']['title']\n",
    "        \n",
    "        if text_to_annotate:\n",
    "            make_url_query(text_to_annotate)\n",
    "            print(URL)\n",
    "\n",
    "            #call api here\n",
    "            call_endpoint_annotator(URL,PARAMS)\n",
    "            #init schema annotations\n",
    "            \n",
    "            item.restart_annotations()\n",
    "            #proccess and store resultsfor section here\n",
    "            for i in range(len(json_api_results)):\n",
    "                item.update_annotations(json_api_results[i]['annotations'][0]['text'],\n",
    "                                        json_api_results[i]['annotatedClass']['links']['ontology'],\n",
    "                                        json_api_results[i]['annotatedClass']['@id']\n",
    "\n",
    "                                           )\n",
    "            item.update_chunks(_chunk['text'],\n",
    "                                   _chunk['section'],\n",
    "                                   item.annotations\n",
    "                                  )\n",
    "        item.update_item_collection(json_record['paper_id'], item.chunks)\n",
    "        print('paper index:',idx,'processed')\n",
    "        item.update_db_main()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['paper_id', 'chunks_paper'])"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# make some test her\n",
    "one_item = dbMain.db[0]\n",
    "one_item.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['texts', 'sections', 'annotations'])"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "one_item['chunks_paper'][0].keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Mycoplasma pneumoniae is a common cause of upper and lower respiratory tract infections. It remains one of the most frequent causes of atypical pneumonia particularly among young adults.[1,2,3,4,5] Although it is highly transmissible, most infections caused by this organism are relatively minor and include pharyngitis, tracheobronchitis, bronchiolitis, and croup with one fifth of infections being asymptomatic.[6,7] Only 3 -10% of infected subjects develop symptoms consistent with bronchopneumonia and mortality from infection is rare.[6,7] The organism is fastidious and difficult to grow on cultures. Therefore, diagnosis of infections caused by this organism is usually confirmed with serological tests or polymerase chain reaction-gene amplification techniques. At King Abdulaziz University Hospital (KAUH), Jeddah, Saudi Arabia, the facility to perform Mycoplasma culture has been available since January 1997. As published information concerning M. pneumoniae infections in Saudi Arabia is scarce,[8,9,10] we wished to study the epidemiology and clinical features of culture-proven infections caused by this organism at this hospital.'"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "one_item['chunks_paper'][0]['texts'][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Introduction'"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "one_item['chunks_paper'][0]['sections'][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'matchTerm': ['MYCOPLASMA PNEUMONIAE',\n",
       "  'MYCOPLASMA PNEUMONIAE',\n",
       "  'MYCOPLASMA PNEUMONIAE',\n",
       "  'MYCOPLASMA PNEUMONIAE',\n",
       "  'IS A',\n",
       "  'COMMON',\n",
       "  'COMMON',\n",
       "  'COMMON',\n",
       "  'COMMON',\n",
       "  'CAUSE',\n",
       "  'CAUSE',\n",
       "  'UPPER',\n",
       "  'UPPER',\n",
       "  'UPPER',\n",
       "  'LOWER RESPIRATORY TRACT INFECTIONS',\n",
       "  'FREQUENT',\n",
       "  'FREQUENT',\n",
       "  'FREQUENT',\n",
       "  'FREQUENT',\n",
       "  'ATYPICAL PNEUMONIA',\n",
       "  'ATYPICAL PNEUMONIA',\n",
       "  'ATYPICAL PNEUMONIA',\n",
       "  'YOUNG',\n",
       "  'YOUNG',\n",
       "  'YOUNG',\n",
       "  'ADULTS',\n",
       "  'ADULTS',\n",
       "  'HIGHLY',\n",
       "  'INFECTIONS',\n",
       "  'INFECTIONS',\n",
       "  'ORGANISM',\n",
       "  'ORGANISM',\n",
       "  'ORGANISM',\n",
       "  'ORGANISM',\n",
       "  'ORGANISM',\n",
       "  'ORGANISM',\n",
       "  'ORGANISM',\n",
       "  'MINOR',\n",
       "  'MINOR',\n",
       "  'MINOR',\n",
       "  'MINOR',\n",
       "  'MINOR',\n",
       "  'MINOR',\n",
       "  'INCLUDE',\n",
       "  'INCLUDE',\n",
       "  'PHARYNGITIS',\n",
       "  'PHARYNGITIS',\n",
       "  'PHARYNGITIS',\n",
       "  'PHARYNGITIS',\n",
       "  'PHARYNGITIS',\n",
       "  'TRACHEOBRONCHITIS',\n",
       "  'TRACHEOBRONCHITIS',\n",
       "  'TRACHEOBRONCHITIS',\n",
       "  'BRONCHIOLITIS',\n",
       "  'BRONCHIOLITIS',\n",
       "  'BRONCHIOLITIS',\n",
       "  'CROUP',\n",
       "  'CROUP',\n",
       "  'CROUP',\n",
       "  'CROUP',\n",
       "  'FIFTH',\n",
       "  'ASYMPTOMATIC',\n",
       "  'ASYMPTOMATIC',\n",
       "  'ASYMPTOMATIC',\n",
       "  'INFECTED',\n",
       "  'INFECTED',\n",
       "  'CONSISTENT WITH',\n",
       "  'BRONCHOPNEUMONIA',\n",
       "  'BRONCHOPNEUMONIA',\n",
       "  'BRONCHOPNEUMONIA',\n",
       "  'BRONCHOPNEUMONIA',\n",
       "  'MORTALITY',\n",
       "  'MORTALITY',\n",
       "  'MORTALITY',\n",
       "  'MORTALITY',\n",
       "  'MORTALITY',\n",
       "  'INFECTION',\n",
       "  'INFECTION',\n",
       "  'RARE',\n",
       "  'RARE',\n",
       "  'RARE',\n",
       "  'DIFFICULT',\n",
       "  'DIFFICULT',\n",
       "  'DIFFICULT',\n",
       "  'DIFFICULT',\n",
       "  'CULTURES',\n",
       "  'DIAGNOSIS',\n",
       "  'DIAGNOSIS',\n",
       "  'DIAGNOSIS',\n",
       "  'DIAGNOSIS',\n",
       "  'DIAGNOSIS',\n",
       "  'DIAGNOSIS',\n",
       "  'DIAGNOSIS',\n",
       "  'DIAGNOSIS',\n",
       "  'DIAGNOSIS',\n",
       "  'DIAGNOSIS',\n",
       "  'DIAGNOSIS',\n",
       "  'CONFIRMED',\n",
       "  'CONFIRMED',\n",
       "  'CONFIRMED',\n",
       "  'CONFIRMED',\n",
       "  'SEROLOGICAL TESTS',\n",
       "  'POLYMERASE CHAIN REACTION',\n",
       "  'POLYMERASE CHAIN REACTION',\n",
       "  'POLYMERASE CHAIN REACTION',\n",
       "  'GENE AMPLIFICATION',\n",
       "  'GENE AMPLIFICATION',\n",
       "  'TECHNIQUES',\n",
       "  'UNIVERSITY HOSPITAL',\n",
       "  'UNIVERSITY HOSPITAL',\n",
       "  'SAUDI ARABIA',\n",
       "  'SAUDI ARABIA',\n",
       "  'SAUDI ARABIA',\n",
       "  'SAUDI ARABIA',\n",
       "  'FACILITY',\n",
       "  'PERFORM',\n",
       "  'MYCOPLASMA CULTURE',\n",
       "  'AVAILABLE',\n",
       "  'AVAILABLE',\n",
       "  '1997',\n",
       "  'PUBLISHED',\n",
       "  'PUBLISHED',\n",
       "  'INFORMATION',\n",
       "  'INFORMATION',\n",
       "  'INFORMATION',\n",
       "  'STUDY',\n",
       "  'STUDY',\n",
       "  'STUDY',\n",
       "  'STUDY',\n",
       "  'EPIDEMIOLOGY',\n",
       "  'EPIDEMIOLOGY',\n",
       "  'EPIDEMIOLOGY',\n",
       "  'EPIDEMIOLOGY',\n",
       "  'CLINICAL',\n",
       "  'CLINICAL',\n",
       "  'CLINICAL',\n",
       "  'CLINICAL',\n",
       "  'CULTURE',\n",
       "  'CULTURE',\n",
       "  'CULTURE',\n",
       "  'CULTURE',\n",
       "  'PROVEN',\n",
       "  'PROVEN',\n",
       "  'PROVEN',\n",
       "  'PROVEN',\n",
       "  'PROVEN',\n",
       "  'HOSPITAL',\n",
       "  'HOSPITAL',\n",
       "  'HOSPITAL',\n",
       "  'HOSPITAL',\n",
       "  'HOSPITAL'],\n",
       " 'ontology': ['http://data.bioontology.org/ontologies/MESH',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/SNOMEDCT',\n",
       "  'http://data.bioontology.org/ontologies/SNOMEDCT',\n",
       "  'http://data.bioontology.org/ontologies/PREMEDONTO',\n",
       "  'http://data.bioontology.org/ontologies/SNOMEDCT',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/SNOMEDCT',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/COVID-19',\n",
       "  'http://data.bioontology.org/ontologies/SNOMEDCT',\n",
       "  'http://data.bioontology.org/ontologies/PREMEDONTO',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/SNOMEDCT',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/SNOMEDCT',\n",
       "  'http://data.bioontology.org/ontologies/MESH',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/PREMEDONTO',\n",
       "  'http://data.bioontology.org/ontologies/MESH',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/HPIO',\n",
       "  'http://data.bioontology.org/ontologies/PREMEDONTO',\n",
       "  'http://data.bioontology.org/ontologies/MESH',\n",
       "  'http://data.bioontology.org/ontologies/SNOMEDCT',\n",
       "  'http://data.bioontology.org/ontologies/SNOMEDCT',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/MESH',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/SNOMEDCT',\n",
       "  'http://data.bioontology.org/ontologies/SNOMEDCT',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/PREMEDONTO',\n",
       "  'http://data.bioontology.org/ontologies/SNOMEDCT',\n",
       "  'http://data.bioontology.org/ontologies/MESH',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/SNOMEDCT',\n",
       "  'http://data.bioontology.org/ontologies/MESH',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/SNOMEDCT',\n",
       "  'http://data.bioontology.org/ontologies/MESH',\n",
       "  'http://data.bioontology.org/ontologies/SNOMEDCT',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/SNOMEDCT',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/COVID-19',\n",
       "  'http://data.bioontology.org/ontologies/SNOMEDCT',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/SNOMEDCT',\n",
       "  'http://data.bioontology.org/ontologies/MESH',\n",
       "  'http://data.bioontology.org/ontologies/SNOMEDCT',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/PREMEDONTO',\n",
       "  'http://data.bioontology.org/ontologies/MESH',\n",
       "  'http://data.bioontology.org/ontologies/MESH',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/COVID-19',\n",
       "  'http://data.bioontology.org/ontologies/HPIO',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/SNOMEDCT',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/PREMEDONTO',\n",
       "  'http://data.bioontology.org/ontologies/SNOMEDCT',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/SNOMEDCT',\n",
       "  'http://data.bioontology.org/ontologies/PREMEDONTO',\n",
       "  'http://data.bioontology.org/ontologies/MESH',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/MESH',\n",
       "  'http://data.bioontology.org/ontologies/COVID-19',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/MESH',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/SNOMEDCT',\n",
       "  'http://data.bioontology.org/ontologies/SNOMEDCT',\n",
       "  'http://data.bioontology.org/ontologies/MESH',\n",
       "  'http://data.bioontology.org/ontologies/PREMEDONTO',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/SNOMEDCT',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/MESH',\n",
       "  'http://data.bioontology.org/ontologies/PREMEDONTO',\n",
       "  'http://data.bioontology.org/ontologies/PREMEDONTO',\n",
       "  'http://data.bioontology.org/ontologies/SNOMEDCT',\n",
       "  'http://data.bioontology.org/ontologies/PREMEDONTO',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/SNOMEDCT',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/PREMEDONTO',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/SNOMEDCT',\n",
       "  'http://data.bioontology.org/ontologies/PREMEDONTO',\n",
       "  'http://data.bioontology.org/ontologies/PREMEDONTO',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/MESH',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/MESH',\n",
       "  'http://data.bioontology.org/ontologies/PREMEDONTO',\n",
       "  'http://data.bioontology.org/ontologies/SNOMEDCT',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/MESH',\n",
       "  'http://data.bioontology.org/ontologies/SNOMEDCT',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/SNOMEDCT',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/SNOMEDCT',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/MESH',\n",
       "  'http://data.bioontology.org/ontologies/OCHV',\n",
       "  'http://data.bioontology.org/ontologies/SNOMEDCT',\n",
       "  'http://data.bioontology.org/ontologies/PREMEDONTO'],\n",
       " 'annotatedClass': ['http://purl.bioontology.org/ontology/MESH/D009177',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#C0026941',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#8394',\n",
       "  'http://purl.bioontology.org/ontology/SNOMEDCT/58720004',\n",
       "  'http://purl.bioontology.org/ontology/SNOMEDCT/116680003',\n",
       "  'http://purl.obolibrary.org/obo/NCIT_C43461',\n",
       "  'http://purl.bioontology.org/ontology/SNOMEDCT/72906007',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#C0205214',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#20409',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#4722',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#C0015127',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#C0547042',\n",
       "  'http://purl.bioontology.org/ontology/SNOMEDCT/261183002',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#38892',\n",
       "  'http://purl.obolibrary.org/obo/HP_0002783',\n",
       "  'http://purl.bioontology.org/ontology/SNOMEDCT/70232002',\n",
       "  'http://purl.obolibrary.org/obo/NCIT_C64649',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#C0332183',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#29329',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#45080',\n",
       "  'http://purl.bioontology.org/ontology/SNOMEDCT/233606009',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#C0702135',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#C0332239',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#29360',\n",
       "  'http://purl.bioontology.org/ontology/SNOMEDCT/84422007',\n",
       "  'http://purl.bioontology.org/ontology/MESH/D000328',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#783',\n",
       "  'http://purl.obolibrary.org/obo/NCIT_C25227',\n",
       "  'http://purl.bioontology.org/ontology/MESH/D007239',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#6691',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#C0029235',\n",
       "  'http://www.semanticweb.org/ontologies/2010/5/22/Ontology1277229984000.owl#HPI:0000001',\n",
       "  'http://purl.obolibrary.org/obo/NCIT_C14250',\n",
       "  'http://purl.bioontology.org/ontology/STY/T001',\n",
       "  'http://purl.bioontology.org/ontology/STY/T001',\n",
       "  'http://purl.bioontology.org/ontology/SNOMEDCT/410607006',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#9023',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#C0026193',\n",
       "  'http://purl.bioontology.org/ontology/MESH/D033242',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#20364',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#8168',\n",
       "  'http://purl.bioontology.org/ontology/SNOMEDCT/255606000',\n",
       "  'http://purl.bioontology.org/ontology/SNOMEDCT/410599005',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#29373',\n",
       "  'http://purl.obolibrary.org/obo/NCIT_C74521',\n",
       "  'http://purl.bioontology.org/ontology/SNOMEDCT/405737000',\n",
       "  'http://purl.bioontology.org/ontology/MESH/D010612',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#C1319008',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#58191',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#9584',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#C0040586',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#12427',\n",
       "  'http://purl.bioontology.org/ontology/SNOMEDCT/13617004',\n",
       "  'http://purl.bioontology.org/ontology/MESH/D001988',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#2186',\n",
       "  'http://purl.bioontology.org/ontology/SNOMEDCT/4120002',\n",
       "  'http://purl.bioontology.org/ontology/MESH/D003440',\n",
       "  'http://purl.bioontology.org/ontology/SNOMEDCT/71186008',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#3477',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#C0010380',\n",
       "  'http://purl.bioontology.org/ontology/SNOMEDCT/32088001',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#22639',\n",
       "  'http://purl.obolibrary.org/obo/OGMS_0000117',\n",
       "  'http://purl.bioontology.org/ontology/SNOMEDCT/84387000',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#C0439663',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#34581',\n",
       "  'http://purl.bioontology.org/ontology/SNOMEDCT/385433004',\n",
       "  'http://purl.bioontology.org/ontology/MESH/D001996',\n",
       "  'http://purl.bioontology.org/ontology/SNOMEDCT/396285007',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#C0006285',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#2193',\n",
       "  'http://purl.obolibrary.org/obo/NCIT_C16880',\n",
       "  'http://purl.bioontology.org/ontology/MESH/Q000401',\n",
       "  'http://purl.bioontology.org/ontology/MESH/D009026',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#8258',\n",
       "  'http://purl.obolibrary.org/obo/OMIT_0010070',\n",
       "  'http://www.semanticweb.org/ontologies/2010/5/22/Ontology1277229984000.owl#HPI:0000022',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#C0021311',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#29330',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#C0332184',\n",
       "  'http://purl.bioontology.org/ontology/SNOMEDCT/89292003',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#29349',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#C0332218',\n",
       "  'http://purl.obolibrary.org/obo/NCIT_C129485',\n",
       "  'http://purl.bioontology.org/ontology/SNOMEDCT/52925006',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#3501',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#44058',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#C0441942',\n",
       "  'http://purl.bioontology.org/ontology/SNOMEDCT/439401001',\n",
       "  'http://purl.obolibrary.org/obo/NCIT_C15220',\n",
       "  'http://purl.bioontology.org/ontology/MESH/D003933',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#34948',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#39667',\n",
       "  'http://purl.bioontology.org/ontology/MESH/Q000175',\n",
       "  'http://purl.obolibrary.org/obo/OGMS_0000073',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#C0011900',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#3850',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#C0750484',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#C0521093',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#48316',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#37854',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#11217',\n",
       "  'http://purl.bioontology.org/ontology/MESH/D016133',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#9923',\n",
       "  'http://purl.bioontology.org/ontology/SNOMEDCT/258066000',\n",
       "  'http://purl.bioontology.org/ontology/SNOMEDCT/12645001',\n",
       "  'http://purl.bioontology.org/ontology/MESH/D005784',\n",
       "  'http://purl.obolibrary.org/obo/NCIT_C16847',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#C0020028',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#6283',\n",
       "  'http://purl.bioontology.org/ontology/SNOMEDCT/223616007',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#C0036243',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#11061',\n",
       "  'http://purl.bioontology.org/ontology/MESH/D012529',\n",
       "  'http://purl.obolibrary.org/obo/NCIT_C62574',\n",
       "  'http://purl.obolibrary.org/obo/NCIT_C38000',\n",
       "  'http://purl.bioontology.org/ontology/SNOMEDCT/104176001',\n",
       "  'http://purl.obolibrary.org/obo/NCIT_C25429',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#36938',\n",
       "  'http://purl.bioontology.org/ontology/SNOMEDCT/312848003',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#C0034037',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#10375',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#51185',\n",
       "  'http://purl.obolibrary.org/obo/NCIT_C48191',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#C0870705',\n",
       "  'http://purl.bioontology.org/ontology/SNOMEDCT/224699009',\n",
       "  'http://purl.obolibrary.org/obo/NCIT_C15206',\n",
       "  'http://purl.obolibrary.org/obo/NCIT_C63536',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#53173',\n",
       "  'http://purl.bioontology.org/ontology/MESH/D004813',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#C0014507',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#4543',\n",
       "  'http://purl.bioontology.org/ontology/MESH/Q000453',\n",
       "  'http://purl.obolibrary.org/obo/NCIT_C25398',\n",
       "  'http://purl.bioontology.org/ontology/SNOMEDCT/58147004',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#C0205210',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#20405',\n",
       "  'http://purl.bioontology.org/ontology/MESH/D003469',\n",
       "  'http://purl.bioontology.org/ontology/SNOMEDCT/703725008',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#C0010453',\n",
       "  'http://purl.bioontology.org/ontology/SNOMEDCT/61594008',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#C0700164',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#36586',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#C0456369',\n",
       "  'http://purl.bioontology.org/ontology/SNOMEDCT/277025001',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#44676',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#6268',\n",
       "  'http://purl.bioontology.org/ontology/MESH/D006761',\n",
       "  'http://sbmi.uth.tmc.edu/ontology/ochv#C0019994',\n",
       "  'http://purl.bioontology.org/ontology/SNOMEDCT/22232009',\n",
       "  'http://purl.obolibrary.org/obo/NCIT_C16696']}"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "one_item['chunks_paper'][0]['annotations'][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "item.store_db_main()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "len(item.db[1]['chunks_paper'][0]['annotations'][0]['annotatedClass'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "load_json_record(samp_r1)['paper_id']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "samp_r2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###  FINAL CODE TO ANNOTATED PAPERS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "len(item.db)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
